--- rpm-4.16.0/build/rpmbuild_internal.h~	2020-05-28 12:04:25.000000000 +0200
+++ rpm-4.16.0/build/rpmbuild_internal.h	2020-11-21 19:32:47.432243301 +0100
@@ -18,7 +18,7 @@
 #undef HTDATATYPE
 
 #define WHITELIST_NAME ".-_+%{}"
-#define WHITELIST_VERREL "._+%{}~^"
+#define WHITELIST_VERREL "._+%{}~^@"
 #define WHITELIST_EVR WHITELIST_VERREL "-:"
 #define LEN_AND_STR(_tag) (sizeof(_tag)-1), (_tag)
 
