--- rpm-4.4.9/macros.in	2009-08-03 18:07:50.426882379 +0300
+++ rpm/macros	2009-08-03 18:06:42.986896052 +0300
@@ -185,7 +185,7 @@
 #	Path to script that creates debug symbols in a /usr/lib/debug
 #	shadow tree.
 %__debug_install_post   \
-   %{_usrlibrpm}/find-debuginfo.sh %{_builddir}/%{?buildsubdir}\
+   %{_usrlibrpm}/find-debuginfo.sh '%{_builddir}/%{?buildsubdir}'\
 %{nil}
 
 #	Template for debug information sub-package.
