--- rpm-6.0.0/lib/rpmlead.hh~	2025-09-22 14:09:24.000000000 +0200
+++ rpm-6.0.0/lib/rpmlead.hh	2025-10-05 17:03:28.863330730 +0200
@@ -1,6 +1,10 @@
 #ifndef _H_RPMLEAD
 #define _H_RPMLEAD
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 /** \ingroup lead
  * \file rpmlead.h
  * Routines to read and write an rpm lead structure for a a package.
@@ -31,4 +35,8 @@
  */
 rpmRC rpmLeadRead(FD_t fd, char **emsg);
 
+#ifdef __cplusplus
+}
+#endif
+
 #endif	/* _H_RPMLEAD */
--- rpm-6.0.0/lib/signature.hh~	2025-09-22 14:09:24.000000000 +0200
+++ rpm-6.0.0/lib/signature.hh	2025-10-05 17:04:07.186664064 +0200
@@ -7,6 +7,10 @@
  */
 #include <rpm/rpmtypes.h>
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 /** \ingroup signature
  * Read (and verify header+payload size) signature header.
  * If an old-style signature is found, we emulate a new style one.
@@ -41,4 +45,8 @@
 			rpm_loff_t size, rpm_loff_t payloadSize, FD_t fd,
 			int rpmver);
 
+#ifdef __cplusplus
+}
+#endif
+
 #endif	/* H_SIGNATURE */
