--- rpm-5.4.15/rpmio/macro.c~	2018-08-19 13:11:28.000000000 +0200
+++ rpm-5.4.15/rpmio/macro.c	2018-08-19 13:18:39.102655981 +0200
@@ -3122,7 +3122,7 @@
     ssize_t nb;
     int rc = -1;
     unsigned char magic[13];
-#if defined(RPM_VENDOR_OPENPKG) || defined(RPM_VENDOR_FEDORA) || defined(RPM_VENDOR_MANDRIVA) /* extension-based-compression-detection */
+#if defined(RPM_VENDOR_OPENPKG) || defined(RPM_VENDOR_FEDORA) || defined(RPM_VENDOR_MANDRIVA) || defined(RPM_VENDOR_PLD) /* extension-based-compression-detection */
     size_t file_len;
 #endif
 
@@ -3139,7 +3139,7 @@
 	return 0;
 #endif
 
-#if defined(RPM_VENDOR_OPENPKG) || defined(RPM_VENDOR_FEDORA) || defined(RPM_VENDOR_MANDRIVA) /* extension-based-compression-detection */
+#if defined(RPM_VENDOR_OPENPKG) || defined(RPM_VENDOR_FEDORA) || defined(RPM_VENDOR_MANDRIVA) || defined(RPM_VENDOR_PLD) /* extension-based-compression-detection */
     file_len = strlen(file);
     if ((file_len > 4 && strcasecmp(file+file_len-4, ".tbz") == 0)
      || (file_len > 4 && strcasecmp(file+file_len-4, ".bz2") == 0)) {
@@ -3208,7 +3208,7 @@
      &&	magic[2] == 'Z' && magic[3] == 'O')	/* lzop */
 	*compressed = COMPRESSED_LZOP;
     else
-#if !defined(RPM_VENDOR_OPENPKG) && !defined(RPM_VENDOR_FEDORA) && !defined(RPM_VENDOR_MANDRIVA) /* extension-based-compression-detection */
+#if !defined(RPM_VENDOR_OPENPKG) && !defined(RPM_VENDOR_FEDORA) && !defined(RPM_VENDOR_MANDRIVA) && !defined(RPM_VENDOR_PLD) /* extension-based-compression-detection */
     /* XXX Ick, LZMA has no magic. See http://lkml.org/lkml/2005/6/13/285 */
     if (magic[ 9] == (unsigned char) 0x00 && magic[10] == (unsigned char) 0x00 &&
 	magic[11] == (unsigned char) 0x00 && magic[12] == (unsigned char) 0x00)	/* lzmash */
