--- rpm-4.16.0/configure.ac~	2020-11-22 19:47:14.000000000 +0100
+++ rpm-4.16.0/configure.ac	2020-11-22 19:49:03.093877834 +0100
@@ -76,6 +76,7 @@
 dnl Find some common programs
 dnl
 AC_PATH_PROGS(__7ZIP, [7zip 7za 7z], /usr/bin/7za, $MYPATH)
+AC_PATH_PROG(__BASH, bash, /bin/bash, $MYPATH)
 AC_PATH_PROG(__BZIP2, bzip2, /usr/bin/bzip2, $MYPATH)
 AC_PATH_PROG(__CAT, cat, /bin/cat, $MYPATH)
 AC_PATH_PROG(__CHGRP, chgrp, /bin/chgrp, $MYPATH)
@@ -91,6 +92,7 @@
 AC_PATH_PROG(__UNZIP, unzip, /usr/bin/unzip, $MYPATH)
 AC_PATH_PROG(__ID, id, /usr/bin/id, $MYPATH)
 AC_PATH_PROG(__INSTALL, install, /usr/bin/install, $MYPATH)
+AC_PATH_PROG(__LN, ln, /bin/ln, $MYPATH)
 AC_PATH_PROG(__LRZIP, lrzip, /usr/bin/lrzip, $MYPATH)
 AC_PATH_PROG(__LZIP, lzip, /usr/bin/lzip, $MYPATH)
 AC_PATH_PROG(__XZ, xz, /usr/bin/xz, $MYPATH)
@@ -104,6 +106,7 @@
 AC_PATH_PROG(__RSH, rsh, /usr/bin/rsh, $MYPATH)
 AC_PATH_PROG(__SED, sed, /bin/sed, $MYPATH)
 AC_PATH_PROG(__SEMODULE, semodule, /usr/bin/semodule, $MYPATH)
+AC_PATH_PROG(__SH, sh, /bin/sh, $MYPATH)
 AC_PATH_PROG(__SSH, ssh, /usr/bin/ssh, $MYPATH)
 AC_PATH_PROG(__TAR, tar, /bin/tar, $MYPATH)
 AC_PATH_PROG(__ZSTD, zstd, /usr/bin/zstd, $MYPATH)
--- rpm-4.16.0/macros.in~	2020-11-21 19:41:19.000000000 +0100
+++ rpm-4.16.0/macros.in	2020-11-22 19:45:24.182620411 +0100
@@ -6,6 +6,7 @@
 # should be added to /etc/rpm/macros, while per-user configuration should
 # be added to ~/.rpmmacros.
 #
+%_rpmversion		@VERSION@
 
 #==============================================================================
 # ---- A macro that expands to nothing.
@@ -25,6 +25,7 @@
 #
 %__7zip			@__7ZIP@
 %__awk			@AWK@
+%__bash			@__BASH@
 %__bzip2		@__BZIP2@
 %__cat			@__CAT@
 %__chgrp		@__CHGRP@
@@ -39,6 +40,7 @@
 %__id			@__ID@
 %__id_u			%{__id} -u
 %__install		@__INSTALL@
+%__ln			@__LN@
 %__ln_s			@LN_S@
 %__lrzip		@__LRZIP@
 %__lzip			@__LZIP@
@@ -54,6 +57,7 @@
 %__rsh			@__RSH@
 %__sed			@__SED@
 %__semodule		@__SEMODULE@
+%__sh			@__SH@
 %__ssh			@__SSH@
 %__tar			@__TAR@
 %__unzip		@__UNZIP@
