--- rpm-4.20.0/CMakeLists.txt~	2024-10-07 11:35:46.000000000 +0200
+++ rpm-4.20.0/CMakeLists.txt	2024-11-24 23:27:34.151396858 +0100
@@ -109,6 +109,7 @@
 	endif()
 
 	findutil(__7ZIP "7za;7z")
+	findutil(__BASH bash)
 	findutil(__BZIP2 bzip2)
 	findutil(__CAT cat)
 	findutil(__CHMOD chmod)
@@ -132,6 +133,7 @@
 	findutil(__PATCH patch)
 	findutil(__RM rm)
 	findutil(__SED sed)
+	findutil(__SH sh)
 	findutil(__TAR tar)
 	findutil(__UNZIP unzip)
 	findutil(__ZSTD zstd)
--- rpm-4.16.0/macros.in~	2020-11-21 19:41:19.000000000 +0100
+++ rpm-4.16.0/macros.in	2020-11-22 19:45:24.182620411 +0100
@@ -6,6 +6,7 @@
 # should be added to /etc/rpm/macros, while per-user configuration should
 # be added to ~/.rpmmacros.
 #
+%_rpmversion		@CMAKE_PROJECT_VERSION@
 
 #==============================================================================
 # ---- A macro that expands to nothing.
@@ -25,6 +25,7 @@
 #
 %__7zip			@__7ZIP@
 %__awk			@__AWK@
+%__bash			@__BASH@
 %__bzip2		@__BZIP2@
 %__cat			@__CAT@
 %__chmod		@__CHMOD@
@@ -54,6 +57,7 @@
 %__patch		@__PATCH@
 %__rm			@__RM@
 %__sed			@__SED@
+%__sh			@__SH@
 %__tar			@__TAR@
 %__unzip		@__UNZIP@
 %__zstd			@__ZSTD@
