--- rpm-4.16.1.3/scripts/find-debuginfo.sh.orig	2021-11-05 00:19:27.239390489 +0100
+++ rpm-4.16.1.3/scripts/find-debuginfo.sh	2021-11-05 00:21:27.102128386 +0100
@@ -352,7 +352,6 @@
 # Build a list of unstripped ELF files and their hardlinks
 touch "$temp/primary"
 find "$RPM_BUILD_ROOT" ! -path "${debugdir}/*.debug" -type f \
-     		     \( -perm -0100 -or -perm -0010 -or -perm -0001 \) \
 		     -print | LC_ALL=C sort |
 file -N -f - | sed -n -e 's/^\(.*\):[ 	]*.*ELF.*, not stripped.*/\1/p' |
 xargs --no-run-if-empty stat -c '%h %D_%i %n' |
