diff -urN rpm-4.5/macros.in rpm-4.5.new//macros.in
--- rpm-4.5/macros.in	2010-05-28 20:56:02.854032755 +0200
+++ rpm-4.5.new//macros.in	2010-05-28 20:59:04.177348180 +0200
@@ -277,6 +277,9 @@
 #
 #%distribution
 
+# PLD Linux Release
+%pld_release    th
+
 #	Configurable distribution URL, same as DistURL: tag in a specfile.
 #	The URL will be used to supply reliable information to tools like
 #	rpmfind.
@@ -1449,7 +1452,7 @@
 # Note: Used iff _use_internal_dependency_generator is non-zero. The
 # helpers are also used by %{_rpmhome}/rpmdeps {--provides|--requires}.
 %__libtool_provides	%{_rpmhome}/libtooldeps.sh --provides %{buildroot} %{name}
-%__libtool_requires	%{_rpmhome}/libtooldeps.sh --requires %{buildroot} %{name}
+%__libtool_requires	%{nil}
 
 #------------------------------------------------------------------------
 # pkgconfig(...) configuration.
@@ -1459,7 +1462,7 @@
 # Note: Used iff _use_internal_dependency_generator is non-zero. The
 # helpers are also used by %{_rpmhome}/rpmdeps {--provides|--requires}.
 %__pkgconfig_provides	%{_rpmhome}/pkgconfigdeps.sh --provides
-%__pkgconfig_requires	%{_rpmhome}/pkgconfigdeps.sh --requires
+%__pkgconfig_requires	%{nil}
 
 #------------------------------------------------------------------------
 # executable(...) configuration.
@@ -1482,6 +1485,9 @@
 # -feliminate-dwarf2-dups disabled until PR ld/3290 is fixed.
 
 %debugcflags	-O0 -g -Wall
+%debugcppflags	%{nil}
+%optldflags	-Wl,--as-needed -Wl,--no-copy-dt-needed-entries -Wl,-z,relro -Wl,-z,combreloc
+%optcppflags	-D_FORTIFY_SOURCE=2
 
 # Warning: those macros are overwritten by macros.build,
 #          left here for compatibility
