--- rpm-5.3.1/macros/macros.rpmbuild.in.wiget	2010-05-30 17:11:07.699227892 +0200
+++ rpm-5.3.1/macros/macros.rpmbuild.in	2010-05-30 17:12:37.700048249 +0200
@@ -102,6 +102,9 @@
 #
 #%distribution
 
+# PLD Linux Release
+%pld_release    th
+
 #	Configurable bug URL, same as BugURL: tag in a specfile.
 #	The URL will be used to supply reliable information to where
 #	to file bugs.
--- rpm-5.3.1/macros/macros.in.wiget	2010-05-30 17:10:50.702560935 +0200
+++ rpm-5.3.1/macros/macros.in	2010-05-30 17:10:50.705892723 +0200
@@ -956,7 +956,8 @@ pakietu oraz przy odpluskwianiu samego p
 # -feliminate-dwarf2-dups disabled until PR ld/3290 is fixed.
 
 %debugcflags	-O0 -g -Wall
-%optldflags	-Wl,--as-needed
+%optldflags	-Wl,--as-needed -Wl,-z,relro -Wl,-z,combreloc
+%optcppflags	-D_FORTIFY_SOURCE=2
 
 # Warning: those macros are overwritten by macros.build,
 #          left here for compatibility
