--- rpm-4.3/rpmio/rpmio.h.orig	2003-05-10 17:20:20.000000000 +0200
+++ rpm-4.3/rpmio/rpmio.h	2003-06-19 23:08:44.000000000 +0200
@@ -501,7 +501,7 @@
 int fdClose( /*@only@*/ void * cookie)
 	/*@globals errno, fileSystem, systemState, internalState @*/
 	/*@modifies *cookie, errno, fileSystem, systemState, internalState @*/;
-#define	fdCLose(_fd)		fdio->close(_fd)
+#define	fdClose(_fd)		fdio->close(_fd)
 
 /**
  */
