--- rpm-3.0.3.perl/configure.in	Fri Oct 22 14:23:17 1999
+++ rpm-3.0.3/configure.in	Fri Oct 22 14:47:01 1999
@@ -906,7 +906,7 @@
 export LIBS INCPATH CONFIG_SITE
 
 AC_CONFIG_SUBDIRS(popt)
-AC_OUTPUT([Makefile rpmrc macros macros.pld lib/Makefile
+AC_OUTPUT([Makefile rpmrc macros macros.pld macros.perl lib/Makefile
 	build/Makefile tools/Makefile scripts/Makefile
 	tests/Makefile tests/rpmrc tests/macros tests/hello-test/Makefile
 	misc/Makefile po/Makefile.in intl/Makefile
--- rpm-3.0.3.perl/macros.in	Fri Oct 22 14:23:17 1999
+++ rpm-3.0.3/macros.in	Fri Oct 22 14:59:08 1999
@@ -248,6 +248,8 @@
 #	%{perl_sitearch}/Image
 #	%dir %{perl_sitearch}/auto/Image
 #
+#
+# NOTE: %{perl_sitearch}, %{perl_archlib} and %{perl_sitelib} macros 
+#	has been moved to macros.perl (see info in this file for details).
+#
 %requires_eq()	%(LC_ALL="C" rpm -q --queryformat 'Requires:%%{NAME} = %%{VERSION}' %1| grep -v "is not")
-%perl_sitearch	%(eval "`perl -V:installsitearch`"; echo $installsitearch)
-%perl_archlib	%(eval "`perl -V:installarchlib`"; echo $installarchlib)
--- rpm-3.0.3.perl/macros.perl.in	Thu Jan  1 01:00:00 1970
+++ rpm-3.0.3/macros.perl.in	Fri Oct 22 15:08:15 1999
@@ -0,0 +1,11 @@
+# Perl specific macro definitions.
+# To make use of these macros insert the following line into your spec file:
+# %include @RPMCONFIGDIR@/macros.perl
+
+%define		__find_requires	@RPMCONFIGDIR@/find-perl-requires
+%define		__find_provides	@RPMCONFIGDIR@/find-perl-provides
+
+%define		perl_sitelib	%(eval "`perl -V:installsitelib`"; echo $installsitelib)
+%define		perl_sitearch	%(eval "`perl -V:installsitearch`"; echo $installsitearch)
+%define		perl_archlib	%(eval "`perl -V:installarchlib`"; echo $installarchlib)
+
--- rpm-3.0.3.perl/scripts/Makefile.am	Sat Jun  5 22:52:30 1999
+++ rpm-3.0.3/scripts/Makefile.am	Fri Oct 22 15:12:12 1999
@@ -4,7 +4,8 @@
 
 EXTRA_DIST = \
 	convertrpmrc.sh find-lang.sh freshen.sh getpo.sh \
-	perl.prov perl.req vpkg-provides.sh
+	perl.prov perl.req find-perl-requires find-perl-provides \
+	vpkg-provides.sh
 
 installprefix = $(DESTDIR)
 
@@ -13,5 +14,6 @@
 configdir = ${prefix}/lib/rpm
 config_SCRIPTS = \
 	convertrpmrc.sh find-lang.sh freshen.sh getpo.sh \
-	perl.prov perl.req vpkg-provides.sh
+	perl.prov perl.req find-perl-requires find-perl-provides \
+	vpkg-provides.sh
 
--- rpm-3.0.3.perl/scripts/find-perl-provides	Thu Jan  1 01:00:00 1970
+++ rpm-3.0.3/scripts/find-perl-provides	Fri Oct 22 12:21:39 1999
@@ -0,0 +1,8 @@
+#!/bin/sh
+ulimit -c 0
+
+filelist=`sed "s/['\"]/\\\&/g"`
+
+echo $filelist|/usr/lib/rpm/find-provides
+/usr/lib/rpm/perl.prov $filelist
+
--- rpm-3.0.3.perl/scripts/find-perl-requires	Thu Jan  1 01:00:00 1970
+++ rpm-3.0.3/scripts/find-perl-requires	Fri Oct 22 12:21:36 1999
@@ -0,0 +1,11 @@
+#!/bin/sh
+ulimit -c 0
+
+filelist=`sed "s/['\"]/\\\&/g"`
+
+requires="`echo $filelist|/usr/lib/rpm/find-requires`"
+requires_perl="`/usr/lib/rpm/perl.req $filelist`"
+requires_mod="`rpm -q --whatprovides --qf "%{NAME}\n"  $requires_perl 2>/dev/null`"
+echo "$requires
+$requires_perl
+$requires_mod"|  sort -u
--- rpm-3.0.3.perl/scripts/perl.prov	Sat Jun  5 22:52:31 1999
+++ rpm-3.0.3/scripts/perl.prov	Fri Oct 22 14:34:50 1999
@@ -30,12 +30,12 @@
   if (length($require{$module}) == 0) {
    print "perl($module)\n";
   } else {
-    print "perl($module)=$require{$module}\n";
+    print "perl($module) = $require{$module}\n";
 
     # we need to print it without the version number until the
     # requires syntax accepts version numbers correctly.
 
-    print "perl($module)\n";
+#    print "perl($module)\n";
   }
 }
 
--- rpm-3.0.3.perl/scripts/perl.req	Sat Jun  5 22:52:31 1999
+++ rpm-3.0.3/scripts/perl.req	Fri Oct 22 12:21:33 1999
@@ -26,16 +26,20 @@
 
 
 foreach $module (sort keys %require) {
   if (length($require{$module}) == 0) {
-    print "perl($module)\n";
+    if ($module =~ /^[0-9._]+$/) {
+      print "perl >= $module\n";
+    } else {
+      print "perl($module)\n";
+    }
   } else {
 
-    print "perl($module)>=$require{$module}\n";
+    print "perl($module) >= $require{$module}\n";
 
     # we need to print it without the version number until the
     # requires syntax accepts version numbers correctly.
 
-    print "perl($module)\n";
+#    print "perl($module)\n";
   }
 }
 
