--- rpm-4.0.4/configure.ac.wiget	Tue Feb 12 23:26:36 2002
+++ rpm-4.0.4/configure.ac	Tue Feb 12 23:27:46 2002
@@ -1314,6 +1314,7 @@
     rpmio/Makefile rpmdb/Makefile lib/Makefile build/Makefile
     scripts/Makefile scripts/macros.perl scripts/macros.php
     scripts/perl.req scripts/perl.prov
+    scripts/php.req scripts/php.prov
     scripts/macros.python tools/Makefile misc/Makefile doc/Makefile
     doc/manual/Makefile doc/fr/Makefile doc/ja/Makefile doc/ko/Makefile
     doc/pl/Makefile doc/ru/Makefile doc/sk/Makefile python/Makefile

