--- rpm-2.92/rpmrc.in.old	Sun Mar 14 01:35:00 1999
+++ rpm-2.92/rpmrc.in	Sat Mar 27 16:41:42 1999
@@ -7,23 +7,28 @@
 # whenever a new version of RPM is installed.
 
 dbpath:		@varprefix@/lib/rpm
-topdir:		@prefix@/src/redhat
+topdir:		@prefix@/src/rpm
 tmppath:	@tmpdir@
 gzipbin:	@GZIPBIN@
 bzip2bin:	@BZIP2BIN@
 defaultdocdir:  @prefix@/doc
 fixperms:	@FIXPERMS@
-rpmfilename:    %{ARCH}/%{NAME}-%{VERSION}-%{RELEASE}.%{ARCH}.rpm
+Vendor:         PLD          
+rpmfilename:    %{NAME}-%{VERSION}-%{RELEASE}.%{ARCH}.rpm
+macrofiles:	@RPMCONFIGDIR@/macros:@RPMCONFIGDIR@/macros.pld
 instchangelog:	5
 
 #############################################################
 
 # Please send new entries to rpm-list@redhat.com
 
+# Plesae send new entries to pld-list@pld.org.pl 
+
 #############################################################
 # Values for RPM_OPT_FLAGS for various platforms
 
-optflags: i386 -O2 -m486 -fno-strength-reduce
+optflags: i386 -O2 -mpentium
+optflags: i686 -O2 -mpentium
 optflags: alpha -O2
 optflags: sparc -O2
 optflags: sparc64 -O2
