--- rpm-4.1/rpmdb/rpmdb.h.wiget	Tue Sep 17 16:56:35 2002
+++ rpm-4.1/rpmdb/rpmdb.h	Thu Sep 19 00:19:26 2002
@@ -9,7 +9,7 @@
 
 #include <assert.h>
 #include <rpmlib.h>
-#include <db.h>
+#include <db4/db.h>
 
 /*@-exportlocal@*/
 /*@unchecked@*/
--- rpm-4.1/configure.ac.wiget	Thu Sep 19 00:14:53 2002
+++ rpm-4.1/configure.ac	Thu Sep 19 00:17:45 2002
@@ -437,10 +437,11 @@
 if test $withval = no ; then
 dnl ------------------ without internal db
 
-AC_CHECK_HEADERS(db3/db.h)
+AC_CHECK_HEADERS(db4/db.h db3/db.h)
 
 dnl Check for Berkeley db3 API.
 AC_CHECK_FUNC(db_create, [DBLIBSRCS="$DBLIBSRCS db3.c"],
+  AC_CHECK_LIB(db-4.0, db_create, [DBLIBSRCS="$DBLIBSRCS db3.c"; libdb3="-ldb-4.0"],
   AC_CHECK_LIB(db-3.2, db_create, [DBLIBSRCS="$DBLIBSRCS db3.c"; libdb3="-ldb-3.2"],
     AC_CHECK_LIB(db-3.1, db_create, [DBLIBSRCS="$DBLIBSRCS db3.c"; libdb3="-ldb-3.1"],
       AC_CHECK_LIB(db-3.0, db_create, [DBLIBSRCS="$DBLIBSRCS db3.c"; libdb3="-ldb-3.0"],
@@ -449,6 +450,7 @@
       )
     )
   )
+  )
 )
 
 if test X"$DBLIBSRCS" = X; then
