--- rpm-5.0.2/build/parsePreamble.c	2007-12-31 18:21:47.000000000 +0200
+++ rpm-5.0.2-epoch0/build/parsePreamble.c	2008-02-17 20:17:38.000000000 +0200
@@ -1082,6 +1082,21 @@
 	headerCopyTags(spec->packages->header, pkg->header,
 			(uint32_t *)copyTagsDuringParse);
 
+    /* Add Epoch: 0 to package header if it was not set by spec */
+    he->tag = RPMTAG_NAME;
+    if (headerGet(spec->packages->header, he, 0) == 0) {
+    	int num = 0;
+
+	he->tag = RPMTAG_EPOCH;
+	he->t = RPM_UINT32_TYPE;
+	he->p.ui32p = &num;
+	he->c = 1;
+	xx = headerPut(pkg->header, he, 0);
+
+	/* also declare %{epoch} to be same */
+	addMacro(spec->macros, "epoch", NULL, "0", RMIL_SPEC);
+    }
+
     if (checkForRequired(pkg->header, NVR) != RPMRC_OK)
 	return RPMRC_FAIL;
 
