--- rpm-4.4.9/macros.in.~1~	2008-10-20 22:23:45.720379526 +0300
+++ rpm-4.4.9/macros.in	2008-10-20 22:24:22.614492828 +0300
@@ -358,7 +358,7 @@
 #		"w9.lzdio"	lzma level 9.
 #
 %_source_payload	w9.gzdio
-%_binary_payload	w9.lzdio
+%_binary_payload	w9.bzdio
 
 #	Archive formats to use for source/binary package payloads.
 #		"cpio"		cpio archive (default)
@@ -1438,8 +1438,8 @@
 #
 # Note: Used iff _use_internal_dependency_generator is non-zero. The
 # helpers are also used by %{_usrlibrpm}/rpmdeps {--provides|--requires}.
-%__libtool_provides	%{_usrlibrpm}/libtooldeps.sh --provides %{buildroot} %{name}
-%__libtool_requires	%{_usrlibrpm}/libtooldeps.sh --requires %{buildroot} %{name}
+#%__libtool_provides	%{_usrlibrpm}/libtooldeps.sh --provides %{buildroot} %{name}
+#%__libtool_requires	%{_usrlibrpm}/libtooldeps.sh --requires %{buildroot} %{name}
 
 #------------------------------------------------------------------------
 # pkgconfig(...) configuration.
@@ -1448,8 +1448,8 @@
 #
 # Note: Used iff _use_internal_dependency_generator is non-zero. The
 # helpers are also used by %{_usrlibrpm}/rpmdeps {--provides|--requires}.
-%__pkgconfig_provides	%{_usrlibrpm}/pkgconfigdeps.sh --provides
-%__pkgconfig_requires	%{_usrlibrpm}/pkgconfigdeps.sh --requires
+#%__pkgconfig_provides	%{_usrlibrpm}/pkgconfigdeps.sh --provides
+#%__pkgconfig_requires	%{_usrlibrpm}/pkgconfigdeps.sh --requires
 
 #------------------------------------------------------------------------
 # executable(...) configuration.
@@ -1470,10 +1470,10 @@
 #
 # Note: Used if _use_internal_dependency_generator is non-zero. The
 # helper is also used by %{_uselibrpm}/rpmdeps --provides
-%__mimetype_provides %{_usrlibrpm}/mimetypedeps.sh --provides
+#%__mimetype_provides %{_usrlibrpm}/mimetypedeps.sh --provides
 
 # PLD rpm macros
-%_enable_debug_packages 1
+%_enable_debug_packages 0
 
 #-----------------------------------------------------------------
 # CFLAGS and LDFLAGS used to build
