--- rpm-4.0.4/rpmpopt.in.wiget	Fri Feb 15 09:45:59 2002
+++ rpm-4.0.4/rpmpopt.in	Fri Feb 15 09:48:54 2002
@@ -171,26 +171,26 @@
 rpm	exec --target		rpmb --target
 rpm	exec --short-circuit	rpmb --short-circuit
 
-rpm	exec --initdb		rpmd --initdb
-rpm	exec --rebuilddb	rpmd --rebuilddb
-rpm	exec --verifydb		rpmd --verifydb
+#rpm	exec --initdb		rpmd --initdb
+#rpm	exec --rebuilddb	rpmd --rebuilddb
+#rpm	exec --verifydb		rpmd --verifydb
 
 #rpm	exec -e			rpme -e
 #rpm	exec --erase		rpme --erase
 #rpm	exec -U			rpmu -U
 #rpm	exec --upgrade		rpmu --upgrade
 
-rpm	exec --addsign		rpmk --addsign
-rpm	exec -K			rpmk -K
-rpm	exec --checksig		rpmk --checksig
-rpm	exec --resign		rpmk --resign
-
-rpm	exec -q			rpmq -q
-rpm	exec --query		rpmq --query
-rpm	exec --querytags	rpmq --querytags
-rpm	exec -V			rpmv -V
-rpm	exec -y			rpmv -y
-rpm	exec --verify		rpmv --verify
+#rpm	exec --addsign		rpmk --addsign
+#rpm	exec -K			rpmk -K
+#rpm	exec --checksig		rpmk --checksig
+#rpm	exec --resign		rpmk --resign
+
+#rpm	exec -q			rpmq -q
+#rpm	exec --query		rpmq --query
+#rpm	exec --querytags	rpmq --querytags
+#rpm	exec -V			rpmv -V
+#rpm	exec -y			rpmv -y
+#rpm	exec --verify		rpmv --verify
 
 rpm	alias --with		--define "_with_!#:+       --with-!#:+" \
 	--POPTdesc=$"enable configure <option> for build" \
@@ -213,6 +213,11 @@
 rpmb	alias --without		--define "_without_!#:+    --without-!#:+" \
 	--POPTdesc=$"disable configure <option> for build" \
 	--POPTargs=$"<option>"
+# Build policies enabled from command line. Last policy applies.
+rpmb alias --buildpolicy --define '__os_install_post @RPMCONFIGDIR@/brp-!#:+' \
+	--POPTdesc=$"set buildroot <policy> (e.g. compress man pages)" \
+	--POPTargs=$"<policy>"
+
 #==============================================================================
 rpmbuild	alias --dbpath		--define '_dbpath !#:+' \
 	--POPTdesc=$"use <dir> as the directory for the database" \
@@ -228,6 +233,11 @@
 rpmbuild alias --without	--define "_without_!#:+  --without-!#:+" \
 	--POPTdesc=$"disable configure <option> for build" \
 	--POPTargs=$"<option>"
+# Build policies enabled from command line. Last policy applies.
+rpmbuild alias --buildpolicy --define '__os_install_post @RPMCONFIGDIR@/brp-!#:+' \
+	--POPTdesc=$"set buildroot <policy> (e.g. compress man pages)" \
+	--POPTargs=$"<policy>"
+
 #==============================================================================
 rpmd	alias --dbpath		--define '_dbpath !#:+' \
 	--POPTdesc=$"use <dir> as the directory for the database" \
