--- rpm-5.4.10/build/files.c~	2013-01-24 14:24:39.161982722 +0100
+++ rpm-5.4.10/build/files.c	2013-01-24 14:27:50.620325354 +0100
@@ -2363,6 +2363,11 @@
     appendLineStringBuf(docScript, "export LC_ALL=C");
     appendStringBuf(docScript, "export ");
     appendLineStringBuf(docScript, sdenv);
+    appendStringBuf(docScript, "if test -d \"$");
+    appendStringBuf(docScript, sdenv);
+    appendStringBuf(docScript, "\" ; then rmdir \"$");
+    appendStringBuf(docScript, sdenv);
+    appendLineStringBuf(docScript, "\" ; fi");
     appendLineStringBuf(docScript, mkdocdir);
 
     for (ARGV_const_t fn = sd->files; fn && *fn; fn++) {
