--- rpm-4.0.2/build/files.c.orig	Thu Jul 25 15:57:52 2002
+++ rpm-4.0.2/build/files.c	Thu Jul 25 16:40:55 2002
@@ -1548,7 +1548,9 @@
     /* Now process special doc, if there is one */
     if (specialDoc) {
 	if (installSpecialDoc) {
-	    doScript(spec, RPMBUILD_STRINGBUF, "%doc", pkg->specialDoc, test);
+	    int rc = doScript(spec, RPMBUILD_STRINGBUF, "%doc", pkg->specialDoc, test);
+	    if (rc)
+	        fl.processingFailed = rc;
 	}
 
 	/* Reset for %doc */
