--- rpm-4.3/rpmio/Makefile.am.wiget	2003-06-01 07:56:41.000000000 +0200
+++ rpm-4.3/rpmio/Makefile.am	2003-06-11 02:02:20.000000000 +0200
@@ -43,7 +43,7 @@
 	if test X"@WITH_BEECRYPT_SUBDIR@" != X; then \
 	${MAKE} -C $(top_builddir)/@WITH_BEECRYPT_SUBDIR@ listobjs ; \
 	for lo in $(BEECRYPTLOBJS); do \
-	  [ -f $$lo ] || $(LN_S) $(top_builddir)/@WITH_BEECRYPT_SUBDIR@/$$lo $$lo ; \
+	  [ -f $$lo ] || $(LN_S) $(top_builddir)/@WITH_BEECRYPT_SUBDIR@/$$lo $$lo ; true ; \
 	done \
 	fi
 	touch $@
