--- rpm-4.4.8/macros.in	2007-04-08 20:36:08.925632417 +0200
+++ rpm-4.4.9/macros.in	2008-01-25 02:16:46.248708080 +0200
@@ -206,7 +206,7 @@
 %endif\
 %{nil}
 
-%_defaultdocdir		%{_usr}/doc
+%_defaultdocdir		%{_usr}/share/doc
 
 #	The path to the pgp executable (legacy, use %{__pgp} instead).
 %_pgpbin		%{__pgp}
@@ -233,12 +233,12 @@
 #	The directory where newly built source packages will be written.
 %_srcrpmdir		%{_topdir}/SRPMS
 
-#	Directory where temporaray files can be created.
-%_tmppath		%{_var}/tmp
+#	Directory where temporary files can be created.
+%_tmppath		%(echo "${TMPDIR:-/tmp}")
 %tmpdir			%{_tmppath}
 
 #	Path to top of build area.
-%_topdir		%{_usrsrc}/rpm
+%_topdir		%(echo $HOME)/rpm
 
 #==============================================================================
 # ---- Optional rpmrc macros.
@@ -248,7 +248,7 @@
 #	Configurable build root path, same as BuildRoot: in a specfile.
 #	(Note: the configured macro value will override the spec file value).
 #
-%buildroot		%{_tmppath}/%{name}-root
+%buildroot		%{_tmppath}/%{name}-%{version}-root-%(id -u -n)
 
 #	The sub-directory (relative to %{_builddir}) where sources are compiled.
 #	This macro is set after processing %setup, either explicitly from the
@@ -347,8 +347,8 @@
 #		"w9.bzdio"	bzip2 level 9.
 #		"w9.lzdio"	lzma level 9.
 #
-#%_source_payload	w9.gzdio
-#%_binary_payload	w9.gzdio
+%_source_payload	w9.gzdio
+%_binary_payload	w9.lzdio
 
 #	Archive formats to use for source/binary package payloads.
 #		"cpio"		cpio archive (default)
@@ -498,7 +498,7 @@
 
 #
 # Path to magic file used for file classification.
-%_rpmfc_magic_path	%{_usr}/lib/rpm/magic
+%_rpmfc_magic_path	/usr/share/file/magic
 
 #==============================================================================
 # ---- Database configuration macros.
@@ -991,9 +991,12 @@
   %{?_javaclasspath:CLASSPATH=\"%{_javaclasspath}\"\
   export CLASSPATH}\
   unset PERL_MM_OPT || :\
-  LANG=C\
-  export LANG\
-  unset DISPLAY || :\
+  export LC_ALL=C\
+  export LANG=C\
+  unset LINGUAS ||:\
+  unset LANGUAGE ||:\
+  unset LC_MESSAGES ||:\
+  unset DISPLAY ||:\
   \
   %{verbose:set -x}%{!verbose:exec > /dev/null}\
   umask 022\
@@ -1127,17 +1130,17 @@
 %_exec_prefix		%{_prefix}
 %_bindir		%{_exec_prefix}/bin
 %_sbindir		%{_exec_prefix}/sbin
-%_libexecdir		%{_exec_prefix}/libexec
+%_libexecdir		%{_exec_prefix}/lib
 %_datadir		%{_prefix}/share
-%_sysconfdir		%{_prefix}/etc
-%_sharedstatedir	%{_prefix}/com
-%_localstatedir		%{_prefix}/var
+%_sysconfdir		/etc
+%_sharedstatedir	/var/lib
+%_localstatedir		/var
 %_lib			lib
 %_libdir		%{_exec_prefix}/%{_lib}
 %_includedir		%{_prefix}/include
 %_oldincludedir		/usr/include
-%_infodir		%{_prefix}/info
-%_mandir		%{_prefix}/man
+%_infodir		%{_prefix}/share/info
+%_mandir		%{_prefix}/share/man
 %_localedir		%{_datadir}/locale
 
 #==============================================================================
@@ -1429,6 +1432,24 @@
 #%__executable_provides	%{_usrlibrpm}/executabledeps.sh --provides
 #%__executable_requires	%{_usrlibrpm}/executabledeps.sh --requires
-%__scriptlet_requires	/bin/bash --rpm-requires
+#%__scriptlet_requires	/bin/bash --rpm-requires
 
+# PLD rpm macros
+%_enable_debug_packages 1
+
+#-----------------------------------------------------------------
+# CFLAGS and LDFLAGS used to build
+
+%debuginfocflags	%{expand:%%define __dic_%{?_enable_debug_packages} 1}%{?__dic_1: -gdwarf-2 -g2}%{expand:%%undefine __dic_%{?_enable_debug_packages}}
+# -feliminate-dwarf2-dups disabled until PR ld/3290 is fixed.
+
+%debugcflags	-O0 -g -Wall
+%optldflags	-Wl,--as-needed -Wl,-z,relro -Wl,-z,-combreloc
+
+# Warning: those macros are overwritten by macros.build,
+#          left here for compatibility
+%rpmcflags	%{?debug:%debugcflags}%{!?debug:%optflags}%{?debuginfocflags}
+%rpmcxxflags	%{rpmcflags}
+%rpmldflags	%{!?no_build_with_as_needed:-Wl,--as-needed}
+  
 # \endverbatim
 #*/
--- rpm-4.4.8/configure.ac.orig	2007-04-08 20:24:47.210783702 +0200
+++ rpm-4.4.8/configure.ac	2007-04-08 20:32:18.216485052 +0200
@@ -1462,7 +1462,7 @@
 	[Full path to rpm system configuration directory (usually /etc/rpm)])
 AC_SUBST(SYSCONFIGDIR)
 
-MACROFILES="${USRLIBRPM}/macros:${USRLIBRPM}/%{_target}/macros:${SYSCONFIGDIR}/macros.*:${SYSCONFIGDIR}/macros:${SYSCONFIGDIR}/%{_target}/macros:~/.rpmmacros"
+MACROFILES="${USRLIBRPM}/macros:${USRLIBRPM}/macros.build:${USRLIBRPM}/%{_target}/macros:${SYSCONFIGDIR}/macros.*:${SYSCONFIGDIR}/macros:${SYSCONFIGDIR}/%{_target}/macros:~/etc/.rpmmacros:~/.rpmmacros"
 AC_DEFINE_UNQUOTED(MACROFILES, "$MACROFILES",
 	[Colon separated paths of macro files to read.])
 AC_SUBST(MACROFILES)
