--- rpm-5.4.10/macros/gstreamer.in.gst	2012-09-26 08:59:44.842368772 +0200
+++ rpm-5.4.10/macros/gstreamer.in	2012-09-26 08:59:48.542348566 +0200
@@ -2,7 +2,7 @@
 # To make use of these macros insert the following line into your spec file:
 # %{load:%{_usrlibrpm}/macros.d/gstreamer}
 
-%__gst_inspect		@__GST_INSPECT
+%__gst_inspect		@__GST_INSPECT@
 
 # Path to scripts to autogenerate gstreamer package dependencies,
 #
