--- rpm-5.4.16/lua/linit.c~	2016-03-16 00:26:03.000000000 +0200
+++ rpm-5.4.16/lua/linit.c	2016-03-16 00:26:42.479945350 +0200
@@ -13,6 +13,8 @@
 #include "lualib.h"
 #include "lauxlib.h"
 
+#include "lposix.h"
+#include "lrexlib.h"
 
 static const luaL_Reg lualibs[] = {
   {"", luaopen_base},
@@ -55,6 +55,9 @@
 #if defined(LUA_COMPAT_BITLIB)
   {LUA_BITLIBNAME, luaopen_bit32},
 #endif
+  {"rex_posix", luaopen_rex_posix},
+  {"rex_pcre", luaopen_rex_pcre},
+  {"posix", luaopen_posix},
   {NULL, NULL}
 };
 
