--- rpm-4.0.4/lib/misc.c.wiget	Sun Jan 20 23:10:19 2002
+++ rpm-4.0.4/lib/misc.c	Tue Feb 12 22:41:39 2002
@@ -514,6 +514,11 @@
     const char * path;
     const char * globURL;
     char * globRoot = NULL;
+#ifdef ENABLE_NLS	
+	char * old_collate = NULL;
+	char * old_ctype = NULL;
+	char * t;
+#endif
     size_t maxb, nb;
     glob_t gl;
     int ut;
@@ -524,6 +529,17 @@
     if (rc)
 	return rc;
 
+#ifdef ENABLE_NLS
+	t = setlocale(LC_COLLATE, NULL);
+	if (t)
+		old_collate = strdup(t);
+	t = setlocale(LC_CTYPE, NULL);
+	if (t)
+		old_ctype = strdup(t);
+	setlocale(LC_COLLATE, "C");
+	setlocale(LC_CTYPE, "C");
+#endif
+	
     for (j = 0; j < ac; j++) {
 	if (!myGlobPatternP(av[j])) {
 	    if (argc == 0)
@@ -609,6 +625,16 @@
 
 
 exit:
+#ifdef ENABLE_NLS
+    if (old_collate) {
+	setlocale(LC_COLLATE, old_collate);
+	free(old_collate);
+    }
+    if (old_ctype) {
+	setlocale(LC_CTYPE, old_ctype);
+	free(old_ctype);
+    }
+#endif
     av = _free(av);
     if (rc || argvPtr == NULL) {
 	if (argv != NULL)
