--- rpm-4.4.9/Doxyfile.in.orig	2007-05-13 05:20:43.000000000 +0200
+++ rpm-4.4.9/Doxyfile.in	2007-05-22 17:41:12.107028314 +0200
@@ -504,7 +504,6 @@
                          @top_srcdir@/doc/manual/spec \
                          @top_srcdir@/doc/manual/triggers \
                          @top_srcdir@/doc/manual/tsort \
-                         @top_srcdir@/file/src/tar.h \
                          @top_srcdir@/lib/cpio.c \
                          @top_srcdir@/lib/cpio.h \
                          @top_srcdir@/lib/depends.c \
