--- rpm-5.4.16/macros/macros.in~	2016-04-20 16:52:06.000000000 +0300
+++ rpm-5.4.16/macros/macros.in	2016-04-20 16:52:40.148730512 +0300
@@ -1054,11 +1054,11 @@
 
 #------------------------------------------------------------------------
 # cmake(...) configuration
-#%%{load:%{_usrlibrpm}/macros.d/cmake}
+%{load:%{_usrlibrpm}/macros.d/cmake}
 
 #------------------------------------------------------------------------
 # gstreamer(...) configuration.
-#%%{load:%{_usrlibrpm}/macros.d/gstreamer}
+%{load:%{_usrlibrpm}/macros.d/gstreamer}
 
 #------------------------------------------------------------------------
 # haskell(...) configuration.
@@ -1066,43 +1066,43 @@
 
 #------------------------------------------------------------------------
 # perl(...) configuration
-#%%{load:%{_usrlibrpm}/macros.d/perl}
+%{load:%{_usrlibrpm}/macros.d/perl}
 
 #------------------------------------------------------------------------
 # python(...) configuration.
-#%%{load:%{_usrlibrpm}/macros.d/python}
+%{load:%{_usrlibrpm}/macros.d/python}
 
 #------------------------------------------------------------------------
 # php(...) configuration.
-#%%{load:%{_usrlibrpm}/macros.d/php}
+%{load:%{_usrlibrpm}/macros.d/php}
 
 #------------------------------------------------------------------------
 # java(...) configuration.
-#%%{load:%{_usrlibrpm}/macros.d/java}
+%{load:%{_usrlibrpm}/macros.d/java}
 
 #------------------------------------------------------------------------
 # libtool(...) configuration.
-#%%{load:%{_usrlibrpm}/macros.d/libtool}
+%{load:%{_usrlibrpm}/macros.d/libtool}
 
 #------------------------------------------------------------------------
 # pkgconfig(...) configuration.
-#%%{load:%{_usrlibrpm}/macros.d/pkgconfig}
+%{load:%{_usrlibrpm}/macros.d/pkgconfig}
 
 #------------------------------------------------------------------------
 # mono(...) configuration.
-#%%{load:%{_usrlibrpm}/macros.d/mono}
+%{load:%{_usrlibrpm}/macros.d/mono}
 
 #------------------------------------------------------------------------
 # ruby(...) configuration.
-#%%{load:%{_usrlibrpm}/macros.d/ruby}
+%{load:%{_usrlibrpm}/macros.d/ruby}
 
 #------------------------------------------------------------------------
 # tcl(...) configuration.
-#%%{load:%{_usrlibrpm}/macros.d/tcl}
+%{load:%{_usrlibrpm}/macros.d/tcl}
 
 #------------------------------------------------------------------------
 # typelib(...) configuration.
-#%%{load:%{_usrlibrpm}/macros.d/typelib}
+%{load:%{_usrlibrpm}/macros.d/typelib}
 
 # \endverbatim
 #*/
