--- rpm-4.4.9/lib/rpmfc.c.orig	2007-05-22 20:19:05.990914000 +0200
+++ rpm-4.4.9/lib/rpmfc.c	2007-05-22 22:12:38.775152636 +0200
@@ -875,7 +875,14 @@
 	if (is_executable)
 	    if (fc->findreq)
 	    xx = rpmfcHelper(fc, 'R', "mono", fc->noautoreq, fc->noautoreq_c);
+    } else
+    if (fc->fcolor->vals[fc->ix] & RPMFC_JAVA) {
+	if (fc->findprov)
+	    xx = rpmfcHelper(fc, 'P', "java", fc->noautoprov, fc->noautoprov_c);
+	if (fc->findreq)
+	    xx = rpmfcHelper(fc, 'R', "java", fc->noautoreq, fc->noautoreq_c);
     }
+
     return 0;
 }
 
--- rpm-4.4.9/macros.in.orig	2007-05-24 21:20:49.657284000 +0200
+++ rpm-4.4.9/macros.in	2007-05-24 21:22:32.823163996 +0200
@@ -1416,8 +1416,8 @@
 #
 # Note: Used iff _use_internal_dependency_generator is non-zero. The
 # helpers are also used by %{_rpmhome}/rpmdeps {--provides|--requires}.
-%__java_provides	%{_rpmhome}/javadeps.sh --provides
-%__java_requires	%{_rpmhome}/javadeps.sh --requires
+%__java_provides	%{nil}
+%__java_requires	%{nil}
 
 #------------------------------------------------------------------------
 # libtool(...) configuration.
