--- rpmrc.in~	Sat Jul  3 16:52:06 1999
+++ rpmrc.in	Sat Jul 10 00:08:09 1999
@@ -4,10 +4,15 @@
 
 # Please send new entries to rpm-list@redhat.com
 
+# Please send new entries to pld-list@pld.org.pl
+
 #############################################################
 # Values for RPM_OPT_FLAGS for various platforms
 
-optflags: i386 -O2 -m486 -fno-strength-reduce
+optflags: i386 -O2 -march=i386
+optflags: i486 -O2 -march=i386
+optflags: i586 -O2 -march=pentium
+optflags: i686 -O2 -march=pentiumpro
 optflags: alpha -O2
 optflags: sparc -O2
 optflags: sparc64 -O2
@@ -253,7 +258,7 @@
 #############################################################
 #macrofiles:	@RPMCONFIGDIR@/macros:@SYSCONFIGDIR@/macros:~/.rpmmacros
 
-macrofiles:	@RPMCONFIGDIR@/macros:@RPMCONFIGDIR@/%{_target}/macros:@SYSCONFIGDIR@/macros:@SYSCONFIGDIR@/%{_target}/macros:~/.rpmmacros
+macrofiles:	@RPMCONFIGDIR@/macros:@RPMCONFIGDIR@/%{_target}/macros:@RPMCONFIGDIR@/macros.pld:@SYSCONFIGDIR@/macros:@SYSCONFIGDIR@/%{_target}/macros:~/.rpmmacros
 
 #include: @RPMCONFIGDIR@/%{_target}/rpmrc
 
