diff -Nru rpm-3.0.6/rpmrc.in rpm-3.0.6.new/rpmrc.in
--- rpm-3.0.6/rpmrc.in	Wed Sep 13 19:55:15 2000
+++ rpm-3.0.6.new/rpmrc.in	Thu Oct 12 15:49:35 2000
@@ -8,7 +8,7 @@
 #############################################################
 # Values for RPM_OPT_FLAGS for various platforms
 
-optflags: i386 -O2 -m486 -fno-strength-reduce
+optflags: i386 -O2 -march=i386
 optflags: i486 -O2 -march=i486
 optflags: i586 -O2 -march=i586
 optflags: i686 -O2 -march=i686
@@ -289,5 +289,5 @@
 buildarch_compat: ia64: noarch
 buildarch_compat: s390: noarch
 
-macrofiles:	@RPMCONFIGDIR@/macros:@RPMCONFIGDIR@/%{_target}/macros:@SYSCONFIGDIR@/macros.specspo:@SYSCONFIGDIR@/macros:@SYSCONFIGDIR@/%{_target}/macros:~/.rpmmacros
+macrofiles:	@RPMCONFIGDIR@/macros:@RPMCONFIGDIR@/macros.pld:@RPMCONFIGDIR@/%{_target}/macros:@SYSCONFIGDIR@/macros.specspo:@SYSCONFIGDIR@/macros:@SYSCONFIGDIR@/%{_target}/macros:~/.rpmmacros
 
