--- rpm-5.4.18/lua/Makefile.am~	2016-08-06 20:39:54.000000000 +0300
+++ rpm-5.4.18/lua/Makefile.am	2017-07-18 19:44:32.874014686 +0300
@@ -326,7 +326,10 @@
 	lzio.c
 
 liblua_la_DEPENDENCIES =
-liblua_la_LIBADD = @WITH_LUA_SUBDIR_LIB@ -lm -lexpat
+liblua_la_LIBADD = @WITH_LUA_SUBDIR_LIB@ -lm
+if USE_LUA_CRYPTO
+liblua_la_LIBADD += -lexpat
+endif
 liblua_la_LDFLAGS = -no-undefined $(LDFLAGS)
 
 liblua.la: $(liblua_la_OBJECTS) $(liblua_la_DEPENDENCIES) 
