--- rpm-2.93/rpmrc.in.rpmrc	Sun Apr  4 22:10:34 1999
+++ rpm-2.93/rpmrc.in	Thu Apr  8 18:32:21 1999
@@ -4,10 +4,15 @@
 
 # Please send new entries to rpm-list@redhat.com
 
+# Plesae send new entries to pld-list@pld.org.pl
+
 #############################################################
 # Values for RPM_OPT_FLAGS for various platforms
 
-optflags: i386 -O2 -m486 -fno-strength-reduce
+optflags: i386 -O2
+optflags: i486 -O2
+optflags: i586 -O2 -mpentium
+optflags: i686 -O2 -mpentium
 optflags: alpha -O2
 optflags: sparc -O2
 optflags: sparc64 -O2
@@ -187,6 +192,6 @@
 #############################################################
 #macrofiles:	@RPMCONFIGDIR@/macros:@SYSCONFIGDIR@/macros:~/.rpmmacros
 
-macrofiles:	@RPMCONFIGDIR@/macros:@RPMCONFIGDIR@/%{_target}/macros:@SYSCONFIGDIR@/macros:@SYSCONFIGDIR@/%{_target}/macros:~/.rpmmacros
+macrofiles:	@RPMCONFIGDIR@/macros:@RPMCONFIGDIR@/%{_target}/macros:@RPMCONFIGDIR@/macros.pld:@SYSCONFIGDIR@/macros:@SYSCONFIGDIR@/%{_target}/macros:~/.rpmmacros
 
 #include: @RPMCONFIGDIR@/%{_target}/rpmrc
