--- rpm-4.0.4/scripts/Makefile.am.wiget	Mon Feb  4 01:11:15 2002
+++ rpm-4.0.4/scripts/Makefile.am	Tue Feb 12 21:32:36 2002
@@ -11,6 +11,7 @@
 	magic.prov magic.req perl.prov perl.req rpmdiff rpmdiff.cgi \
 	rpm.daily rpm.log rpm.xinetd rpm2cpio.sh \
 	sql.prov sql.req tcl.req trpm u_pkg.sh \
+ 	find-perl-requires find-perl-provides \
 	vpkg-provides.sh vpkg-provides2.sh
 
 installprefix = $(DESTDIR)
@@ -27,4 +28,5 @@
 	magic.prov magic.req perl.prov perl.req rpmdiff rpmdiff.cgi \
 	rpm.daily rpm.log rpm.xinetd rpm2cpio.sh \
 	sql.prov sql.req tcl.req trpm u_pkg.sh \
+ 	find-perl-requires find-perl-provides \
 	vpkg-provides.sh vpkg-provides2.sh
--- rpm-4.0.4/scripts/perl.prov.wiget	Thu Jan 31 23:22:06 2002
+++ rpm-4.0.4/scripts/perl.prov	Tue Feb 12 21:25:39 2002
@@ -47,7 +47,9 @@
 
 if ("@ARGV") {
   foreach (@ARGV) {
-    process_file($_);
+     if (! m=(/(doc|usr/src)/|\.(so|gz|ph|pod|h|html)$)=) {
+      process_file($_);
+    }
   }
 } else {
 
@@ -55,7 +57,9 @@
   # contents of the file.
 
   foreach (<>) {
-    process_file($_);
+     if (! m=(/(doc|usr/src)/|\.(so|gz|ph|pod|h|html)$)=) {
+      process_file($_);
+    }
   }
 }
 
--- rpm-4.0.4/scripts/perl.req.wiget	Thu Jan 31 21:32:12 2002
+++ rpm-4.0.4/scripts/perl.req	Tue Feb 12 21:37:11 2002
@@ -1,4 +1,4 @@
-#!/usr/bin/perl
+#!@PERL@
 
 # RPM (and it's source code) is covered under two separate licenses. 
 
@@ -41,7 +41,11 @@
 
 if ("@ARGV") {
   foreach (@ARGV) {
-    process_file($_);
+    if (m=/usr/(sbin|bin|lib|share|X11R6/(lib|bin))/=) {
+      if (! m=(/(doc|man|info|usr/src)/|\.(so|ph|h|html|pod)$)=) {
+        process_file($_);
+      }
+    }
   }
 } else {
   
@@ -49,14 +53,22 @@
   # contents of the file.
   
   foreach (<>) {
-    process_file($_);
+     if (m=/usr/(sbin|bin|lib|share|X11R6/(lib|bin))/=) {
+       if (! m=(/(doc|man|info|usr/src)/|\.(so|ph|h|html|pod)$)=) {
+         process_file($_);
+       }
+     }
   }
 }
 
 
 foreach $module (sort keys %require) {
   if (length($require{$module}) == 0) {
-    print "perl($module)\n";
+     if ($module =~ /^[0-9._]+$/) {
+       print "perl >= $module\n";
+     } else {
+         print "perl($module)\n";
+     }
   } else {
 
     # I am not using rpm3.0 so I do not want spaces arround my
@@ -202,7 +214,31 @@
       # will be included with the name sys/systeminfo.ph so only use the
       # basename of *.ph files
 
-      ($module  =~ m/\.ph$/) && next;
+      # ($module  =~ m/\.ph$/) && ($module =~ s!.*/!!g );
+ 
+      # there is no need to generate dependencies for ph, pl or test files
+      # so let's just skip them.
+
+      ($module =~ m/\.(ph|pl|t)$/) && next;
+
+      # skip all modules for platforms other than linux.
+
+      ($module =~ m/Mac|OS2|MSDOS|Win32|VMS|vmsish/) && next;
+
+      # if the module name starts in a dot it is not a module name.
+
+      ($module =~ m/^\./) && next;
+
+      # if the module ends with .pm strip it to leave only basename.
+
+      $module =~ s/\.pm$//;
+
+      $module =~ s/\//::/;
+
+      # trim off trailing parenthesis if any.  Sometimes people pass
+      # the module an empty list.
+
+      $module =~ s/\(\s*\)$//;
 
       $require{$module}=$version;
       $line{$module}=$_;
--- rpm-4.0.4/configure.in.wiget	Wed Jan 23 02:01:43 2002
+++ rpm-4.0.4/configure.in	Tue Feb 12 21:30:54 2002
@@ -228,6 +228,7 @@
     AC_PATH_PROG(__MKDIR, mkdir, /bin/mkdir, $MYPATH)
     AC_PATH_PROG(__MV, mv, /bin/mv, $MYPATH)
     AC_PATH_PROG(__PATCH, patch, /usr/bin/patch, $MYPATH)
+    AC_PATH_PROG(PERL, perl, /usr/bin/perl, $MYPATH)
     AC_MSG_CHECKING(old version of patch)
     PATCHVERSION=`patch --version 2>&1`
 
@@ -1164,6 +1165,7 @@
 AC_CONFIG_SUBDIRS(popt beecrypt zlib $WITH_DB_SUBDIR)
 
 AC_OUTPUT([ Doxyfile Makefile rpmrc macros platform rpmpopt rpm.spec
+	macros.perl scripts/perl.req scripts/perl.prov
 	rpmio/Makefile rpmdb/Makefile lib/Makefile build/Makefile
 	scripts/Makefile scripts/brp-redhat tools/Makefile
 	tests/Makefile tests/rpmrc tests/macros tests/hello-test/Makefile
--- rpm-4.0.4/macros.in.wiget	Tue Feb 12 21:25:39 2002
+++ rpm-4.0.4/macros.in	Tue Feb 12 21:29:18 2002
@@ -898,8 +898,10 @@
 #	%dir %{perl_sitearch}/auto/Image
 #
 %requires_eq()	%(LC_ALL="C" echo '%*' | xargs -r rpm -q --qf 'Requires: %%{name} = %%{epoch}:%%{version}\\n' | sed -e 's/ (none):/ /' -e 's/ 0:/ /' | grep -v "is not")
-%perl_sitearch	%(eval "`perl -V:installsitearch`"; echo $installsitearch)
-%perl_archlib	%(eval "`perl -V:installarchlib`"; echo $installarchlib)
+#
+# NOTE: %{perl_sitearch}, %{perl_archlib} and %{perl_sitelib} macros 
+#	has been moved to macros.perl (see info in this file for details).
+#
 
 #------------------------------------------------------------------------------
 # arch macro for all Intel i?86 compatibile processors
