--- rpm-4.3/macros.in.orig	2003-06-25 23:53:55.000000000 +0200
+++ rpm-4.3/macros.in	2003-06-25 23:58:35.000000000 +0200
@@ -321,7 +321,9 @@
 
 #
 # Use internal dependency generator rather than external helpers?
-%_use_internal_dependency_generator	1
+# note: we wrap around internal generator (invoked by rpmdeps) to
+#       postprocess provides/requires lists and apply _noauto*
+%_use_internal_dependency_generator	0
 
 #
 # Path to scripts to autogenerate package dependencies,
@@ -329,8 +331,12 @@
 # Note: Used iff _use_internal_dependency_generator is zero.
 #%__find_provides	@RPMCONFIGDIR@/rpmdeps --provides
 #%__find_requires	@RPMCONFIGDIR@/rpmdeps --requires
-%__find_provides	@RPMCONFIGDIR@/find-provides
-%__find_requires	@RPMCONFIGDIR@/find-requires
+# these are ELF-only generators
+#%__find_provides	@RPMCONFIGDIR@/find-provides
+#%__find_requires	@RPMCONFIGDIR@/find-requires
+# rpmdeps wrappers
+%__find_provides	@RPMCONFIGDIR@/find-provides-wrapper%{?_perl_deps: --with-perl}%{?_php_deps: --with-php}%{?_noautoprovfiles:%{?buildroot: --buildroot='%{buildroot}'} --noautoprovfiles='%{_noautoprovfiles}'}%{?_noautoprov: --noautoprov='%{_noautoprov}'}
+%__find_requires	@RPMCONFIGDIR@/find-requires-wrapper%{?_perl_deps: --with-perl}%{?_php_deps: --with-php}%{?buildroot: --buildroot='%{buildroot}'}%{?_noautoprovfiles: --noautoprovfiles='%{_noautoprovfiles}'}%{?_noautoprov: --noautoprov='%{_noautoprov}'}%{?_noautoreqfiles: --noautoreqfiles='%{_noautoreqfiles}'}%{?_noautoreq: --noautoreq='%{_noautoreq}'}%{?_noautoreqdep: --noautoreqdep='%{_noautoreqdep}'}
 #%__find_conflicts	???
 #%__find_obsoletes	???
 
