diff -urN rpm-5.1.4/lib/rpmfc.c rpm-5.1.4.new/lib/rpmfc.c
--- rpm-5.1.4/lib/rpmfc.c	2008-07-29 19:36:38.000000000 +0200
+++ rpm-5.1.4.new/lib/rpmfc.c	2008-07-29 19:43:42.000000000 +0200
@@ -527,6 +527,8 @@
   { "troff or preprocessor input",	RPMFC_MANPAGE|RPMFC_INCLUDE },
   { "GNU Info",			RPMFC_MANPAGE|RPMFC_INCLUDE },
 
+  { "Desktop Entry",		RPMFC_DESKTOP_FILE|RPMFC_INCLUDE },
+
   { "perl script text",		RPMFC_PERL|RPMFC_INCLUDE },
   { "Perl5 module source text", RPMFC_PERL|RPMFC_MODULE|RPMFC_INCLUDE },
 
@@ -875,6 +877,10 @@
 	    xx = rpmfcHelper(fc, 'P', "java", fc->noautoprov, fc->noautoprov_c);
 	if (fc->findreq)
 	    xx = rpmfcHelper(fc, 'R', "java", fc->noautoreq, fc->noautoreq_c);
+    } else
+    if (fc->fcolor->vals[fc->ix] & RPMFC_DESKTOP_FILE) {
+	if (fc->findprov)
+	    xx = rpmfcHelper(fc, 'P', "mimetype", fc->noautoprov, fc->noautoprov_c);
     }
 
     return 0;
@@ -957,7 +963,7 @@
 /*@unchecked@*/
 static struct rpmfcApplyTbl_s rpmfcApplyTable[] = {
     { rpmfcELF,		RPMFC_ELF },
-    { rpmfcSCRIPT,	(RPMFC_SCRIPT|RPMFC_PERL|RPMFC_PYTHON|RPMFC_LIBTOOL|RPMFC_PKGCONFIG|RPMFC_BOURNE|RPMFC_JAVA|RPMFC_PHP|RPMFC_MONO) },
+    { rpmfcSCRIPT,	(RPMFC_SCRIPT|RPMFC_PERL|RPMFC_PYTHON|RPMFC_LIBTOOL|RPMFC_PKGCONFIG|RPMFC_BOURNE|RPMFC_JAVA|RPMFC_PHP|RPMFC_MONO|RPMFC_DESKTOP_FILE) },
     { NULL, 0 }
 };
 
@@ -1311,6 +1317,9 @@
 	    else if (_suffix(s, ".php"))
 		ftype = "PHP script text";
 
+	    else if (_suffix(s, ".desktop"))
+		ftype = "Desktop Entry";
+
 	    /* XXX skip all files in /dev/ which are (or should be) %dev dummies. */
 	    else if (slen >= fc->brlen+sizeof("/dev/") && !strncmp(s+fc->brlen, "/dev/", sizeof("/dev/")-1))
 		ftype = "";
diff -urN rpm-5.1.4/lib/rpmfc.h rpm-5.1.4.new/lib/rpmfc.h
--- rpm-5.1.4/lib/rpmfc.h	2008-07-29 19:36:38.000000000 +0200
+++ rpm-5.1.4.new/lib/rpmfc.h	2008-07-29 19:44:27.000000000 +0200
@@ -28,7 +28,8 @@
 #define	RPMFC_ELF	(RPMFC_ELF32|RPMFC_ELF64|RPMFC_ELFMIPSN32)
 	/* (1 << 3) leaks into package headers, reserved */
 
-	/* bits 4-7 unused */
+	/* bits 4-6 unused */
+    RPMFC_DESKTOP_FILE		= (1 <<  7),
     RPMFC_PKGCONFIG		= (1 <<  8),
     RPMFC_LIBTOOL		= (1 <<  9),
     RPMFC_BOURNE		= (1 << 10),
diff -urN rpm-5.1.4/macros.in rpm-5.1.4.new/macros.in
--- rpm-5.1.4/macros.in	2008-07-29 19:36:38.000000000 +0200
+++ rpm-5.1.4.new/macros.in	2008-07-29 19:45:45.000000000 +0200
@@ -1617,6 +1617,16 @@
 %__mono_requires	%{nil}
 
 #------------------------------------------------------------------------
+# mimetype(...) configuration.
+#
+# Path to script to autogenerate mimetype(foo) provides, based on MimeType
+# key from desktop files.
+#
+# Note: Used if _use_internal_dependency_generator is non-zero. The
+# helper is also used by %{_uselibrpm}/rpmdeps --provides
+%__mimetype_provides %{_usrlibrpm}/mimetypedeps.sh --provides
+
+#------------------------------------------------------------------------
 # executable(...) configuration.
 #
 # Path to scripts to autogenerate executable(foo) script dependencies,
