--- rpm-4.1/beecrypt/configure.ac.orig	Sat Jun 15 20:49:02 2002
+++ rpm-4.1/beecrypt/configure.ac	Sun Nov 24 19:51:39 2002
@@ -82,11 +82,6 @@
 AC_PROG_CXX
 AC_PROG_LIBTOOL
 
-# Initialize tool flags
-CFLAGS=""
-CPPFLAGS=""
-LDFLAGS=""
-
 # Checks for compiler characteristics and flags
 if test "$ac_cv_prog_gcc" != yes; then
   AC_CACHE_CHECK([whether we are using Sun Workshop C], ac_cv_prog_SUNPRO_CC, [
@@ -706,7 +697,6 @@
   fi
 else
   if test "$ac_cv_prog_gcc" = yes; then
-    CFLAGS="$CFLAGS -O3 -g"
     if test "$ac_enable_optimized" = yes; then
       case $target_cpu in
 # This switch makes the mp32 routines slower by about 10%, so it's disabled
@@ -717,16 +707,16 @@
         CFLAGS="$CFLAGS -fomit-frame-pointer"
         ;;
       i386)
-        CFLAGS="$CFLAGS -march=i386 -mcpu=i686 -fomit-frame-pointer"
+        CFLAGS="$CFLAGS -fomit-frame-pointer"
         ;;
       i486)
-        CFLAGS="$CFLAGS -march=i386 -mcpu=i686 -fomit-frame-pointer"
+        CFLAGS="$CFLAGS -fomit-frame-pointer"
         ;;
       i586)
-        CFLAGS="$CFLAGS -march=i386 -mcpu=i686 -fomit-frame-pointer"
+        CFLAGS="$CFLAGS -fomit-frame-pointer"
         ;;
       i686)
-	CFLAGS="$CFLAGS -march=i386  -mcpu=i686 -fomit-frame-pointer"
+	CFLAGS="$CFLAGS -fomit-frame-pointer"
         ;;
       sparc)
         CFLAGS="$CFLAGS -m32 -mtune=ultrasparc"
@@ -787,8 +777,6 @@
     if test "$ac_enable_threads" = yes; then
       CFLAGS="$CFLAGS -pthread"
     fi
-  else
-    CFLAGS="$CFLAGS -O2"
   fi
 fi
 
