Index: rpmsx.c
===================================================================
RCS file: /cvs/devel/rpm/lib/rpmsx.c,v
retrieving revision 2.6
diff -u -u -r2.6 rpmsx.c
--- rpm/lib/rpmsx.c	25 Dec 2003 23:00:45 -0000	2.6
+++ rpm/lib/rpmsx.c	17 Jan 2004 17:52:19 -0000
@@ -316,7 +316,10 @@
 /*@=branchstate@*/
 
     if ((fp = fopen(fn, "r")) == NULL) {
-	perror(fn);
+	if (ENOENT == errno)
+	    fprintf(stderr, _("Parsing SELinux file contexts skipped: %s: %s\n"), fn, strerror(errno));
+	else
+    	    perror(fn);
 	return -1;
     }
 

