--- rpm-5.1b2/macros.in~	2008-04-13 19:16:37.000000000 +0300
+++ rpm-5.1b2/macros.in	2008-04-13 19:16:57.396520131 +0300
@@ -282,6 +282,9 @@
 #
 #%distribution
 
+# PLD Linux Release
+%pld_release    th
+
 #	Configurable distribution URL, same as DistURL: tag in a specfile.
 #	The URL will be used to supply reliable information to tools like
 #	rpmfind.
@@ -1625,7 +1625,7 @@
 # -feliminate-dwarf2-dups disabled until PR ld/3290 is fixed.
 
 %debugcflags	-O0 -g -Wall
-%optldflags	-Wl,--as-needed
+%optldflags	-Wl,--as-needed -Wl,-z,relro -Wl,-z,-combreloc
 
 # Warning: those macros are overwritten by macros.build,
 #          left here for compatibility
