--- rpm-4.4.3/macros.in	2005-11-19 01:19:50.657854672 +0100
+++ rpm-4.4.6/macros.in	2006-05-24 11:49:38.438332239 +0300
@@ -219,7 +219,7 @@
 %_tmppath		%{_var}/tmp
 
 #	Path to top of build area.
-%_topdir		%{_usrsrc}/rpm
+%_topdir		%(echo $HOME)/rpm
 
 #	The path to the unzip executable (legacy, use %{__unzip} instead).
 %_unzipbin		%{__unzip}
@@ -330,8 +330,8 @@
 #		"w9.gzdio"	gzip level 9 (default).
 #		"w9.bzdio"	bzip2 level 9.
 #
-#%_source_payload	w9.gzdio
-#%_binary_payload	w9.gzdio
+%_source_payload	w9.gzdio
+%_binary_payload	w9.lzdio
 
 #	Archive formats to use for source/binary package payloads.
 #		"cpio"		cpio archive (default)
@@ -932,10 +932,10 @@
   export RPM_BUILD_ROOT}\
   %{?_javaclasspath:CLASSPATH=\"%{_javaclasspath}\"\
   export CLASSPATH}\
-  unset PERL_MM_OPT\
+  unset PERL_MM_OPT || :\
   LANG=C\
   export LANG\
-  unset DISPLAY\
+  unset DISPLAY || :\
   \
   %{verbose:set -x}%{!verbose:exec > /dev/null}\
   umask 022\
@@ -1374,4 +1374,17 @@
 #%__executable_requires	@RPMCONFIGDIR@/executabledeps.sh --requires
 
+# PLD rpm macros
+%_enable_debug_packages 1
+
+#-----------------------------------------------------------------
+# CFLAGS and LDFLAGS used to build
+
+%debuginfocflags	%{expand:%%define __dic_%{?_enable_debug_packages} 1}%{?__dic_1: -ggdb}%{expand:%%undefine __dic_%{?_enable_debug_packages}}
+
+%debugcflags	-O0 -g -Wall
+%rpmcflags	%{?debug:%debugcflags}%{!?debug:%optflags}%{?debuginfocflags}
+%rpmcxxflags	%{rpmcflags}
+%rpmldflags	%{nil}
+
 # \endverbatim
 #*/
