--- rpm-5.3.1/macros/macros.rpmbuild.in.wiget	2010-03-31 14:22:06.000000000 +0200
+++ rpm-5.3.1/macros/macros.rpmbuild.in	2010-05-30 16:51:46.906008986 +0200
@@ -383,7 +383,7 @@ echo "Patch #%{__patch_number} (%{basena
 %{?buildroot:%{__rm} -rf '%{buildroot}'; %{__mkdir_p} '%{buildroot}'}
 %__spec_install_body	%{___build_body}
 %__spec_install_post\
-%{?__debug_package:%{__debug_install_post}}\
+%{expand:%%define __sip_%{?_enable_debug_packages} 1}%{?__sip_1:%{?__debug_package:%{__debug_install_post}}}%{expand:%%undefine __sip_%{?_enable_debug_packages}}\
 %{__arch_install_post}\
 %{__os_install_post}\
 %{nil}
--- rpm-5.3.1/macros/macros.in.wiget	2010-05-30 16:47:23.389365448 +0200
+++ rpm-5.3.1/macros/macros.in	2010-05-30 16:47:23.412699985 +0200
@@ -131,15 +131,21 @@
 %debug_package \
 %ifnarch noarch\
 %global __debug_package 1\
-%package debug\
+%package debuginfo\
 Summary: Debug information for package %{name}\
+Summary(pl.UTF-8): Informacje dla debuggera dla pakietu %{name}\
 Group: Development/Debug\
+Requires: filesystem-debuginfo >= 3.0-16\
 AutoReqProv: 0\
-%description debug\
+%description debuginfo\
 This package provides debug information for package %{name}.\
 Debug information is useful when developing applications that use this\
 package or when debugging this package.\
-%files debug -f debugfiles.list\
+%description debuginfo -l pl.UTF-8\
+Ten pakiet dostarcza informacje dla debuggera dla pakietu %{name}.\
+Informacje te są przydatne przy rozwijaniu aplikacji używających tego\
+pakietu oraz przy odpluskwianiu samego pakietu.\
+%files debuginfo -f debugfiles.list\
 %defattr(-,root,root)\
 %endif\
 %{nil}
--- rpm-5.3.1/scripts/find-debuginfo.sh.wiget	2009-09-10 03:52:13.000000000 +0200
+++ rpm-5.3.1/scripts/find-debuginfo.sh	2010-05-30 16:53:33.275994510 +0200
@@ -92,7 +92,8 @@ strip_to_debug()
   $strip_g && case "$(file -bi "$2")" in
   application/x-sharedlib,*) g=-g ;;
   esac
-  eu-strip --remove-comment $g -f "$1" "$2" || exit
+  objcopy --only-keep-debug --remove-section .comment "$2" "$1" || exit
+  objcopy --add-gnu-debuglink="$1" "$2" || exit
   chmod 444 "$1" || exit
 }
 
