--- rpm-4.4.4/lib/poptI.c.orig	2005-11-27 13:50:16.000000000 +0000
+++ rpm-4.4.4/lib/poptI.c	2005-12-28 02:43:01.029574304 +0000
@@ -16,8 +16,7 @@
 /*@unchecked@*/
 struct rpmInstallArguments_s rpmIArgs = {
     0,			/* transFlags */
-			/* probFilter */
-    (RPMPROB_FILTER_REPLACEOLDFILES | RPMPROB_FILTER_REPLACENEWFILES),
+    0,			/* probFilter */
     0,			/* installInterfaceFlags */
     0,			/* eraseInterfaceFlags */
     0,			/* qva_flags */
@@ -274,9 +273,6 @@
 	N_("skip files with leading component <path> "),
 	N_("<path>") },
 
- { "fileconflicts", '\0', POPT_BIT_CLR, &rpmIArgs.probFilter,
-	(RPMPROB_FILTER_REPLACEOLDFILES | RPMPROB_FILTER_REPLACENEWFILES),
-	N_("detect file conflicts between packages"), NULL},
  { "force", '\0', 0, NULL, RPMCLI_POPT_FORCE,
 	N_("short hand for --replacepkgs --replacefiles"), NULL},
 
