--- rpm-4.4.2.2/rpmio/macro.c~	2007-12-20 01:27:39.588605566 +0200
+++ rpm-4.4.2.2/rpmio/macro.c	2007-12-20 01:29:52.512813009 +0200
@@ -2016,12 +2016,13 @@
 
 	/* Read macros from each file. */
 	for (i = 0; i < ac; i++) {
-	    if (strstr(av[i], ".rpmnew") || 
-		strstr(av[i], ".rpmsave") ||
-		strstr(av[i], ".rpmorig")) {
-		continue;
+	    size_t slen = strlen(av[i]);
+	    if ((av[i])[slen-1] != '~' &&
+		(slen < 8 || strcmp((av[i] + slen - 7), ".rpmnew")) &&
+		(slen < 9 || (strcmp((av[i] + slen - 8), ".rpmorig") &&
+			      strcmp((av[i] + slen - 8), ".rpmsave")))) {
+		(void) rpmLoadMacroFile(mc, av[i]);
 	    }
-	    (void) rpmLoadMacroFile(mc, av[i]);
 	    av[i] = _free(av[i]);
 	}
 	av = _free(av);
