--- rpm-5.4.18/build/files.c~	2017-07-18 18:00:51.000000000 +0300
+++ rpm-5.4.18/build/files.c	2017-07-18 18:01:41.326747116 +0300
@@ -1041,7 +1041,6 @@
 	/* XXX FIXME: this is easy to do as macro expansion */
 
 	    if (! fl->passedSpecialDoc) {
-	    	char *compress_doc;
 	    	char *mkdir_p;
 
 		pkg->specialDoc = rpmiobNew(0);
@@ -1064,11 +1064,6 @@
 		mkdir_p = _free(mkdir_p);
 		pkg->specialDoc = rpmiobAppend(pkg->specialDoc, " \"$DOCDIR\"", 1);
 
-		compress_doc = rpmExpand("%{__compress_doc}", NULL);
-		if (compress_doc && *compress_doc != '%')
-	    	    pkg->specialDoc = rpmiobAppend(pkg->specialDoc, compress_doc, 1);
-		compress_doc = _free(compress_doc);
-
 		*fileName = buf;
 		fl->passedSpecialDoc = 1;
 		fl->isSpecialDoc = 1;
@@ -1071,6 +1065,15 @@
 	    pkg->specialDoc = rpmiobAppend(pkg->specialDoc, "cp -pr ", 0);
 	    pkg->specialDoc = rpmiobAppend(pkg->specialDoc, specialDocBuf, 0);
 	    pkg->specialDoc = rpmiobAppend(pkg->specialDoc, " \"$DOCDIR\"", 1);
+
+	    {
+	    	char *compress_doc;
+
+		compress_doc = rpmExpand("%{__compress_doc}", NULL);
+		if (compress_doc && *compress_doc != '%')
+	    	    pkg->specialDoc = rpmiobAppend(pkg->specialDoc, compress_doc, 1);
+		compress_doc = _free(compress_doc);
+	    }
 	}
     }
 
