diff -urN rpm-4.3.org/Makefile.am rpm-4.3/Makefile.am
--- rpm-4.3.org/Makefile.am	2003-12-28 01:33:43.917858224 +0100
+++ rpm-4.3/Makefile.am	2003-12-28 02:01:02.060822640 +0100
@@ -14,7 +14,7 @@
 	po/*.in po/*.po po/rpm.pot \
 	rpm.magic rpmpopt-$(VERSION) rpmqv.c rpm.c
 
-SUBDIRS = intl po @WITH_ZLIB_SUBDIR@ @WITH_ELFUTILS_SUBDIR@ file @WITH_DB_SUBDIR@ @WITH_BEECRYPT_SUBDIR@ lua rpmio rpmdb lib build misc @WITH_PYTHON_SUBDIR@ tools scripts tests doc .
+SUBDIRS = intl po @WITH_ZLIB_SUBDIR@ @WITH_ELFUTILS_SUBDIR@ file @WITH_BEECRYPT_SUBDIR@ lua rpmio rpmdb lib build misc @WITH_PYTHON_SUBDIR@ tools scripts tests doc .
 
 INCLUDES = \
 	-I$(top_srcdir)/build \
diff -urN rpm-4.3.org/rpmdb/Makefile.am rpm-4.3/rpmdb/Makefile.am
--- rpm-4.3.org/rpmdb/Makefile.am	2003-12-28 01:33:46.160517288 +0100
+++ rpm-4.3/rpmdb/Makefile.am	2003-12-28 01:56:22.000000000 +0100
@@ -25,18 +25,15 @@
 noinst_HEADERS = fprint.h header_internal.h legacy.h
 
 pkgbindir = @RPMCONFIGDIR@
-pkgbin_PROGRAMS = \
-	rpmdb_deadlock rpmdb_dump rpmdb_load rpmdb_svc rpmdb_stat rpmdb_verify
-noinst_PROGRAMS = \
-	rpmdb_archive rpmdb_checkpoint rpmdb_printlog rpmdb_recover \
-	rpmdb_upgrade
+pkgbin_PROGRAMS =
+noinst_PROGRAMS =
 
 mylibs = librpmdb.la
 
 LIBS =
 
 # XXX watchout, ../db3/libdb.la created by this Makefile may surprise
-libdb_la = $(top_builddir)/$(WITH_DB_SUBDIR)/libdb.la
+libdb_la = 
 
 # XXX grrr, RPM_BUILD_ROOT prevents build pollution if/when -lrpm different
 LDFLAGS = -L$(RPM_BUILD_ROOT)$(usrlibdir) -L$(DESTDIR)$(usrlibdir)
@@ -52,7 +49,7 @@
 	$(top_builddir)/rpmio/librpmio.la \
 	@WITH_POPT_LIB@ \
 	@WITH_LIBELF_LIB@
-librpmdb_la_LIBADD = $(DBLIBOBJS) $(libdb_la)
+librpmdb_la_LIBADD = $(DBLIBOBJS) $(libdb_la) -ldb
 librpmdb_la_DEPENDENCIES = $(DBLIBOBJS) $(libdb_la)
 
 # XXX make sure that db.h symlink is generated
