--- rpm-4.4.1/lib/rpmrc.c.orig	2005-07-19 18:53:09.096470264 +0200
+++ rpm-4.4.1/lib/rpmrc.c	2005-07-19 19:00:48.610613496 +0200
@@ -1,6 +1,12 @@
 /*@-bounds@*/
 #include "system.h"
 
+#ifdef USE_HOME_ETC
+#include <home_etc.h>
+#else
+#define _HE(x) (x)
+#endif
+
 #include <stdarg.h>
 #if defined(__linux__) && defined(__powerpc__)
 #include <setjmp.h>
@@ -1815,13 +1821,13 @@
 		rc = 1;
 		break;
 	    }
-	    if (strlen(home) > (sizeof(fn) - strlen(r))) {
+	    if (strlen(_HE(home)) > (sizeof(fn) - strlen(r))) {
 		rpmError(RPMERR_RPMRC, _("Cannot read %s, HOME is too large.\n"),
 				r);
 		rc = 1;
 		break;
 	    }
-	    strcpy(fn, home);
+	    strcpy(fn, _HE(home));
 	    r++;
 	}
 	strncat(fn, r, sizeof(fn) - (strlen(fn) + 1));
--- rpm-4.4.2/lib/Makefile.am.orig	2005-07-23 13:40:23.191135552 +0200
+++ rpm-4.4.2/lib/Makefile.am	2005-07-23 13:41:49.474018560 +0200
@@ -46,6 +46,7 @@
 	$(top_builddir)/rpmio/librpmio.la \
 	@WITH_POPT_LIB@ \
 	@WITH_SELINUX_LIB@ \
+	@HOME_ETC_LIB@ \
 	@INTLLIBS@
 
 getdate.c: getdate.y
--- rpm-4.4.1/configure.ac.orig	2005-07-19 18:53:09.385426000 +0200
+++ rpm-4.4.1/configure.ac	2005-07-19 19:10:49.732229120 +0200
@@ -945,6 +945,14 @@
 fi
 AC_SUBST(WITH_SELINUX_LIB)
 
+withval=no
+AC_ARG_WITH(home_etc, [  --with-home_etc         build with home_etc support ])
+if test $withval != no ; then
+  AC_DEFINE(USE_HOME_ETC, 1, [Build with home_etc support?])
+  HOME_ETC_LIB="-lhome_etc"
+fi
+AC_SUBST(HOME_ETC_LIB)
+
 WITH_LUA_LIB=
 WITH_LUA_SUBDIR=
 WITH_LUA_INCLUDE=
