--- rpm-4.1/configure.ac.nointernallib	Thu Sep 19 04:02:42 2002
+++ rpm-4.1/configure.ac	Thu Sep 19 04:03:04 2002
@@ -1194,7 +1194,7 @@
 dnl # XXX Propagate -lucb to popt ...
 dnl export LIBS INCPATH CONFIG_SITE
 
-AC_CONFIG_SUBDIRS(popt beecrypt zlib libelf db3)
+AC_CONFIG_SUBDIRS(popt beecrypt db3)
 
 AC_CONFIG_FILES([ Doxyfile Makefile rpmrc macros platform rpmpopt rpm.spec m4/Makefile 
 	macros.perl scripts/perl.req scripts/perl.prov
--- rpm-4.1/Makefile.am.nointernallib	Thu Sep 19 04:03:19 2002
+++ rpm-4.1/Makefile.am	Thu Sep 19 04:04:00 2002
@@ -12,7 +12,7 @@
 	po/*.in po/*.po po/rpm.pot \
 	rpm.magic rpmpopt-$(VERSION) rpmqv.c rpm.c
 
-SUBDIRS = intl po @WITH_ZLIB_SUBDIR@ libelf @WITH_DB_SUBDIR@ popt beecrypt rpmio rpmdb lib build misc @WITH_PYTHON_SUBDIR@ tools scripts tests doc .
+SUBDIRS = intl po @WITH_DB_SUBDIR@ popt beecrypt rpmio rpmdb lib build misc @WITH_PYTHON_SUBDIR@ tools scripts tests doc .
 
 INCLUDES = \
 	-I$(top_srcdir)/build \
