--- rpm-4.4.2.2/build/files.c~	2007-12-18 01:15:59.584854868 +0200
+++ rpm-4.4.2.2/build/files.c	2007-12-18 01:17:06.891921768 +0200
@@ -2042,10 +2042,12 @@
     fl.docDirs[fl.docDirCount++] = xstrdup("/usr/share/man");
     fl.docDirs[fl.docDirCount++] = xstrdup("/usr/share/info");
     fl.docDirs[fl.docDirCount++] = xstrdup("/usr/share/gtk-doc/html");
+    fl.docDirs[fl.docDirCount++] = xstrdup("/usr/src/examples");
     fl.docDirs[fl.docDirCount++] = rpmGetPath("%{_docdir}", NULL);
     fl.docDirs[fl.docDirCount++] = rpmGetPath("%{_mandir}", NULL);
     fl.docDirs[fl.docDirCount++] = rpmGetPath("%{_infodir}", NULL);
     fl.docDirs[fl.docDirCount++] = rpmGetPath("%{_javadocdir}", NULL);
+    fl.docDirs[fl.docDirCount++] = rpmGetPath("%{_examplesdir}", NULL);
     
     fl.fileList = NULL;
     fl.fileListRecsAlloced = 0;
