--- rpm-3.0.4/rpmpopt.in.wiget	Wed Feb 23 23:40:53 2000
+++ rpm-3.0.4/rpmpopt.in	Wed Mar 22 17:39:52 2000
@@ -81,3 +81,12 @@
 
 # Build policies enabled from command line. Last policy applies.
 rpm alias --buildpolicy --define '__spec_install_post @RPMCONFIGDIR@/brp-!#:+'
+
+# Make RPM build tree
+rpm exec --install-build-tree install-build-tree
+
+# Compiling with debuginfo may be enabled by --debug
+rpm alias --debug --define 'debug 1'
+
+# Don't remove $BUILDROOT if --noclean
+rpm alias --noclean --define '__spec_clean_pre exit 0'
\ No newline at end of file
