diff -ur rpm-4.0.4/build/files.c rpm-4.0.4-/build/files.c
--- rpm-4.0.4/build/files.c	Tue Jul 23 10:38:11 2002
+++ rpm-4.0.4-/build/files.c	Tue Jul 23 10:37:57 2002
@@ -1943,7 +1943,10 @@
     /* Now process special doc, if there is one */
     if (specialDoc) {
 	if (installSpecialDoc) {
-	    (void) doScript(spec, RPMBUILD_STRINGBUF, "%doc", pkg->specialDoc, test);
+	    int rc = doScript(spec, RPMBUILD_STRINGBUF, "%doc", 
+	    		      pkg->specialDoc, test);
+	    if (rc)
+	        fl.processingFailed = rc;
 	}
 
 	/* Reset for %doc */
Only in rpm-4.0.4-/build: tags
