--- rpm-4.5/macros.in~	2008-06-10 02:17:16.000000000 +0300
+++ rpm-4.5/macros.in	2008-06-10 02:17:47.682505213 +0300
@@ -350,7 +350,7 @@
 #		"w9.lzdio"	lzma level 9.
 #
 %_source_payload	w9.gzdio
-%_binary_payload	w9.lzdio
+%_binary_payload	w9.bzdio
 
 #	Archive formats to use for source/binary package payloads.
 #		"cpio"		cpio archive (default)
@@ -1448,8 +1448,8 @@
 #
 # Note: Used iff _use_internal_dependency_generator is non-zero. The
 # helpers are also used by %{_rpmhome}/rpmdeps {--provides|--requires}.
-%__libtool_provides	%{_rpmhome}/libtooldeps.sh --provides %{buildroot} %{name}
-%__libtool_requires	%{_rpmhome}/libtooldeps.sh --requires %{buildroot} %{name}
+#%__libtool_provides	%{_rpmhome}/libtooldeps.sh --provides %{buildroot} %{name}
+#%__libtool_requires	%{_rpmhome}/libtooldeps.sh --requires %{buildroot} %{name}
 
 #------------------------------------------------------------------------
 # pkgconfig(...) configuration.
@@ -1458,8 +1458,8 @@
 #
 # Note: Used iff _use_internal_dependency_generator is non-zero. The
 # helpers are also used by %{_rpmhome}/rpmdeps {--provides|--requires}.
-%__pkgconfig_provides	%{_rpmhome}/pkgconfigdeps.sh --provides
-%__pkgconfig_requires	%{_rpmhome}/pkgconfigdeps.sh --requires
+#%__pkgconfig_provides	%{_rpmhome}/pkgconfigdeps.sh --provides
+#%__pkgconfig_requires	%{_rpmhome}/pkgconfigdeps.sh --requires
 
 #------------------------------------------------------------------------
 # executable(...) configuration.
@@ -1451,7 +1451,7 @@
 #%__scriptlet_requires	/bin/bash --rpm-requires
 
 # PLD rpm macros
-%_enable_debug_packages 1
+%_enable_debug_packages 0
 
 #-----------------------------------------------------------------
 # CFLAGS and LDFLAGS used to build
--- rpm/rpm.sysconfig~	2012-11-08 05:23:51.082114980 +0200
+++ rpm/rpm.sysconfig	2012-11-08 06:49:38.238855609 +0200
@@ -12,15 +12,3 @@
 # You can also disable this per service, if service supports it from
 # /etc/sysconfig/SERVICE
 #RPM_SKIP_AUTO_RESTART=yes
-
-# Should package upgrades enable SysV service?
-# You don't want to disable this unless you're planning to fully
-# migrate to systemd or want serviceless system.
-RPM_ENABLE_SYSV_SERVICE=yes
-
-# Should package upgrades enable systemd service?
-# This means migrating init scripts from SysV to systemd.
-# If you disable this, systemd units will not be enabled for
-# upgraded services. Newly installed packages will still have
-# systemd services enabled.
-RPM_ENABLE_SYSTEMD_SERVICE=yes
--- rpm-4.5/scripts/find-debuginfo.sh~	2012-11-08 06:51:17.000000000 +0200
+++ rpm-4.5/scripts/find-debuginfo.sh	2012-11-08 06:57:02.627280611 +0200
@@ -87,7 +87,6 @@
 strip_to_debug()
 {
   local t=$(mktemp "/tmp/rpm.stripped.XXXXXX")
-  objcopy --compress-debug-sections "$2" || exit
   eu-strip --remove-comment -f "$1" "$2" -o "$t" || exit
   rm -f "$t"
 }
