--- rpm-5.4.18/lib/psm.c~	2017-07-18 19:37:38.000000000 +0300
+++ rpm-5.4.18/lib/psm.c	2017-07-18 19:40:53.713076772 +0300
@@ -2905,7 +2905,7 @@
 
 	psm->rpmio_flags = te = (char *) xmalloc(sizeof("w9.zstdio")+32);
 	*te = '\0';
-	te = stpcpy(te, ((psm->goal == PSM_PKGSAVE) ? "w9" : "r"));
+	te = stpcpy(te, ((psm->goal == PSM_PKGSAVE) ? "w6" : "r"));
 	if (!strcmp(payload_compressor, "gzip"))
 	    te = stpcpy(te, ".gzdio");
 	if (!strcmp(payload_compressor, "bzip2"))
