--- rpm-5.3.1/scripts/find-debuginfo.sh.wiget	2009-09-10 03:52:13.000000000 +0200
+++ rpm-5.3.1/scripts/find-debuginfo.sh	2010-05-30 16:53:33.275994510 +0200
@@ -92,8 +92,10 @@ strip_to_debug()
   esac
   [ -n "$EXCLUDE_FULL_REGEXP" ] && grep -E -q "$EXCLUDE_FULL_REGEXP" <<< "$2" && g=-g
 
-  eu-strip --remove-comment $g $([ -n "$DISABLE_DEBUG" ] || echo -f "$1") "$2" || exit  
-  [ -n "$DISABLE_DEBUG" ] || chmod 444 "$1" || exit
+  [ -n "$DISABLE_DEBUG" ] && return
+  objcopy --only-keep-debug --remove-section .comment "$2" "$1" || exit
+  objcopy --add-gnu-debuglink="$1" "$2" || exit
+  chmod 444 "$1" || exit
 }
 
 # Make a relative symlink to $1 called $3$2
