diff -ur rpm.org/file/doc/Makefile.am rpm/file/doc/Makefile.am
--- rpm.org/file/doc/Makefile.am	2004-11-18 15:04:33.000000000 +0100
+++ rpm/file/doc/Makefile.am	2004-11-22 01:02:30.114807597 +0100
@@ -1,4 +1,4 @@
-MAGIC	 = $(pkgdatadir)/magic
+MAGIC	 = /usr/share/file/magic
 if FSECT5
 man_MAGIC = magic.5
 else
Tylko w rpm/file/doc: Makefile.am~
diff -ur rpm.org/file/Makefile.am rpm/file/Makefile.am
--- rpm.org/file/Makefile.am	2004-11-18 15:04:32.000000000 +0100
+++ rpm/file/Makefile.am	2004-11-22 01:01:36.817958788 +0100
@@ -3,4 +3,4 @@
 
 EXTRA_DIST = ChangeLog LEGAL.NOTICE MAINT README
 
-SUBDIRS = src magic doc
+SUBDIRS = src magic
Tylko w rpm/file: Makefile.am.orig
diff -ur rpm.org/file/src/Makefile.am rpm/file/src/Makefile.am
--- rpm.org/file/src/Makefile.am	2004-11-18 15:04:33.000000000 +0100
+++ rpm/file/src/Makefile.am	2004-11-22 01:01:36.823963503 +0100
@@ -3,7 +3,7 @@
 
 #MAGIC	 = $(pkgdatadir)/magic
 pkglibdir =  @prefix@/lib/rpm
-MAGIC    = $(pkglibdir)/magic
+MAGIC    = /usr/share/file/magic
 
 AM_CPPFLAGS = -DMAGIC='"$(MAGIC)"'
 
diff -ur rpm.org/tools/Makefile.am rpm/tools/Makefile.am
--- rpm.org/tools/Makefile.am	2004-11-22 00:56:22.484205000 +0100
+++ rpm/tools/Makefile.am	2004-11-22 01:03:05.178339563 +0100
@@ -30,7 +30,7 @@
 
 pkgbindir = @RPMCONFIGDIR@
 pkgbin_PROGRAMS = debugedit javadeps rpmcache rpmdeps rpmfile
-MAGIC	= $(pkgbindir)/magic
+MAGIC	= /usr/share/file/magic
 
 bin_PROGRAMS =	rpmgraph
 
