--- rpm-5.4.10/lib/rpmfc.c~	2012-12-01 21:13:39.985319735 +0100
+++ rpm-5.4.10/lib/rpmfc.c	2013-01-24 12:21:26.379891972 +0100
@@ -1142,7 +1142,7 @@
     /* Generate package and per-file dependencies. */
     for (fc->ix = 0; fc->fn[fc->ix] != NULL; fc->ix++) {
 
-	/* XXX Insure that /usr/lib{,64}/python files are marked RPMFC_PYTHON */
+	/* XXX Insure that /usr/{share,lib{,64}}/python files are marked RPMFC_PYTHON */
 	/* XXX HACK: classification by path is intrinsically stupid. */
 	{   fn = strstr(fc->fn[fc->ix], "/usr/lib");
 	    if (fn) {
@@ -1166,6 +1166,22 @@
 			fc->fcolor->vals[fc->ix] & RPMFC_LIBRARY)
 		    fc->fcolor->vals[fc->ix] |= (RPMFC_MODULE|RPMFC_SCRIPT);
 	    }
+	    fn = strstr(fc->fn[fc->ix], "/usr/share");
+	    if (fn) {
+		fn += sizeof("/usr/share")-1;
+		if (!strncmp(fn, "/python", sizeof("/python")-1))
+		    fc->fcolor->vals[fc->ix] |= RPMFC_PYTHON;
+		else if (!strncmp(fn, "/ruby", sizeof("/ruby")-1)) {
+		    fc->fcolor->vals[fc->ix] |= RPMFC_RUBY;
+		    if ((strstr(fn, ".gemspec") || strstr(fn, "rbconfig.rb"))) {
+			miRE mire = mireNew(RPMMIRE_REGEX, RPMTAG_FILEPATHS);
+			if (!mireRegcomp(mire, ".*/(specifications/.*\\.gemspec|rbconfig\\.rb)$"))
+			    if (mireRegexec(mire, fc->fn[fc->ix], (size_t) 0) >= 0)
+				fc->fcolor->vals[fc->ix] |= RPMFC_MODULE;
+			mire = mireFree(mire);
+		    }
+		}
+	    }
 	}
 
        /* XXX ugly quick & dirty integration of haskell() dependencies */
