--- rpm-4.1/build/Makefile.am.orig	Mon Jul 22 00:06:19 2002
+++ rpm-4.1/build/Makefile.am	Sun Nov 24 19:25:53 2002
@@ -23,12 +23,7 @@
 	parseFiles.c parsePreamble.c parsePrep.c parseReqs.c parseScript.c \
 	parseSpec.c poptBT.c reqprov.c spec.c
 librpmbuild_la_LDFLAGS = -release @VERSION@
-
-# XXX Add internal libtool dependence
-install-data-local:
-	@cd $(DESTDIR)/$(libdir) && \
-	sed -e "s|^dependency_libs='|& -lrpm-@VERSION@|" < librpmbuild.la > .librpmbuild.la && \
-	mv .librpmbuild.la librpmbuild.la
+librpmbuild_la_LIBADD = ../lib/librpm.la
 
 #BUILT_SOURCES = rpmbuild.lcd
 
--- rpm-4.1/lib/Makefile.am.orig	Fri Aug 30 13:41:54 2002
+++ rpm-4.1/lib/Makefile.am	Sun Nov 24 19:29:23 2002
@@ -38,12 +38,7 @@
 	signature.c stringbuf.c transaction.c \
 	verify.c
 librpm_la_LDFLAGS = -release @VERSION@
-
-# XXX Add internal libtool dependence
-install-data-local:
-	@cd $(DESTDIR)/$(libdir) && \
-	sed -e "s|^dependency_libs='|& -lrpmdb-@VERSION@ -lrpmio-@VERSION@ -lpopt|" < librpm.la > .librpm.la && \
-	mv .librpm.la librpm.la
+librpm_la_LIBADD = ../rpmdb/librpmdb.la ../rpmio/librpmio.la -lpopt
 
 getdate.c: getdate.y
 	@echo expect 10 shift/reduce conflicts
--- rpm-4.1/rpmdb/Makefile.am.orig	Fri Aug 30 15:12:01 2002
+++ rpm-4.1/rpmdb/Makefile.am	Sun Nov 24 19:30:41 2002
@@ -40,18 +40,12 @@
 	poptDB.c rpmhash.c rpmdb.c \
 	tagname.c tagtbl.c
 librpmdb_la_LDFLAGS = -release @VERSION@ @libdb3@
-librpmdb_la_LIBADD = $(DBLIBOBJS) $(DB3LOBJS) @WITH_LIBELF_LIB@
+librpmdb_la_LIBADD = ../rpmio/librpmio.la $(DBLIBOBJS) $(DB3LOBJS) @WITH_LIBELF_LIB@
 librpmdb_la_DEPENDENCIES = $(DBLIBOBJS) .created
 
 # XXX make sure that db.h symlink is generated
 dbconfig.c: db.h
 
-# XXX Add internal libtool dependence
-install-data-local:
-	@cd $(DESTDIR)/$(libdir) && \
-	sed -e "s|^dependency_libs='|& -lrpmio-@VERSION@|" < librpmdb.la > .librpmdb.la && \
-	mv .librpmdb.la librpmdb.la
-
 tagtbl.c: Makefile.am $(top_srcdir)/lib/rpmlib.h 
 	@echo '/*@-bounds@*/' > $@
 	@echo '#include "system.h"' >> $@
--- rpm-4.1/rpmio/Makefile.am.orig	Sun Nov 24 19:22:06 2002
+++ rpm-4.1/rpmio/Makefile.am	Sun Nov 24 19:31:18 2002
@@ -26,15 +26,9 @@
 	rpmio.c rpmlog.c rpmmalloc.c \
 	rpmpgp.c rpmrpc.c strcasecmp.c stubs.c url.c ugid.c
 librpmio_la_LDFLAGS = -release @VERSION@ @WITH_ZLIB_LIB@ -lrt -lpthread
-librpmio_la_LIBADD = $(BEECRYPTLOBJS)
+librpmio_la_LIBADD = -lpopt $(BEECRYPTLOBJS)
 librpmio_la_DEPENDENCIES = .created
 
-# XXX Add internal libtool dependence
-install-data-local:
-	@cd $(DESTDIR)/$(libdir) && \
-	sed -e "s|^dependency_libs='|& -lpopt|" < librpmio.la > .librpmio.la && \
-	mv .librpmio.la librpmio.la
-
 $(top_builddir)/beecrypt/listobjs:
 	${MAKE} -C $(top_builddir)/beecrypt listobjs
 
