diff -x '*~' -durN rpm-4.3.orig/configure.ac rpm-4.3/configure.ac
--- rpm-4.3.orig/configure.ac	2004-12-19 08:52:05.000000000 +0100
+++ rpm-4.3/configure.ac	2004-12-19 09:24:03.466542416 +0100
@@ -1178,9 +1178,6 @@
 
 dnl XXX Choose /usr/lib or /usr/lib64 for library installs.
 MARK64=
-case "${target_cpu}" in
-x86_64*|ppc64*|powerpc64*|sparc64*|s390x*)	MARK64=64 ;;
-esac
 AC_SUBST(MARK64)
 
 dnl Determine the canonical arch-vendor-os for the build machine
