--- rpm-5.0.2/lib/transaction.c~	2008-02-19 22:01:59.000000000 +0200
+++ rpm-5.0.2/lib/transaction.c	2008-02-19 22:02:00.000000000 +0200
@@ -1332,7 +1332,7 @@
 	    }
 
 	    if (rpmteFd(p) != NULL) {
-		fi = rpmfiNew(ts, p->h, RPMTAG_BASENAMES, 1);
+		fi = rpmfiNew(ts, p->h, RPMTAG_BASENAMES, 0);
 		if (fi != NULL) {	/* XXX can't happen */
 		    fi->te = p;
 		    p->fi = fi;
@@ -1915,7 +1915,7 @@
 
 /*@-nullpass@*/
 	    if (rpmteFd(p) != NULL) {
-	    	p->fi = rpmfiNew(ts, p->h, RPMTAG_BASENAMES, 1);
+	    	p->fi = rpmfiNew(ts, p->h, RPMTAG_BASENAMES, 0);
 	    	if (p->fi != NULL)	/* XXX can't happen */
 	    	    p->fi->te = p;
 /*@-compdef -usereleased@*/	/* p->fi->te undefined */
