--- rpm-5.4.16/rpmio/rpmdav.c~	2016-03-16 18:36:34.000000000 +0200
+++ rpm-5.4.16/rpmio/rpmdav.c	2016-03-16 18:40:59.883064408 +0200
@@ -1722,6 +1722,7 @@ int davReq(FD_t ctrl, const char * httpC
 {
     urlinfo u;
     int rc = 0;
+    const ne_status *status;
 
 assert(ctrl != NULL);
     u = (urlinfo) ctrl->u;
@@ -2032,10 +2032,21 @@
     }
 
 /* XXX somewhere else instead? */
-if (_dav_debug) {		/* HACK: tools/wget.c */
-    const ne_status *status = ne_get_status((ne_request *)ctrl->req);
-fprintf(stderr, "HTTP request sent, awaiting response... %d %s\n", status->code, status->reason_phrase);
-}
+    status = ne_get_status((ne_request *)ctrl->req);
+    switch (status->code) {
+	case 200:
+	case 201:			/* 201 Created. */
+	    break;
+	case 204:			/* HACK: if overwriting, 204 No Content. */
+	case 403:			/* 403 Forbidden. */
+	    rc = FTPERR_UNKNOWN;
+	    break;
+	default:
+	    rc = FTPERR_FILE_NOT_FOUND;
+	    break;
+    }
+    if (rc || _dav_debug)
+	fprintf(stderr, "HTTP request sent, awaiting response... %d %s\n", status->code, status->reason_phrase);
 
     if (rc)
 	goto errxit;
