--- rpm/lib/poptI.c.orig	2007-07-24 16:36:47.000000000 +0200
+++ rpm/lib/poptI.c	2007-10-02 15:11:29.213848100 +0200
@@ -17,7 +17,7 @@
 /*@-fullinitblock@*/
 /*@unchecked@*/
 struct rpmQVKArguments_s rpmIArgs = {
-    .probFilter = (RPMPROB_FILTER_REPLACEOLDFILES | RPMPROB_FILTER_REPLACENEWFILES),
+    .probFilter = 0,
 };
 /*@=fullinitblock@*/
 
@@ -275,10 +275,6 @@
 	N_("skip files with leading component <path> "),
 	N_("<path>") },
 
- { "fileconflicts", '\0', POPT_BIT_CLR, &rpmIArgs.probFilter,
-	(RPMPROB_FILTER_REPLACEOLDFILES | RPMPROB_FILTER_REPLACENEWFILES),
-	N_("detect file conflicts between packages"), NULL},
-
  { "freshen", 'F', POPT_BIT_SET, &rpmIArgs.installInterfaceFlags,
 	(INSTALL_UPGRADE|INSTALL_FRESHEN|INSTALL_INSTALL),
 	N_("upgrade package(s) if already installed"),

