--- rpm-3.0.3/configure.in.wiget	Thu Sep 30 23:25:29 1999
+++ rpm-3.0.3/configure.in	Fri Oct 29 11:43:58 1999
@@ -903,7 +903,7 @@
 export LIBS INCPATH CONFIG_SITE
 
 AC_CONFIG_SUBDIRS(popt)
-AC_OUTPUT([Makefile rpmrc macros lib/Makefile
+AC_OUTPUT([Makefile rpmrc macros macros.pld lib/Makefile
 	build/Makefile tools/Makefile scripts/Makefile
 	tests/Makefile tests/rpmrc tests/macros tests/hello-test/Makefile
 	misc/Makefile po/Makefile.in intl/Makefile
--- rpm-3.0.3/macros.in.wiget	Thu Sep 30 01:29:57 1999
+++ rpm-3.0.3/macros.in	Fri Oct 29 11:50:10 1999
@@ -60,8 +60,8 @@
 %_builddir		%{_topdir}/BUILD
 %_buildshell		/bin/sh
 %_bzip2bin		@BZIP2BIN@
-%_dbpath		%{_var}/lib/rpm
-%_defaultdocdir		%{_usr}/doc
+%_dbpath		%{_var}/state/rpm
+%_defaultdocdir		%{_usr}/share/doc
 #
 %__find_provides	@FINDPROVIDES@
 %__find_requires	@FINDREQUIRES@
@@ -70,8 +70,8 @@
 #%__find_obsoletes	???
 #
 # XXX fixowner, fixgroup, and fixperms are run at the end of hardcoded setup
-%_fixowner		[ `%{__id} -u` = '0' ] && %{__chown} -Rhf root
-%_fixgroup		[ `%{__id} -u` = '0' ] && %{__chgrp} -Rhf @ROOT_GROUP@
+%_fixowner		[ `%{__id} -u` = '0' ] && %{__chown} -Rhf root.@ROOT_GROUP@
+#%_fixgroup		[ `%{__id} -u` = '0' ] && %{__chgrp} -Rhf @ROOT_GROUP@
 %_fixperms		%{__chmod} -Rf @FIXPERMS@
 #
 %_gzipbin		@GZIPBIN@
@@ -80,13 +80,13 @@
 %_rpmdir		%{_topdir}/RPMS
 #
 # XXX Note escaped %% for use in headerSprintf
-%_rpmfilename		%%{ARCH}/%%{NAME}-%%{VERSION}-%%{RELEASE}.%%{ARCH}.rpm
+%_rpmfilename		%%{NAME}-%%{VERSION}-%%{RELEASE}.%%{ARCH}.rpm
 %_signature		none
 %_sourcedir		%{_topdir}/SOURCES
 %_specdir		%{_topdir}/SPECS
 %_srcrpmdir		%{_topdir}/SRPMS
 %_tmppath		%{_var}/tmp
-%_topdir		%{_usrsrc}/redhat
+%_topdir		%(echo $HOME)/rpm
 #
 #==============================================================================
 # ---- Optional rpmrc macros.
@@ -160,14 +160,14 @@
 %_sbindir		%{_exec_prefix}/sbin
 %_libexecdir		%{_exec_prefix}/libexec
 %_datadir		%{_prefix}/share
-%_sysconfdir		%{_prefix}/etc
+%_sysconfdir		/etc
 %_sharedstatedir	%{_prefix}/com
 %_localstatedir		%{_prefix}/var
 %_libdir		%{_exec_prefix}/lib
 %_includedir		%{_prefix}/include
 %_oldincludedir		/usr/include
-%_infodir		%{_prefix}/info
-%_mandir		%{_prefix}/man
+%_infodir		%{_prefix}/share/info
+%_mandir		%{_prefix}/share/man
 
 #==============================================================================
 # ---- config.guess platform macros.
