--- rpm-5.4.15/rpmdb/rpmdb.h.orig	2015-01-27 21:01:38.147301133 +0100
+++ rpm-5.4.15/rpmdb/rpmdb.h	2015-02-16 20:01:30.571602097 +0100
@@ -939,7 +939,7 @@
 /*@unused@*/ static inline
 int rpmlgcGet(rpmdb rpmdb, DB_LSN * _lsn, DBT * data, uint32_t flags)
 {
-    DB_LOGC * _logc = rpmdb->db_logc;
+    DB_LOGC * _logc = (DB_LOGC*)rpmdb->db_logc;
     int rc = (_logc ? _logc->get(_logc, _lsn, data, flags) : ENOTSUP);
 if (_rpmdb_debug)
 fprintf(stderr, "<-- %s(%p,%p,%p,0x%x) rc %d\n", "logc->get", _logc, _lsn, data, flags, rc);
@@ -949,7 +949,7 @@
 /*@unused@*/ static inline
 int rpmlgcClose(rpmdb rpmdb)
 {
-    DB_LOGC * _logc = rpmdb->db_logc;
+    DB_LOGC * _logc = (DB_LOGC*)rpmdb->db_logc;
     uint32_t _flags = 0;
     int rc = (_logc ? _logc->close(_logc, _flags) : ENOTSUP);
     rpmdb->db_logc = NULL;
@@ -1027,7 +1027,7 @@
 /*@unused@*/ static inline
 int rpmmpfClose(rpmdb rpmdb)
 {
-    DB_MPOOLFILE * mpf = rpmdb->db_mpf;
+    DB_MPOOLFILE * mpf = (DB_MPOOLFILE*)rpmdb->db_mpf;
     uint32_t _flags = 0;
     int rc = (mpf ? mpf->close(mpf, _flags) : ENOTSUP);
     rpmdb->db_mpf = NULL;
@@ -1039,7 +1039,7 @@
 /*@unused@*/ static inline
 int rpmmpfGet(rpmdb rpmdb, uint32_t * _pgnop, uint32_t _flags, void ** _pagep)
 {
-    DB_MPOOLFILE * mpf = rpmdb->db_mpf;
+    DB_MPOOLFILE * mpf = (DB_MPOOLFILE*)rpmdb->db_mpf;
     DB_TXN * _txnid = (DB_TXN*)rpmdb->db_txn;
     int rc = mpf->get(mpf, _pgnop, _txnid, _flags, _pagep);
 if (_rpmdb_debug)
@@ -1073,7 +1073,7 @@
 /*@unused@*/ static inline
 int rpmmpfPut(rpmdb rpmdb, void * _page, uint32_t flags)
 {
-    DB_MPOOLFILE * mpf = rpmdb->db_mpf;
+    DB_MPOOLFILE * mpf = (DB_MPOOLFILE*)rpmdb->db_mpf;
     uint32_t _priority = DB_PRIORITY_DEFAULT;
     int rc = mpf->put(mpf, _page, (DB_CACHE_PRIORITY)_priority, flags);
 if (_rpmdb_debug)
@@ -1084,7 +1084,7 @@
 /*@unused@*/ static inline
 int rpmmpfSync(rpmdb rpmdb)
 {
-    DB_MPOOLFILE * mpf = rpmdb->db_mpf;
+    DB_MPOOLFILE * mpf = (DB_MPOOLFILE*)rpmdb->db_mpf;
     int rc = (mpf ? mpf->sync(mpf) : ENOTSUP);
 if (_rpmdb_debug)
 fprintf(stderr, "<-- %s(%p) rc %d\n", "mpf->close", mpf, rc);
