--- rpm-3.0.1.old/rpmrc.in	Mon May 17 19:46:25 1999
+++ rpm-3.0.1/rpmrc.in	Mon May 17 20:12:06 1999
@@ -96,12 +96,12 @@
 buildarchtranslate: osfmach3_i486: i386
 buildarchtranslate: osfmach3_i386: i386
 
-buildarchtranslate: i986: i386
-buildarchtranslate: i886: i386
-buildarchtranslate: i786: i386
-buildarchtranslate: i686: i386
-buildarchtranslate: i586: i386
-buildarchtranslate: i486: i386
+buildarchtranslate: i986: i686
+buildarchtranslate: i886: i686
+buildarchtranslate: i786: i686
+buildarchtranslate: i686: i686
+buildarchtranslate: i586: i586
+buildarchtranslate: i486: i486
 buildarchtranslate: i386: i386
 
 buildarchtranslate: osfmach3_ppc: ppc
