--- rpm-2.93/macros.in.macros	Thu Apr  8 17:51:13 1999
+++ rpm-2.93/macros.in	Thu Apr  8 18:02:34 1999
@@ -66,13 +66,13 @@
 %_rpmdir		%{_topdir}/RPMS
 #
 # XXX Note escaped %% for use in headerSprintf
-%_rpmfilename		%%{ARCH}/%%{NAME}-%%{VERSION}-%%{RELEASE}.%%{ARCH}.rpm
+%_rpmfilename		%%{NAME}-%%{VERSION}-%%{RELEASE}.%%{ARCH}.rpm
 %_signature		none
 %_sourcedir		%{_topdir}/SOURCES
 %_specdir		%{_topdir}/SPECS
 %_srcrpmdir		%{_topdir}/SRPMS
 %_tmppath		%{_var}/tmp
-%_topdir		%{_usrsrc}/redhat
+%_topdir		%(echo $HOME)/rpm
 #
 #==============================================================================
 # ---- Optional rpmrc macros.
@@ -96,7 +96,7 @@
 #%_pgp_path
 #%_provides
 #%_timecheck
-#%vendor
+%vendor			PLD
 
 #==============================================================================
 # ---- per-platform macros.
