--- rpm-5.4.10/lib/psm.c~	2013-03-17 13:34:31.534601434 +0100
+++ rpm-5.4.10/lib/psm.c	2013-03-25 13:46:18.326658890 +0100
@@ -3017,7 +3017,7 @@
 
 	psm->rpmio_flags = t = (char *) xmalloc(sizeof("w9.gzdio"));
 	*t = '\0';
-	t = stpcpy(t, ((psm->goal == PSM_PKGSAVE) ? "w9" : "r"));
+	t = stpcpy(t, ((psm->goal == PSM_PKGSAVE) ? "w6" : "r"));
 	if (!strcmp(payload_compressor, "gzip"))
 	    t = stpcpy(t, ".gzdio");
 	if (!strcmp(payload_compressor, "bzip2"))
