diff -urBb rpm-4.4.2.org/macros.in rpm-4.4.2/macros.in
--- rpm-4.4.2.org/macros.in	2005-10-08 13:38:33.470982000 +0200
+++ rpm-4.4.2/macros.in	2005-10-08 13:39:51.327848250 +0200
@@ -1249,7 +1249,7 @@
 #	%{perl_sitearch}/Image
 #	%dir %{perl_sitearch}/auto/Image
 #
-%requires_eq()	%(LC_ALL="C" echo '%*' | xargs -r rpm -q --qf 'Requires: %%{name} = %%{epoch}:%%{version}\\n' | sed -e 's/ (none):/ /' -e 's/ 0:/ /' | grep -v "is not")
+%requires_eq()	%(echo '%*' | LC_ALL="C" xargs -r rpm -q --qf 'Requires: %%{name} = %%{epoch}:%%{version}\\n' | sed -e 's/ (none):/ /' -e 's/ 0:/ /' | grep -v "is not")
 %perl_sitearch	%(eval "`%{__perl} -V:installsitearch`"; echo $installsitearch)
 %perl_sitelib	%(eval "`%{__perl} -V:installsitelib`"; echo $installsitelib)
 %perl_vendorarch %(eval "`%{__perl} -V:installvendorarch`"; echo $installvendorarch)
