diff -ur rpm-3.0.1.old/configure.in rpm-3.0.1/configure.in
--- rpm-3.0.1.old/configure.in	Fri Jun  4 18:21:31 1999
+++ rpm-3.0.1/configure.in	Fri Jun  4 18:22:00 1999
@@ -781,7 +781,7 @@
 export LIBS INCPATH
 
 AC_CONFIG_SUBDIRS(popt)
-AC_OUTPUT([Makefile rpmrc macros lib/Makefile
+AC_OUTPUT([Makefile rpmrc macros macros.pld lib/Makefile
 	build/Makefile tools/Makefile scripts/Makefile
 	tests/Makefile tests/rpmrc tests/macros tests/hello-test/Makefile
 	misc/Makefile po/Makefile.in intl/Makefile],
Only in rpm-3.0.1: configure.in~
diff -ur rpm-3.0.1.old/macros.in rpm-3.0.1/macros.in
--- rpm-3.0.1.old/macros.in	Fri May 14 20:59:43 1999
+++ rpm-3.0.1/macros.in	Fri Jun  4 18:28:25 1999
@@ -60,12 +60,12 @@
 %_builddir		%{_topdir}/BUILD
 %_buildshell		/bin/sh
 %_bzip2bin		@BZIP2BIN@
-%_dbpath		%{_var}/lib/rpm
-%_defaultdocdir		%{_usr}/doc
+%_dbpath		%{_var}/db/rpm
+%_defaultdocdir		%{_usr}/share/doc
 #
 # XXX fixowner, fixgroup, and fixperms are run at the end of hardcoded setup
-%_fixowner		[ `%{__id} -u` = '0' ] && %{__chown} -Rf root
-%_fixgroup		[ `%{__id} -u` = '0' ] && %{__chgrp} -Rf @ROOT_GROUP@
+%_fixowner		[ `%{__id} -u` = '0' ] && %{__chown} -Rf root.@ROOT_GROUP@
+#%_fixgroup		[ `%{__id} -u` = '0' ] && %{__chgrp} -Rf @ROOT_GROUP@
 %_fixperms		%{__chmod} -Rf @FIXPERMS@
 #
 %_gzipbin		@GZIPBIN@
@@ -74,13 +74,13 @@
 %_rpmdir		%{_topdir}/RPMS
 #
 # XXX Note escaped %% for use in headerSprintf
-%_rpmfilename		%%{ARCH}/%%{NAME}-%%{VERSION}-%%{RELEASE}.%%{ARCH}.rpm
+%_rpmfilename		%%{NAME}-%%{VERSION}-%%{RELEASE}.%%{ARCH}.rpm
 %_signature		none
 %_sourcedir		%{_topdir}/SOURCES
 %_specdir		%{_topdir}/SPECS
 %_srcrpmdir		%{_topdir}/SRPMS
 %_tmppath		%{_var}/tmp
-%_topdir		%{_usrsrc}/redhat
+%_topdir		%(echo $HOME)/rpm
 #
 #==============================================================================
 # ---- Optional rpmrc macros.
@@ -159,8 +159,8 @@
 %_libdir		%{_exec_prefix}/lib
 %_includedir		%{_prefix}/include
 %_oldincludedir		/usr/include
-%_infodir		%{_prefix}/include
-%_mandir		%{_prefix}/man
+%_infodir		/usr/share/info
+%_mandir		%{_prefix}/share/man
 
 #==============================================================================
 # ---- config.guess platform macros.
Only in rpm-3.0.1: macros.in~
