diff -urN rpm-4.4.9/lib/rpmfc.c rpm-4.4.9.new/lib/rpmfc.c
--- rpm-4.4.9/lib/rpmfc.c	2008-07-27 13:26:12.000000000 +0200
+++ rpm-4.4.9.new/lib/rpmfc.c	2008-07-27 13:31:18.000000000 +0200
@@ -549,6 +549,11 @@
 
   { "Java ",			RPMFC_JAVA|RPMFC_INCLUDE },
 
+  /* .NET executables and libraries. file(1) cannot differ it from native win32 executables unfortunatelly */
+  { "Mono/.Net assembly",	RPMFC_MONO|RPMFC_INCLUDE },
+  { "PE executable",		RPMFC_MONO|RPMFC_INCLUDE },
+  { "executable PE",		RPMFC_MONO|RPMFC_INCLUDE },
+
   { "current ar archive",	RPMFC_STATIC|RPMFC_LIBRARY|RPMFC_ARCHIVE|RPMFC_INCLUDE },
 
   { "Zip archive data",		RPMFC_COMPRESSED|RPMFC_ARCHIVE|RPMFC_INCLUDE },
@@ -915,6 +920,35 @@
 }
 
 /**
+ * Extract .NET dependencies.
+ * @param fc		file classifier
+ * @return		0 on success
+ */
+static int rpmfcMONO(rpmfc fc)
+	/*@globals rpmGlobalMacroContext, h_errno, fileSystem, internalState @*/
+	/*@modifies fc, rpmGlobalMacroContext, fileSystem, internalState @*/
+{
+    const char * fn = fc->fn[fc->ix];
+    FILE * fp;
+    int xx;
+
+    fp = fopen(fn, "r");
+    if (fp == NULL || ferror(fp)) {
+	if (fp) (void) fclose(fp);
+	return -1;
+    }
+
+    (void) fclose(fp);
+
+    if (fc->findprov)
+	xx = rpmfcHelper(fc, 'P', "mono", fc->noautoprov, fc->noautoprov_c);
+    if (fc->findreq)
+	xx = rpmfcHelper(fc, 'R', "mono", fc->noautoreq, fc->noautoreq_c);
+
+    return 0;
+}
+
+/**
  * Extract Elf dependencies.
  * @param fc		file classifier
  * @return		0 on success
@@ -946,6 +980,7 @@
 static struct rpmfcApplyTbl_s rpmfcApplyTable[] = {
     { rpmfcELF,		RPMFC_ELF },
     { rpmfcSCRIPT,	(RPMFC_SCRIPT|RPMFC_PERL|RPMFC_PYTHON|RPMFC_LIBTOOL|RPMFC_PKGCONFIG|RPMFC_BOURNE|RPMFC_JAVA|RPMFC_PHP) },
+    { rpmfcMONO,	RPMFC_MONO },
     { NULL, 0 }
 };
 
diff -urN rpm-4.4.9/lib/rpmfc.h rpm-4.4.9.new/lib/rpmfc.h
--- rpm-4.4.9/lib/rpmfc.h	2008-07-27 13:26:12.000000000 +0200
+++ rpm-4.4.9.new/lib/rpmfc.h	2008-07-27 13:29:10.000000000 +0200
@@ -59,6 +59,7 @@
     RPMFC_PYTHON		= (1 << 26),
     RPMFC_PHP			= (1 << 27),
     RPMFC_TCL			= (1 << 28),
+    RPMFC_MONO			= (1 <<  6),
 
     RPMFC_WHITE			= (1 << 29),
     RPMFC_INCLUDE		= (1 << 30),
diff -urN rpm-4.4.9/macros.in rpm-4.4.9.new/macros.in
--- rpm-4.4.9/macros.in	2008-07-27 13:26:12.000000000 +0200
+++ rpm-4.4.9.new/macros.in	2008-07-27 13:29:10.000000000 +0200
@@ -1407,6 +1407,9 @@
 #%__php_provides		%{_usrlibrpm}/php.prov
 #%__php_requires		%{_usrlibrpm}/php.req
 
+%__mono_provides	%{nil}
+%__mono_requires	%{nil}
+
 #------------------------------------------------------------------------
 # java(...) configuration.
 #
