diff -ur rpm-4.4.9.org/macros.in rpm-4.4.9/macros.in
--- rpm-4.4.9.org/macros.in	2008-07-24 08:52:35.205363000 +0200
+++ rpm-4.4.9/macros.in	2008-07-24 09:01:00.351355186 +0200
@@ -275,6 +275,9 @@
 #
 #%distribution
 
+# PLD Linux Release
+%pld_release    th
+
 #	Configurable distribution URL, same as DistURL: tag in a specfile.
 #	The URL will be used to supply reliable information to tools like
 #	rpmfind.
--- rpm-4.5/macros.in.orig	2008-11-26 21:59:11.346746301 +0100
+++ rpm-4.5/macros.in	2008-11-26 22:01:02.728818761 +0100
@@ -1490,8 +1490,9 @@
 #          left here for compatibility
 %rpmcflags	%{?debug:%debugcflags}%{!?debug:%optflags}%{?debuginfocflags}
 %rpmcxxflags	%{rpmcflags}
-%rpmldflags	%{!?no_build_with_as_needed:-Wl,--as-needed}
-  
+%rpmldflags	%{!?no_build_with_as_needed:-Wl,--as-needed} -Wl,-z,relro -Wl,-z,combreloc
+%optcppflags	-D_FORTIFY_SOURCE=2
+
 #------------------------------------------------------------------------
 # mimetype(...) configuration.
 #
