--- rpm-4.2/rpmrc.in.wiget	Fri Jan 10 21:34:26 2003
+++ rpm-4.2/rpmrc.in	Thu Mar 27 18:56:27 2003
@@ -11,55 +11,56 @@
 #############################################################
 # Values for RPM_OPT_FLAGS for various platforms
 
-optflags: i386 -O2 -g -march=i386 -mcpu=i686
-optflags: i486 -O2 -g -march=i486
-optflags: i586 -O2 -g -march=i586
-optflags: i686 -O2 -g -march=i686
-optflags: athlon -O2 -g -march=athlon
-optflags: ia64 -O2 -g
-optflags: x86_64 -O2 -g
-
-optflags: alpha -O2 -g -mieee
-optflags: alphaev5 -O2 -g -mieee -mcpu=ev5
-optflags: alphaev56 -O2 -g -mieee -mcpu=ev56
-optflags: alphapca56 -O2 -g -mieee -mcpu=pca56
-optflags: alphaev6 -O2 -g -mieee -mcpu=ev6
-optflags: alphaev67 -O2 -g -mieee -mcpu=ev67
-
-optflags: sparc -O2 -g -m32 -mtune=ultrasparc
-optflags: sparcv9 -O2 -g -m32 -mcpu=ultrasparc
-optflags: sparc64 -O2 -g -m64 -mcpu=ultrasparc
-
-optflags: m68k -O2 -g -fomit-frame-pointer
-
-optflags: ppc -O2 -g -fsigned-char
-optflags: ppciseries -O2 -g -fsigned-char
-optflags: ppcpseries -O2 -g -fsigned-char
-optflags: ppc64 -O2 -g -fsigned-char
-
-optflags: parisc -O2 -g -mpa-risc-1-0
-optflags: hppa1.0 -O2 -g -mpa-risc-1-0
-optflags: hppa1.1 -O2 -g -mpa-risc-1-0
-optflags: hppa1.2 -O2 -g -mpa-risc-1-0
-optflags: hppa2.0 -O2 -g -mpa-risc-1-0
-
-optflags: mips -O2 -g
-optflags: mipsel -O2 -g
-
-optflags: armv3l -O2 -g -fsigned-char -fomit-frame-pointer -march=armv3
-optflags: armv4b -O2 -g -fsigned-char -fomit-frame-pointer -march=armv4
-optflags: armv4l -O2 -g -fsigned-char -fomit-frame-pointer -march=armv4
-
-optflags: atarist -O2 -g -fomit-frame-pointer
-optflags: atariste -O2 -g -fomit-frame-pointer
-optflags: ataritt -O2 -g -fomit-frame-pointer
-optflags: falcon -O2 -g -fomit-frame-pointer
-optflags: atariclone -O2 -g -fomit-frame-pointer
-optflags: milan -O2 -g -fomit-frame-pointer
-optflags: hades -O2 -g -fomit-frame-pointer
+optflags: i386 -O2 -march=i386
+#optflags: i386 -O2 -march=i386 -mcpu=i686
+optflags: i486 -O2 -march=i486
+optflags: i586 -O2 -march=i586
+optflags: i686 -O2 -march=i686
+optflags: athlon -O2 -march=athlon
+optflags: ia64 -O2 
+optflags: x86_64 -O2 
+
+optflags: alpha -O2 -mieee
+optflags: alphaev5 -O2  -mieee -mcpu=ev5
+optflags: alphaev56 -O2  -mieee -mcpu=ev56
+optflags: alphapca56 -O2  -mieee -mcpu=pca56
+optflags: alphaev6 -O2  -mieee -mcpu=ev6
+optflags: alphaev67 -O2  -mieee -mcpu=ev67
+
+optflags: sparc -O2  -m32 -mtune=ultrasparc
+optflags: sparcv9 -O2  -m32 -mcpu=ultrasparc
+optflags: sparc64 -O2  -m64 -mcpu=ultrasparc
+
+optflags: m68k -O2  -fomit-frame-pointer
+
+optflags: ppc -O2  -fsigned-char
+optflags: ppciseries -O2  -fsigned-char
+optflags: ppcpseries -O2  -fsigned-char
+optflags: ppc64 -O2  -fsigned-char
+
+optflags: parisc -O2  -mpa-risc-1-0
+optflags: hppa1.0 -O2  -mpa-risc-1-0
+optflags: hppa1.1 -O2  -mpa-risc-1-0
+optflags: hppa1.2 -O2  -mpa-risc-1-0
+optflags: hppa2.0 -O2  -mpa-risc-1-0
+
+optflags: mips -O2 
+optflags: mipsel -O2 
+
+optflags: armv3l -O2  -fsigned-char -fomit-frame-pointer -march=armv3
+optflags: armv4b -O2  -fsigned-char -fomit-frame-pointer -march=armv4
+optflags: armv4l -O2  -fsigned-char -fomit-frame-pointer -march=armv4
+
+optflags: atarist -O2  -fomit-frame-pointer
+optflags: atariste -O2  -fomit-frame-pointer
+optflags: ataritt -O2  -fomit-frame-pointer
+optflags: falcon -O2  -fomit-frame-pointer
+optflags: atariclone -O2  -fomit-frame-pointer
+optflags: milan -O2  -fomit-frame-pointer
+optflags: hades -O2  -fomit-frame-pointer
 
-optflags: s390 -O2 -g
-optflags: s390x -O2 -g
+optflags: s390 -O2 
+optflags: s390x -O2 
 
 #############################################################
 # Canonical arch names and numbers
