--- rpm-4.5/scripts/perl.req	2008-07-09 12:38:31.000000000 +0300
+++ rpm-4.4.9/scripts/perl.req.in	2008-08-12 23:08:33.518675858 +0300
@@ -277,20 +277,20 @@
 	  # $V-style
 	  if ($ver =~ m/5\.([0-5])(\.([0-9]+))?$/) {
 	    if (defined $3) {
-	      print "perl >= 0:5.00$1_$3\n";
+	      print "perl-base >= 0:5.00$1_$3\n";
 	    } else {
-	      print "perl >= 0:5.00$1\n";
+	      print "perl-base >= 0:5.00$1\n";
 	    }
 	  } else {
-	    print "perl >= 1:$ver\n";
+	    print "perl-base >= 1:$ver\n";
 	  }
 	} else {
 	  # $]-style
 	  if ($ver =~ m/5\.(00[0-5])_?([0-9]+)?$/) {
 	    if (defined $2) {
-	      print "perl >= 0:5.$1_$2\n";
+	      print "perl-base >= 0:5.$1_$2\n";
 	    } else {
-	      print "perl >= 0:5.$1\n";
+	      print "perl-base >= 0:5.$1\n";
 	    }
 	  } else {
 	    # expand to M.NNN_nnn form
@@ -299,7 +299,7 @@
 	    $ver =~ s/^([0-9]\.[0-9]{3})_?([0-9]{3})0*$/$1_$2/;
 	    # match trimming leading 0s
 	    $ver =~ m/^([0-9])\.0*([1-9][0-9]*|0)_0*([1-9][0-9]*|0)$/;
-	    print "perl >= 1:$1.$2.$3\n";
+	    print "perl-base >= 1:$1.$2.$3\n";
 	  }
 	}
 	next;
@@ -326,6 +326,10 @@
 
       ($module =~ m/\.(ph|pl|t)$/) && next;
 
+      # skip all modules for platforms other than linux.
+
+      ($module =~ m/Mac|OS2|MSDOS|Win32|VMS|vmsish/) && next;
+
       # if the module name starts in a dot it is not a module name.
 
       ($module =~ m/^\./) && next;
