--- rpm-4.3/po/pl.po.wiget	2004-03-29 17:13:44.119089525 +0200
+++ rpm-4.3/po/pl.po	2004-03-29 17:30:33.363456251 +0200
@@ -4149,3 +4149,7 @@
 
 msgid "Help options:"
 msgstr "Opcje pomocy:"
+
+#: lib/rpmrc.c:1764
+msgid "Cannot read %s, HOME_ETC is too large.\n"
+msgstr "Nie mo¿na odczytaæ %s, HOME_ETC jest zbyt du¿e.\n"
--- rpm-4.3/lib/rpmrc.c.wiget	2004-03-29 13:16:41.000000000 +0200
+++ rpm-4.3/lib/rpmrc.c	2004-03-29 17:13:44.218073938 +0200
@@ -1828,27 +1828,39 @@
 	/* Expand ~/ to $HOME/ */
 	fn[0] = '\0';
 	if (r[0] == '~' && r[1] == '/') {
+			const char * etc_dir = getenv("HOME_ETC");
 	    const char * home = getenv("HOME");
-	    if (home == NULL) {
+			if (etc_dir) {
+	    	if (strlen(etc_dir) > (sizeof(fn) - strlen(r))) {
+					rpmError(RPMERR_RPMRC, _("Cannot read %s, HOME_ETC is too large.\n"),r);
+					rc = 1;
+					break;
+	    	}
+	    	strcpy(fn, etc_dir);
+				strncat(fn, "/", sizeof(fn) - strlen(fn));				
+	  	  r+=2;
+		
+			} else {
+	    	if (home == NULL) {
 	    /* XXX Only /usr/lib/rpm/rpmrc must exist in default rcfiles list */
-		if (rcfiles == defrcfiles && myrcfiles != r)
-		    continue;
-		rpmError(RPMERR_RPMRC, _("Cannot expand %s\n"), r);
-		rc = 1;
-		break;
-	    }
-	    if (strlen(home) > (sizeof(fn) - strlen(r))) {
-		rpmError(RPMERR_RPMRC, _("Cannot read %s, HOME is too large.\n"),
+			if (rcfiles == defrcfiles && myrcfiles != r)
+		    	continue;
+			rpmError(RPMERR_RPMRC, _("Cannot expand %s\n"), r);
+			rc = 1;
+			break;
+	    	}
+	    	if (strlen(home) > (sizeof(fn) - strlen(r))) {
+			rpmError(RPMERR_RPMRC, _("Cannot read %s, HOME is too large.\n"),
 				r);
-		rc = 1;
-		break;
-	    }
-	    strcpy(fn, home);
-	    r++;
-	}
-	strncat(fn, r, sizeof(fn) - (strlen(fn) + 1));
-	fn[sizeof(fn)-1] = '\0';
-
+			rc = 1;
+			break;
+	    	}
+	    	strcpy(fn, home);
+	  	  r++;
+			}
+		}
+		strncat(fn, r, sizeof(fn) - (strlen(fn) + 1));
+		fn[sizeof(fn)-1] = '\0';
 	/* Read another rcfile */
 	fd = Fopen(fn, "r.fpio");
 	if (fd == NULL || Ferror(fd)) {
