--- rpm-3.0.3/rpmpopt.wiget	Fri Sep 24 03:50:04 1999
+++ rpm-3.0.3/rpmpopt	Tue Oct 19 16:56:16 1999
@@ -72,3 +72,5 @@
 
 rpm alias --redhatprovides -q --define '_dbpath /usr/lib/rpmdb/%{_arch}-%{_vendor}-%{_os}/redhat' --whatprovides
 rpm alias --redhatrequires -q --define '_dbpath /usr/lib/rpmdb/%{_arch}-%{_vendor}-%{_os}/redhat' --whatrequires
+
+rpm exec --install-build-tree install-build-tree
