--- rpm-4.4.9/macros.in	2008-01-13 19:07:45.023017755 +0200
+++ rpm-4.4.9/macros.in	2008-01-10 13:18:10.454983604 +0200
@@ -350,7 +350,7 @@
 #		"w9.lzdio"	lzma level 9.
 #
 %_source_payload	w9.gzdio
-%_binary_payload	w9.lzdio
+%_binary_payload	w9.bzdio
 
 #	Archive formats to use for source/binary package payloads.
 #		"cpio"		cpio archive (default)
@@ -1426,8 +1426,8 @@
 #
 # Note: Used iff _use_internal_dependency_generator is non-zero. The
 # helpers are also used by %{_usrlibrpm}/rpmdeps {--provides|--requires}.
-%__libtool_provides	%{_usrlibrpm}/libtooldeps.sh --provides %{buildroot} %{name}
-%__libtool_requires	%{_usrlibrpm}/libtooldeps.sh --requires %{buildroot} %{name}
+#%__libtool_provides	%{_usrlibrpm}/libtooldeps.sh --provides %{buildroot} %{name}
+#%__libtool_requires	%{_usrlibrpm}/libtooldeps.sh --requires %{buildroot} %{name}
 
 #------------------------------------------------------------------------
 # pkgconfig(...) configuration.
@@ -1436,8 +1436,8 @@
 #
 # Note: Used iff _use_internal_dependency_generator is non-zero. The
 # helpers are also used by %{_usrlibrpm}/rpmdeps {--provides|--requires}.
-%__pkgconfig_provides	%{_usrlibrpm}/pkgconfigdeps.sh --provides
-%__pkgconfig_requires	%{_usrlibrpm}/pkgconfigdeps.sh --requires
+#%__pkgconfig_provides	%{_usrlibrpm}/pkgconfigdeps.sh --provides
+#%__pkgconfig_requires	%{_usrlibrpm}/pkgconfigdeps.sh --requires
 
 #------------------------------------------------------------------------
 # executable(...) configuration.
@@ -1451,7 +1451,7 @@
 #%__scriptlet_requires	/bin/bash --rpm-requires
 
 # PLD rpm macros
-%_enable_debug_packages 1
+%_enable_debug_packages 0
 
 #-----------------------------------------------------------------
 # CFLAGS and LDFLAGS used to build
