--- rpm-5.4.10/build/files.c~	2013-01-25 13:14:36.692060288 +0100
+++ rpm-5.4.10/build/files.c	2013-02-25 12:03:26.110910465 +0100
@@ -1041,7 +1041,6 @@
 	/* XXX FIXME: this is easy to do as macro expansion */
 
 	    if (! fl->passedSpecialDoc) {
-	    	char *compress_doc;
 	    	char *mkdir_p;
 
 		pkg->specialDoc = rpmiobNew(0);
@@ -1056,11 +1055,6 @@
 		mkdir_p = _free(mkdir_p);
 		pkg->specialDoc = rpmiobAppend(pkg->specialDoc, " \"$DOCDIR\"", 1);
 
-		compress_doc = rpmExpand("%{__compress_doc}", NULL);
-		if (compress_doc && *compress_doc != '%')
-	    	    pkg->specialDoc = rpmiobAppend(pkg->specialDoc, compress_doc, 1);
-		compress_doc = _free(compress_doc);
-
 		/*@-temptrans@*/
 		*fileName = buf;
 		/*@=temptrans@*/
@@ -1071,6 +1065,15 @@
 	    pkg->specialDoc = rpmiobAppend(pkg->specialDoc, "cp -pr ", 0);
 	    pkg->specialDoc = rpmiobAppend(pkg->specialDoc, specialDocBuf, 0);
 	    pkg->specialDoc = rpmiobAppend(pkg->specialDoc, " \"$DOCDIR\"", 1);
+
+	    {
+	    	char *compress_doc;
+
+		compress_doc = rpmExpand("%{__compress_doc}", NULL);
+		if (compress_doc && *compress_doc != '%')
+	    	    pkg->specialDoc = rpmiobAppend(pkg->specialDoc, compress_doc, 1);
+		compress_doc = _free(compress_doc);
+	    }
 	}
     }
 
