diff -ur rpm-4.0-/lib/depends.c rpm-4.0/lib/depends.c
--- rpm-4.0-/lib/depends.c	Sun Jan 28 22:08:47 2001
+++ rpm-4.0/lib/depends.c	Sun Jan 28 22:10:40 2001
@@ -899,8 +899,10 @@
 
     /*
      * Check if dbiOpen/dbiPut failed (e.g. permissions), we can't cache.
+     *
+     * We can't relay on cache when we need sugesstions.
      */
-    if (_cacheDependsRC) {
+    if (_cacheDependsRC && suggestion == NULL) {
 	dbiIndex dbi;
 	dbi = dbiOpen(rpmdep->rpmdb, RPMDBI_DEPENDS, 0);
 	if (dbi == NULL)
