--- rpm-4.2/rpmdb/Makefile.am.wiget	Fri Mar 28 00:48:52 2003
+++ rpm-4.2/rpmdb/Makefile.am	Fri Mar 28 01:06:18 2003
@@ -26,11 +26,16 @@
 noinst_HEADERS = fprint.h header_internal.h legacy.h
 
 pkgbindir = @RPMCONFIGDIR@
+if WITHOUT_INTERNAL_DB
+pkgbin_PROGRAMS =
+noinst_PROGRAMS =
+else
 pkgbin_PROGRAMS = \
 	rpmdb_deadlock rpmdb_dump rpmdb_load rpmdb_svc rpmdb_stat rpmdb_verify
 noinst_PROGRAMS = \
 	rpmdb_archive rpmdb_checkpoint rpmdb_printlog rpmdb_recover \
 	rpmdb_upgrade
+endif
 
 mylibs = librpmdb.la
 
--- rpm-4.2/configure.ac.wiget	Fri Mar 28 01:06:49 2003
+++ rpm-4.2/configure.ac	Fri Mar 28 01:11:34 2003
@@ -500,6 +497,7 @@
 	[libdb3="$libdb3 $libthread"; AC_MSG_RESULT(yes)])
    LIBS="$saveLIBS"
 fi
+without_internal_db=yes
 
 else
 dnl ------------------ with    internal db
@@ -510,7 +508,9 @@
   libdb3="# \$(top_builddir)/db3/libdb.la"
   libdb3a="\$(top_builddir)/db3/libdb.a"
 dnl  INCPATH="-I\$(top_builddir)/$(WITH_DB_SUBDIR) $INCPATH"
+  without_internal_db=no
 fi
+AM_CONDITIONAL(WITHOUT_INTERNAL_DB, test x$without_internal_db = xyes)
 
 AC_SUBST(WITH_DB_SUBDIR)
 AC_SUBST(WITH_INTERNAL_DB)
