diff --color -urN rpm-4.16.0.orig/fileattrs/pythondist.attr rpm-4.16.0/fileattrs/pythondist.attr
--- rpm-4.16.0.orig/fileattrs/pythondist.attr	2020-12-09 12:38:21.203961862 +0100
+++ rpm-4.16.0/fileattrs/pythondist.attr	2020-12-09 12:58:43.178952892 +0100
@@ -1,3 +1,3 @@
 %__pythondist_provides	%{_rpmconfigdir}/pythondistdeps.py --provides --majorver-provides
 %__pythondist_requires	%{_rpmconfigdir}/pythondistdeps.py --requires
-%__pythondist_path		/lib(64|x32)?/python[[:digit:]]\\.[[:digit:]]+/site-packages/[^/]+\\.(dist-info|egg-info|egg-link)$
+%__pythondist_path		/(lib(64|x32)?|share)/python[[:digit:]]\\.[[:digit:]]+/site-packages/[^/]+\\.(dist-info|egg-info|egg-link)$
