--- rpm-4.3/build/files.c.orig	Thu Jul 10 12:05:14 2003
+++ rpm-4.3/build/files.c	Thu Jul 10 12:07:25 2003
@@ -1977,9 +1977,11 @@
     fl.docDirs[fl.docDirCount++] = xstrdup("/usr/share/doc");
     fl.docDirs[fl.docDirCount++] = xstrdup("/usr/share/man");
     fl.docDirs[fl.docDirCount++] = xstrdup("/usr/share/info");
+    fl.docDirs[fl.docDirCount++] = xstrdup("/usr/src/examples");
     fl.docDirs[fl.docDirCount++] = rpmGetPath("%{_docdir}", NULL);
     fl.docDirs[fl.docDirCount++] = rpmGetPath("%{_mandir}", NULL);
     fl.docDirs[fl.docDirCount++] = rpmGetPath("%{_infodir}", NULL);
+    fl.docDirs[fl.docDirCount++] = rpmGetPath("%{_examplesdir}", NULL);
     
     fl.fileList = NULL;
     fl.fileListRecsAlloced = 0;
