--- rpm-4.4.9/macros.in	2008-04-04 23:15:28.121279270 +0300
+++ rpm-4.5/macros.in	2008-09-03 16:47:30.861766714 +0300
@@ -109,6 +109,10 @@
 %__automake		automake
 %__autoconf		autoconf
 
+# compiler used to build kernel and kernel modules
+%kgcc		%{__cc}
+%kgcc_package	gcc
+
 #==============================================================================
 # Conditional build stuff.
 
@@ -206,7 +210,7 @@
 %endif\
 %{nil}
 
-%_defaultdocdir		%{_usr}/doc
+%_defaultdocdir		%{_usr}/share/doc
 
 #	The path to the pgp executable (legacy, use %{__pgp} instead).
 %_pgpbin		%{__pgp}
@@ -233,12 +237,12 @@
 #	The directory where newly built source packages will be written.
 %_srcrpmdir		%{_topdir}/SRPMS
 
-#	Directory where temporaray files can be created.
-%_tmppath		%{_var}/tmp
+#	Directory where temporary files can be created.
+%_tmppath		%(echo "${TMPDIR:-/tmp}")
 %tmpdir			%{_tmppath}
 
 #	Path to top of build area.
-%_topdir		%{_usrsrc}/rpm
+%_topdir		%(echo $HOME)/rpm
 
 #==============================================================================
 # ---- Optional rpmrc macros.
@@ -248,7 +252,7 @@
 #	Configurable build root path, same as BuildRoot: in a specfile.
 #	(Note: the configured macro value will override the spec file value).
 #
-%buildroot		%{_tmppath}/%{name}-root
+%buildroot		%{_tmppath}/%{name}-%{version}-root-%(id -u -n)
 
 #	The sub-directory (relative to %{_builddir}) where sources are compiled.
 #	This macro is set after processing %setup, either explicitly from the
@@ -347,8 +351,8 @@
 #		"w9.bzdio"	bzip2 level 9.
 #		"w9.lzdio"	lzma level 9.
 #
-#%_source_payload	w9.gzdio
-#%_binary_payload	w9.gzdio
+%_source_payload	w9.gzdio
+%_binary_payload	w9.lzdio
 
 #	Archive formats to use for source/binary package payloads.
 #		"cpio"		cpio archive (default)
@@ -508,7 +512,7 @@
 
 #
 # Path to magic file used for file classification.
-%_rpmfc_magic_path	%{_rpmhome}/magic
+%_rpmfc_magic_path	/usr/share/file/magic
 
 #==============================================================================
 # ---- Database configuration macros.
@@ -954,7 +954,7 @@
 
 %___build_shell		%{?_buildshell:%{_buildshell}}%{!?_buildshell:/bin/sh}
 %___build_args		-e
-%___build_cmd		%{?_sudo:%{_sudo} }%{?_remsh:%{_remsh} %{_remhost} }%{?_remsudo:%{_remsudo} }%{?_remchroot:%{_remchroot} %{_remroot} }%{___build_shell} %{___build_args}
+%___build_cmd		%{?_sudo:%{_sudo} }%{?_remsh:%{_remsh} %{_remhost} }%{?_remsudo:%{_remsudo} }%{?_remchroot:%{_remchroot} %{_remroot} }%{?_clean_env:%{_clean_env} }%{___build_shell} %{___build_args}
 %___build_pre	\
   RPM_SOURCE_DIR=\"%{u2p:%{_sourcedir}}\"\
   RPM_BUILD_DIR=\"%{u2p:%{_builddir}}\"\
@@ -991,9 +995,12 @@
   %{?_javaclasspath:CLASSPATH=\"%{_javaclasspath}\"\
   export CLASSPATH}\
   unset PERL_MM_OPT || :\
-  LANG=C\
-  export LANG\
-  unset DISPLAY || :\
+  export LC_ALL=C\
+  export LANG=C\
+  unset LINGUAS ||:\
+  unset LANGUAGE ||:\
+  unset LC_MESSAGES ||:\
+  unset DISPLAY ||:\
   \
   %{verbose:set -x}%{!verbose:exec > /dev/null}\
   umask 022\
@@ -1127,17 +1134,17 @@
 %_exec_prefix		%{_prefix}
 %_bindir		%{_exec_prefix}/bin
 %_sbindir		%{_exec_prefix}/sbin
-%_libexecdir		%{_exec_prefix}/libexec
+%_libexecdir		%{_exec_prefix}/lib
 %_datadir		%{_prefix}/share
-%_sysconfdir		%{_prefix}/etc
-%_sharedstatedir	%{_prefix}/com
-%_localstatedir		%{_prefix}/var
+%_sysconfdir		/etc
+%_sharedstatedir	/var/lib
+%_localstatedir		/var
 %_lib			lib
 %_libdir		%{_exec_prefix}/%{_lib}
 %_includedir		%{_prefix}/include
 %_oldincludedir		/usr/include
-%_infodir		%{_prefix}/info
-%_mandir		%{_prefix}/man
+%_infodir		%{_prefix}/share/info
+%_mandir		%{_prefix}/share/man
 %_localedir		%{_datadir}/locale
 
 #==============================================================================
@@ -1429,6 +1436,22 @@
 #%__executable_provides	%{_usrlibrpm}/executabledeps.sh --provides
 #%__executable_requires	%{_usrlibrpm}/executabledeps.sh --requires
-%__scriptlet_requires	/bin/bash --rpm-requires
+#%__scriptlet_requires	/bin/bash --rpm-requires
+
+# PLD rpm macros
+%_enable_debug_packages 1
+
+#-----------------------------------------------------------------
+# CFLAGS and LDFLAGS used to build
+
+%debuginfocflags	%{expand:%%define __dic_%{?_enable_debug_packages} 1}%{?__dic_1: -gdwarf-4 -fno-debug-types-section -g2}%{expand:%%undefine __dic_%{?_enable_debug_packages}}
+
+%debugcflags	-O0 -g -Wall
 
+# Warning: those macros are overwritten by macros.build,
+#          left here for compatibility
+%rpmcflags	%{?debug:%debugcflags}%{!?debug:%optflags}%{?debuginfocflags}
+%rpmcxxflags	%{rpmcflags}
+%rpmldflags	%{!?no_build_with_as_needed:-Wl,--as-needed}
+  
 # \endverbatim
 #*/
@@ -1480,7 +1504,7 @@
 #
 # Note: Used if _use_internal_dependency_generator is non-zero. The
 # helper is also used by %{_rpmhome}/rpmdeps --provides
-#%__mimetype_provides %{_rpmhome}/mimetypedeps.sh --provides
+%__mimetype_provides %{_rpmhome}/mimetypedeps.sh --provides
 
 # \endverbatim
 #*/
--- rpm-4.5/configure.ac~	2008-06-10 02:03:07.000000000 +0300
+++ rpm-4.5/configure.ac	2008-06-10 02:04:18.395836371 +0300
@@ -1479,7 +1479,7 @@
 	[Full path to rpm system configuration directory (usually /etc/rpm)])
 AC_SUBST(SYSCONFIGDIR)
 
-MACROFILES="${USRLIBRPM}/${VERSION}/macros:${USRLIBRPM}/%{_target}/macros:${SYSCONFIGDIR}/%{_host_vendor}/macros:${SYSCONFIGDIR}/%{_host_vendor}/%{_target}/macros:${SYSCONFIGDIR}/macros.*:${SYSCONFIGDIR}/macros.d/*.macros:${SYSCONFIGDIR}/macros:${SYSCONFIGDIR}/%{_target}/macros:~/.rpmmacros"
+MACROFILES="${USRLIBRPM}/macros:${USRLIBRPM}/macros.build:${USRLIBRPM}/%{_target}/macros:${SYSCONFIGDIR}/macros.*:${SYSCONFIGDIR}/macros:${SYSCONFIGDIR}/%{_target}/macros:~/etc/.rpmmacros:~/.rpmmacros"
 AC_DEFINE_UNQUOTED(MACROFILES, "$MACROFILES",
 	[Colon separated paths of macro files to read.])
 AC_SUBST(MACROFILES)
