--- rpm-4.0.4/configure.in.wiget	Tue Feb 12 23:26:36 2002
+++ rpm-4.0.4/configure.in	Tue Feb 12 23:27:46 2002
@@ -1166,6 +1166,7 @@
 
 AC_OUTPUT([ Doxyfile Makefile rpmrc macros platform rpmpopt rpm.spec
 	macros.perl scripts/perl.req scripts/perl.prov
+	macros.python macros.php scripts/php.req scripts/php.prov
 	rpmio/Makefile rpmdb/Makefile lib/Makefile build/Makefile
 	scripts/Makefile scripts/brp-redhat tools/Makefile
 	tests/Makefile tests/rpmrc tests/macros tests/hello-test/Makefile
