diff -Nur rpm-3.0.1/macros.in rpm-3.0.1.pld/macros.in
--- rpm-3.0.1/macros.in	Tue May 11 11:24:08 1999
+++ rpm-3.0.1.pld/macros.in	Mon May 17 20:07:52 1999
@@ -55,8 +55,8 @@
 %_builddir		%{_topdir}/BUILD
 %_buildshell		/bin/sh
 %_bzip2bin		@BZIP2BIN@
-%_dbpath		%{_var}/lib/rpm
-%_defaultdocdir		%{_usr}/doc
+%_dbpath		%{_var}/db/rpm
+%_defaultdocdir		%{_usr}/share/doc
 #
 # XXX fixowner, fixgroup, and fixperms are run at the end of hardcoded setup
 %_fixowner		[ `%{__id} -u` = '0' ] && %{__chown} -Rf root
@@ -69,13 +69,13 @@
 %_rpmdir		%{_topdir}/RPMS
 #
 # XXX Note escaped %% for use in headerSprintf
-%_rpmfilename		%%{ARCH}/%%{NAME}-%%{VERSION}-%%{RELEASE}.%%{ARCH}.rpm
+%_rpmfilename		%%{NAME}-%%{VERSION}-%%{RELEASE}.%%{ARCH}.rpm
 %_signature		none
 %_sourcedir		%{_topdir}/SOURCES
 %_specdir		%{_topdir}/SPECS
 %_srcrpmdir		%{_topdir}/SRPMS
 %_tmppath		%{_var}/tmp
-%_topdir		%{_usrsrc}/redhat
+%_topdir		%(echo $HOME)/rpm
 #
 #==============================================================================
 # ---- Optional rpmrc macros.
@@ -154,8 +154,8 @@
 %_libdir		%{_exec_prefix}/lib
 %_includedir		%{_prefix}/include
 %_oldincludedir		/usr/include
-%_infodir		%{_prefix}/include
-%_mandir		%{_prefix}/man
+%_infodir		/usr/share/info
+%_mandir		%{_prefix}/share/man
 
 #==============================================================================
 # ---- config.guess platform macros.
diff -ru rpm-3.0.1.old/configure.in rpm-3.0.1/configure.in
--- rpm-3.0.1.old/configure.in	Wed May 19 10:17:34 1999
+++ rpm-3.0.1/configure.in	Wed May 19 10:18:32 1999
@@ -781,7 +781,7 @@
 export LIBS INCPATH
 
 AC_CONFIG_SUBDIRS(popt)
-AC_OUTPUT([Makefile rpmrc macros lib/Makefile
+AC_OUTPUT([Makefile rpmrc macros macros.pld lib/Makefile
 	build/Makefile tools/Makefile scripts/Makefile
 	tests/Makefile tests/rpmrc tests/macros tests/hello-test/Makefile
 	misc/Makefile po/Makefile.in intl/Makefile],
