diff -Nur rpm-4.0.1.orig/rpmio/macro.c rpm-4.0.1/rpmio/macro.c
--- rpm-4.0.1.orig/rpmio/macro.c	Fri Jan 19 02:47:25 2001
+++ rpm-4.0.1/rpmio/macro.c	Tue Feb 13 13:11:08 2001
@@ -801,7 +801,7 @@
     /* Build argv array */
     argv = (const char **) alloca((argc + 1) * sizeof(char *));
     be[-1] = ' ';	/*  be - 1 == b + strlen(b) == buf + strlen(buf)  */
-    buf[0] = '\0';
+    be[0] = '\0';
     b = buf;
     for (c = 0; c < argc; c++) {
 	argv[c] = b;
