--- rpm-4.5/macros.in~	2009-06-05 00:50:12.000000000 +0300
+++ rpm-4.5/macros.in	2009-06-05 01:25:32.648494029 +0300
@@ -746,8 +746,8 @@
 
 # Horowitz Key Protocol server configuration
 #
-%_hkp_keyserver         hkp://subkeys.pgp.net
-%_hkp_keyserver_query   %{_hkp_keyserver}/pks/lookup?op=get&search=0x
+#%_hkp_keyserver         hkp://subkeys.pgp.net
+#%_hkp_keyserver_query   %{_hkp_keyserver}/pks/lookup?op=get&search=0x
 
 #==============================================================================
 # ---- Transaction macros.
