diff -Nur rpm-4.4.2.orig/build/Makefile.am rpm-4.4.2.as_needed-fix/build/Makefile.am
--- rpm-4.4.2.orig/build/Makefile.am	2006-08-04 22:17:16.000000000 +0200
+++ rpm-4.4.2.as_needed-fix/build/Makefile.am	2006-08-04 22:17:43.000000000 +0200
@@ -33,6 +33,7 @@
 	$(top_builddir)/lib/librpm.la \
 	$(top_builddir)/rpmdb/librpmdb.la \
 	$(top_builddir)/rpmio/librpmio.la \
+	@WITH_MAGIC_LIB@ \
 	@WITH_LIBELF_LIB@
 
 rpmfile.h:
