--- rpm-4.4.9/macros.in	2008-02-20 20:05:29.045275299 +0200
+++ rpm-4.4.9/macros.in	2008-03-04 21:34:41.519716362 +0200
@@ -192,7 +192,7 @@
 Summary: Debug information for package %{name}\
 Summary(pl.UTF-8): Informacje dla debuggera dla pakietu %{name}\
 Group: Development/Debug\
-Requires: filesystem-debuginfo >= 3.0-16\
+Requires: filesystem-debuginfo >= 2.0-7\
 AutoReqProv: 0\
 %description debuginfo\
 This package provides debug information for package %{name}.\
@@ -237,7 +237,7 @@
 %_srcrpmdir		%{_topdir}/SRPMS
 
 #	Directory where temporary files can be created.
-%_tmppath		%(echo "${TMPDIR:-/tmp}")
+%_tmppath		%{_var}/tmp
 %tmpdir			%{_tmppath}
 
 #	Path to top of build area.
@@ -1156,6 +1156,7 @@
 %_infodir		%{_prefix}/share/info
 %_mandir		%{_prefix}/share/man
 %_localedir		%{_datadir}/locale
+%_x_libraries	/usr/X11R6/%{_lib}
 
 #==============================================================================
 # ---- config.guess platform macros.
