--- rpm-5.4.16/lua/Makefile.am~	2016-04-19 19:19:35.000000000 +0300
+++ rpm-5.4.16/lua/Makefile.am	2016-04-20 16:51:20.574586445 +0300
@@ -326,7 +326,10 @@
 	lzio.c
 
 liblua_la_DEPENDENCIES =
-liblua_la_LIBADD = @WITH_LUA_SUBDIR_LIB@ -lm -lexpat
+liblua_la_LIBADD = @WITH_LUA_SUBDIR_LIB@ -lm
+if USE_LUA_CRYPTO
+liblua_la_LIBADD += -lexpat
+endif
 liblua_la_LDFLAGS = $(LDFLAGS)
 
 liblua.la: $(liblua_la_OBJECTS) $(liblua_la_DEPENDENCIES) 
