--- rpm-5.3.1/configure.ac.wiget	2010-05-30 16:43:06.000000000 +0200
+++ rpm-5.3.1/configure.ac	2010-05-30 16:43:38.482725491 +0200
@@ -2199,6 +2199,7 @@ AC_CONFIG_FILES([ po/Makefile.in
 AC_CONFIG_FILES([Makefile
 	rpmio/Makefile lib/Makefile build/Makefile sign/Makefile
 	po/Makefile.in scripts/Makefile scripts/perl.req scripts/perl.prov fileattrs/Makefile
+	scripts/php.req scripts/php.prov
 	misc/Makefile 
 	doc/Makefile
 	python/Makefile
--- rpm-4.15.1/fileattrs/Makefile.am~	2019-06-26 23:17:31.000000000 +0900
+++ rpm-4.15.1/fileattrs/Makefile.am	2019-12-29 00:23:02.120085588 +0900
@@ -8,6 +8,6 @@
 fattrs_DATA = \
 	debuginfo.attr desktop.attr elf.attr font.attr libtool.attr metainfo.attr \
 	perl.attr perllib.attr pkgconfig.attr python.attr pythondist.attr ocaml.attr \
-	script.attr
+	script.attr php.attr
 
 EXTRA_DIST = $(fattrs_DATA)
--- /dev/null	2019-12-23 19:28:16.334291624 +0900
+++ rpm-4.15.1/fileattrs/php.attr	2019-12-29 00:22:52.989864428 +0900
@@ -0,0 +1,6 @@
+%__php_provides		%{_rpmconfigdir}/php.prov
+# define 'php_req_new' in ~/.rpmmacros to use php version of req finder
+%__php_requires		env PHP_MIN_VERSION=%{?php_min_version} %{_rpmconfigdir}/php.req%{?php_req_new:.php}
+%__php_magic		^PHP script.*
+%__php_path		\\.php$
+%__php_flags		magic_and_path
