diff -urN rpm-4.3.org/configure.ac rpm-4.3/configure.ac
--- rpm-4.3.org/configure.ac	2004-01-07 02:28:38.420706689 +0100
+++ rpm-4.3/configure.ac	2004-01-07 02:29:17.078719715 +0100
@@ -425,6 +425,18 @@
 AC_SUBST(WITH_BEECRYPT_INCLUDE)
 AC_SUBST(WITH_BEECRYPT_LIB)
 
+WITH_POPT_SUBDIR=
+WITH_POPT_LIB=
+AC_CHECK_HEADERS([popt.h])
+AC_CHECK_LIB(popt, poptGetContext, [
+       AC_DEFINE(HAVE_LIBPOPT, 1, [Define to 1 if you have libpopt.])
+       WITH_POPT_LIB="-lpopt"
+       ],[WITH_POPT_SUBDIR="popt"
+       WITH_POPT_LIB="\$(top_builddir)/popt/libpopt.la"])
+
+AC_SUBST(WITH_POPT_SUBDIR)
+AC_SUBST(WITH_POPT_LIB)
+
 AC_CHECK_FUNC(setreuid, [], [
     AC_CHECK_LIB(ucb, setreuid, [if echo $LIBS | grep -- -lucb >/dev/null ;then :; else LIBS="$LIBS -lc -lucb" USEUCB=y;fi])
 ])
@@ -1193,7 +1205,7 @@
 dnl # XXX Propagate -lucb to popt ...
 dnl export LIBS INCPATH CONFIG_SITE
 
-AC_CONFIG_SUBDIRS(popt beecrypt zlib elfutils file db3)
+AC_CONFIG_SUBDIRS(file)
 
 AC_CONFIG_FILES([ Doxyfile Makefile rpmrc macros platform rpmpopt rpm.spec
 	scripts/perl.req scripts/perl.prov
@@ -1216,7 +1228,7 @@
 	python/rpmdb/Makefile
 	python/rpmdb/test/Makefile
   ])
-AC_CONFIG_COMMANDS([default],[[	echo timestamp > popt/stamp-h.in
+AC_CONFIG_COMMANDS([default],[[	
 	echo timestamp > stamp-h.in
   
 ]],[[]])
diff -urN rpm-4.3.org/lib/Makefile.am rpm-4.3/lib/Makefile.am
--- rpm-4.3.org/lib/Makefile.am	2004-01-07 02:27:38.909001150 +0100
+++ rpm-4.3/lib/Makefile.am	2004-01-07 02:29:44.348085397 +0100
@@ -40,7 +40,7 @@
 librpm_la_LDFLAGS = -release 4.3 $(LDFLAGS) \
 	$(top_builddir)/rpmdb/librpmdb.la \
 	$(top_builddir)/rpmio/librpmio.la \
-	$(top_builddir)/popt/libpopt.la \
+	-lpopt \
 	@WITH_SELINUX_LIB@
 
 getdate.c: getdate.y
diff -urN rpm-4.3.org/Makefile.am rpm-4.3/Makefile.am
--- rpm-4.3.org/Makefile.am	2004-01-07 02:27:37.031389024 +0100
+++ rpm-4.3/Makefile.am	2004-01-07 02:29:17.082718889 +0100
@@ -14,7 +14,7 @@
 	po/*.in po/*.po po/rpm.pot \
 	rpm.magic rpmpopt-$(VERSION) rpmqv.c rpm.c
 
-SUBDIRS = intl po @WITH_ZLIB_SUBDIR@ @WITH_ELFUTILS_SUBDIR@ file @WITH_DB_SUBDIR@ popt @WITH_BEECRYPT_SUBDIR@  rpmio rpmdb lib build misc @WITH_PYTHON_SUBDIR@ tools scripts tests doc .
+SUBDIRS = intl po @WITH_ZLIB_SUBDIR@ @WITH_ELFUTILS_SUBDIR@ file @WITH_DB_SUBDIR@ @WITH_BEECRYPT_SUBDIR@  rpmio rpmdb lib build misc @WITH_PYTHON_SUBDIR@ tools scripts tests doc .
 
 INCLUDES = \
 	-I$(top_srcdir)/build \
@@ -22,7 +22,6 @@
 	-I$(top_srcdir)/rpmdb \
 	-I$(top_srcdir)/rpmio \
 	@WITH_BEECRYPT_INCLUDE@ \
-	-I$(top_srcdir)/popt \
 	@WITH_LIBELF_INCLUDE@ \
 	@INCPATH@
 
@@ -34,7 +33,7 @@
 	$(top_builddir)/lib/librpm.la \
 	$(top_builddir)/rpmdb/librpmdb.la \
 	$(top_builddir)/rpmio/librpmio.la \
-	$(top_builddir)/popt/libpopt.la \
+	-lpopt \
 	@WITH_ZLIB_LIB@ \
 	@INTLLIBS@ \
 	@LIBMISC@
@@ -107,7 +106,6 @@
 	    -load lib/rpmlib.lcd \
 	    -load rpmdb/rpmdb.lcd \
 	    -load rpmio/rpmio.lcd \
-	    -load popt/popt.lcd \
 		$(DEFS) $(INCLUDES) rpmqv.c $(rpmb_SOURCES)
 
 .PHONY:	lint
@@ -117,9 +115,7 @@
 		`make -s sources -C lib` \
 		`make -s sources -C rpmdb` \
 		`make -s sources -C rpmio` \
-		`make -s sources -C beecrypt` \
-		`make -s sources -C file/src` \
-		`make -s sources -C popt`
+		`make -s sources -C file/src`
 
 CVSTAG = r$(subst .,-,$(VERSION))
 
diff -urN rpm-4.3.org/rpmdb/Makefile.am rpm-4.3/rpmdb/Makefile.am
--- rpm-4.3.org/rpmdb/Makefile.am	2004-01-07 02:28:11.926180317 +0100
+++ rpm-4.3/rpmdb/Makefile.am	2004-01-07 02:29:17.087717856 +0100
@@ -9,7 +9,6 @@
 	-I$(top_srcdir)/lib \
 	-I$(top_srcdir)/rpmio \
 	@WITH_BEECRYPT_INCLUDE@ \
-	-I$(top_srcdir)/popt \
 	@WITH_LIBELF_INCLUDE@ \
 	@INCPATH@
 
@@ -51,7 +50,7 @@
 	tagname.c tagtbl.c
 librpmdb_la_LDFLAGS = -release 4.3 \
 	$(top_builddir)/rpmio/librpmio.la \
-	$(top_builddir)/popt/libpopt.la \
+	@WITH_POPT_LIB@ \
 	@WITH_LIBELF_LIB@
 librpmdb_la_LIBADD = $(DBLIBOBJS) $(libdb_la)
 librpmdb_la_DEPENDENCIES = $(DBLIBOBJS) $(libdb_la)
diff -urN rpm-4.3.org/rpmio/Makefile.am rpm-4.3/rpmio/Makefile.am
--- rpm-4.3.org/rpmio/Makefile.am	2004-01-07 02:27:43.915966813 +0100
+++ rpm-4.3/rpmio/Makefile.am	2004-01-07 02:29:17.090717237 +0100
@@ -11,7 +11,6 @@
 INCLUDES = -I. \
 	-I$(top_srcdir) \
 	@WITH_BEECRYPT_INCLUDE@ \
-	-I$(top_srcdir)/popt \
 	@INCPATH@
 
 pkgincdir = $(pkgincludedir)
@@ -65,34 +64,34 @@
 	$(LINT) $(DEFS) $(INCLUDES) $(librpmio_la_SOURCES)
 
 tdigest_SOURCES = tdigest.c
-tdigest_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la -lgcrypt
+tdigest_LDADD = librpmio.la -lpopt -lgcrypt
 
 tdir_SOURCES = tdir.c
 tdir_LDFLAGS = -all-static
-tdir_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la
+tdir_LDADD = librpmio.la -lpopt
 
 tfts_SOURCES = tfts.c
 tfts_LDFLAGS = -all-static
-tfts_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la
+tfts_LDADD = librpmio.la -lpopt
 
 tglob_SOURCES = tglob.c
 tglob_LDFLAGS = -all-static
-tglob_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la
+tglob_LDADD = librpmio.la -lpopt
 
 tinv_SOURCES = tinv.c
 tinv_LDFLAGS = -all-static
-tinv_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la
+tinv_LDADD = librpmio.la -lpopt
 
 tkey_SOURCES = tkey.c
 tkey_LDFLAGS = -all-static
-tkey_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la
+tkey_LDADD = librpmio.la -lpopt
 
 tring_SOURCES = tring.c
 tring_LDFLAGS = -all-static
-tring_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la
+tring_LDADD = librpmio.la -lpopt
 
 trpmio_SOURCES = trpmio.c
-trpmio_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la
+trpmio_LDADD = librpmio.la -lpopt
 
 tsw_SOURCES = tsw.c
 tsw_LDFLAGS = librpmio.la
diff -urN rpm-4.3.org/tools/Makefile.am rpm-4.3/tools/Makefile.am
--- rpm-4.3.org/tools/Makefile.am	2004-01-07 02:27:44.830777831 +0100
+++ rpm-4.3/tools/Makefile.am	2004-01-07 02:29:17.096715997 +0100
@@ -9,7 +9,6 @@
 	-I$(top_srcdir)/rpmdb \
 	-I$(top_srcdir)/rpmio \
 	@WITH_BEECRYPT_INCLUDE@ \
-	-I$(top_srcdir)/popt \
 	@WITH_LIBELF_INCLUDE@ \
 	@WITH_LIBDWARF_INCLUDE@ \
 	-I$(top_srcdir)/file/src \
@@ -38,7 +37,7 @@
 
 debugedit_SOURCES =	debugedit.c hashtab.c
 debugedit_LDADD =	@LDFLAGS_STATIC@ @WITH_LIBELF_LIB@ \
-	$(top_builddir)/popt/libpopt.la
+	-lpopt
 
 javadeps_SOURCES =	javadeps.c
 
