diff -ur rpm.org/macros.in rpm/macros.in
--- rpm.org/macros.in	2007-10-02 20:31:56.905775000 +0200
+++ rpm/macros.in	2007-10-02 20:35:30.122401828 +0200
@@ -195,15 +195,21 @@
 %debug_package \
 %ifnarch noarch\
 %global __debug_package 1\
-%package debug\
+%package debuginfo\
 Summary: Debug information for package %{name}\
+Summary(pl.UTF-8): Informacje dla debuggera dla pakietu %{name}\
 Group: Development/Debug\
+Requires: filesystem-debuginfo >= 3.0-16\
 AutoReqProv: 0\
-%description debug\
+%description debuginfo\
 This package provides debug information for package %{name}.\
 Debug information is useful when developing applications that use this\
 package or when debugging this package.\
-%files debug -f debugfiles.list\
+%description debuginfo -l pl_PL.UTF-8\
+Ten pakiet dostarcza informacje dla debuggera dla pakietu %{name}.\
+Informacje te są przydatne przy rozwijaniu aplikacji używających tego\
+pakietu oraz przy odpluskwianiu samego pakietu.\
+%files debuginfo -f debugfiles.list\
 %defattr(-,root,root)\
 %endif\
 %{nil}
@@ -1123,7 +1129,7 @@
 %{?buildroot:%{__rm} -rf '%{buildroot}'}
 %__spec_install_body	%{___build_body}
 %__spec_install_post\
-%{?__debug_package:%{__debug_install_post}}\
+%{expand:%%define __sip_%{?_enable_debug_packages} 1}%{?__sip_1:%{?__debug_package:%{__debug_install_post}}}%{expand:%%undefine __sip_%{?_enable_debug_packages}}\
 %{__arch_install_post}\
 %{__os_install_post}\
 %{nil}
--- rpm.org/scripts/find-debuginfo.sh	2007-09-06 09:36:38.000000000 +0200
+++ rpm/scripts/find-debuginfo.sh	2007-10-02 20:38:57.745519759 +0200
@@ -92,7 +92,8 @@
   $strip_g && case "$(file -bi "$2")" in
   application/x-sharedlib,*) g=-g ;;
   esac
-  eu-strip --remove-comment $g -f "$1" "$2" || exit
+  objcopy --only-keep-debug --remove-section .comment "$2" "$1" || exit
+  objcopy --add-gnu-debuglink="$1" "$2" || exit
 }
 
 # Make a relative symlink to $1 called $3$2
