--- rpm-4.3/lib/rpmfi.c.orig	2004-01-12 09:39:32.000000000 +0000
+++ rpm-4.3/lib/rpmfi.c	2004-01-12 10:34:20.000000000 +0000
@@ -13,6 +13,8 @@
 
 #include "rpmds.h"
 
+#include "legacy.h"
+
 #define	_RPMFI_INTERNAL
 #include "rpmfi.h"
 
--- rpm-4.3/lib/rpmrc.c.orig	2004-01-12 09:39:32.000000000 +0000
+++ rpm-4.3/lib/rpmrc.c	2004-01-12 10:54:42.000000000 +0000
@@ -14,6 +14,7 @@
 #define __power_pc() 0
 #endif
 
+#include "rpmio_internal.h"
 #include <rpmcli.h>
 #include <rpmmacro.h>
 #include <rpmlua.h>

