diff -Nru rpm-4.0/rpmpopt.in rpm-4.0.new/rpmpopt.in
--- rpm-4.0/rpmpopt.in	Wed Nov 15 19:18:35 2000
+++ rpm-4.0.new/rpmpopt.in	Wed Nov 15 19:18:15 2000
@@ -80,6 +80,21 @@
 # Build policies enabled from command line. Last policy applies.
 rpm alias --buildpolicy --define '__os_install_post @RPMCONFIGDIR@/brp-!#:+'
 
+# Make RPM build tree
+rpm exec --install-build-tree install-build-tree \
+	--POPTdesc=$"make all needed dirs for building binary rpms"
+
+# Compiling with debuginfo may be enabled by --debug
+rpm alias --debug --define 'debug 1' \
+	--POPTdesc=$"build packages with debug informations"
+rpmb alias --debug --define 'debug 1' \
+	--POPTdesc=$"build packages with debug informations"
+rpmbuild alias --debug --define 'debug 1' \
+	--POPTdesc=$"build packages with debug informations"
+
+# Conditional building 
+rpm exec  --bcond   find-spec-bcond \
+	--POPTdesc=$"find all --with/--without values"
 # Choose db interface:
 #	0	same as 1
 #	1	native db1 interface (e.g. linux glibc libdb1 routines).
--- rpm-4.1/po/pl.po.orig	Wed Oct 23 23:01:03 2002
+++ rpm-4.1/po/pl.po	Wed Oct 23 23:11:20 2002
@@ -3789,4 +3789,13 @@
 #: ../rpmpopt:410 ../rpmpopt:418
 msgid "DIRECTORY"
 msgstr "KATALOG"
+
+msgid "make all needed dirs for building binary rpms"
+msgstr "utwórz wszystkie katalogi potrzebne do budowania pakietów RPM"
+
+msgid "build packages with debug informations"
+msgstr "buduj pakiety z informacjami do odpluskwiania"
+
+msgid "find all --with/--without values"
+msgstr "znajd¼ wszystkie u¿ywane --with/--without"
 
