--- rpm-4.3/lib/psm.c.orig	2003-05-10 17:20:15.000000000 +0200
+++ rpm-4.3/lib/psm.c	2003-08-24 21:41:29.637316776 +0200
@@ -550,7 +550,10 @@
     xx = headerNVR(h, &n, &v, &r);
 
     /* XXX bash must have functional libtermcap.so.2 */
-    if (!strcmp(n, "libtermcap"))
+    /* if (!strcmp(n, "libtermcap"))
+     * -- always run ldconfig, these checks didn't work when few packages with
+     *    shared libs were installed just one after another in the same
+     *    transaction */
 	ldconfig_done = 0;
 
     /*
