diff -urN rpm-4.3.org/macros.in rpm-4.3/macros.in
--- rpm-4.3.org/macros.in	2004-05-20 22:09:48.582353920 +0200
+++ rpm-4.3/macros.in	2004-05-20 22:40:19.156064416 +0200
@@ -118,15 +118,20 @@
 %debug_package \
 %ifnarch noarch\
 %global __debug_package 1\
-%package debug\
+%package debuginfo\
 Summary: Debug information for package %{name}\
+Summary(pl): Informacje dla debuggera dla pakietu %{name}\
 Group: Development/Debug\
 AutoReqProv: 0\
-%description debug\
+%description debuginfo\
 This package provides debug information for package %{name}.\
 Debug information is useful when developing applications that use this\
 package or when debugging this package.\
-%files debug -f debugfiles.list\
+%description debuginfo -l pl\
+Ten pakiet dostarcza informacje dla debuggera dla pakietu %{name}.\
+Informacje te s± przydatne przy rozwijaniu aplikacji u¿ywaj±cych tego\
+pakietu oraz przy odpluskwianiu samego pakietu.\
+%files debuginfo -f debugfiles.list\
 %defattr(-,root,root)\
 %endif\
 %{nil}
@@ -1247,7 +1252,12 @@
 unset DISPLAY ||:\
 %{nil}
 
-%install %{?_enable_debug_packages:%{?buildsubdir:%{debug_package}}}\
+%install \
+%if %{?_enable_debug_packages:1}%{!?_enable_debug_packages:0}\
+%if %{_enable_debug_packages}\
+%{?buildsubdir:%{debug_package}}\
+%endif\
+%endif\
 %%install\
 LANG=C\
 export LANG\
@@ -1404,7 +1414,7 @@
     %{?verbose:set -x;} \
     if [ -d "$RPM_BUILD_ROOT" ]; then \
 	echo "Strip executable binaries, archives and shared object files."; \
-	filelist=`find $RPM_BUILD_ROOT -type f ! -regex ".*ld-[0-9.]*so.*" %{?_noautostrip:! -regex "%{_noautostrip}"}`; \
+	filelist=`find $RPM_BUILD_ROOT -type f ! -regex ".*ld-[0-9.]*so.*" ! -regex ".*/usr/lib[0-9]*/debug/.*\.debug" %{?_noautostrip:! -regex "%{_noautostrip}"}`; \
 	elfexelist=`echo $filelist | xargs -r file | \
 		awk '/ELF.*executable/ {print $1}' | cut -d: -f1`; \
 	elfsharedlist=`echo $filelist | xargs -r file | \
diff -urN rpm-4.3.org/scripts/find-debuginfo.sh rpm-4.3/scripts/find-debuginfo.sh
--- rpm-4.3.org/scripts/find-debuginfo.sh	2004-05-20 22:09:25.767822256 +0200
+++ rpm-4.3/scripts/find-debuginfo.sh	2004-05-20 23:12:43.367499352 +0200
@@ -26,12 +26,14 @@
 
 	mkdir -p "${debugdn}"
 	echo extracting debug info from $f
-	/usr/lib/rpm/debugedit -b "$RPM_BUILD_DIR" -d /usr/src/debug -l "$SOURCEFILE" "$f"
+	/usr/bin/debugedit -b "$RPM_BUILD_DIR" -d /usr/src/debug -l "$SOURCEFILE" "$f"
 	if test -w "$f"; then
-		eu-strip -f "${debugfn}" "$f" || :
+		eu-strip -f "${debugfn}" -o "$f.tmp" "$f" || :
+		rm -f "$f.tmp"
 	else
 		chmod u+w "$f"
-		eu-strip -f "${debugfn}" "$f" || :
+		eu-strip -f "${debugfn}" -o "$f.tmp" "$f" || :
+		rm -f "$f.tmp"
 		chmod u-w "$f"
 	fi
 done
