--- rpm-5.4.18/lib/rpmgi.c~	2017-07-14 15:07:43.000000000 +0300
+++ rpm-5.4.18/lib/rpmgi.c	2017-07-18 20:07:54.541517378 +0300
@@ -462,7 +462,7 @@
 		pat = ae;
 		got++;
 	    }
-	    if (got) {
+	    if (got >= 0) {
 SPEW("\tav %p[%d]: \"%s\" -> %s ~= \"%s\"\n", gi->argv, (int)(av - gi->argv), *av, tagName((rpmTag)tag), pat);
 		got = rpmmiAddPattern(gi->mi, (rpmTag)tag, RPMMIRE_DEFAULT, pat);
 	    }
