diff -ur rpm-4.4.9.org/macros.in rpm-4.4.9/macros.in
--- rpm-4.4.9.org/macros.in	2008-07-24 08:52:35.205363000 +0200
+++ rpm-4.4.9/macros.in	2008-07-24 09:01:00.351355186 +0200
@@ -275,6 +275,9 @@
 #
 #%distribution
 
+# PLD Linux Release
+%pld_release    th
+
 #	Configurable distribution URL, same as DistURL: tag in a specfile.
 #	The URL will be used to supply reliable information to tools like
 #	rpmfind.
--- rpm/macros.in~	2008-07-29 15:33:40.289757059 +0200
+++ rpm/macros.in	2008-07-29 15:37:14.559735562 +0200
@@ -1635,7 +1635,8 @@
 # -feliminate-dwarf2-dups disabled until PR ld/3290 is fixed.
 
 %debugcflags	-O0 -g -Wall
-%optldflags	-Wl,--as-needed
+%optldflags	-Wl,--as-needed -Wl,-z,relro -Wl,-z,-combreloc
+%optcppflags	-D_FORTIFY_SOURCE=2
 
 # Warning: those macros are overwritten by macros.build,
 #          left here for compatibility
