diff -urN rpm-4.5/lib/rpmfc.c rpm-4.5.new/lib/rpmfc.c
--- rpm-4.5/lib/rpmfc.c	2008-11-09 15:05:53.000000000 +0100
+++ rpm-4.5.new/lib/rpmfc.c	2008-11-09 15:07:53.000000000 +0100
@@ -555,6 +555,7 @@
   { "Java ",			RPMFC_JAVA|RPMFC_INCLUDE },
 
   /* .NET executables and libraries. file(1) cannot differ it from native win32 executables unfortunatelly */
+  { "Mono/.Net assembly",	RPMFC_MONO|RPMFC_INCLUDE },
   { "PE executable",		RPMFC_MONO|RPMFC_INCLUDE },
   { "executable PE",		RPMFC_MONO|RPMFC_INCLUDE },
 
