--- rpm-2.92/macros.in.old	Sun Mar 14 01:34:58 1999
+++ rpm-2.92/macros.in	Sun Mar 28 17:18:20 1999
@@ -38,19 +38,19 @@
 #	These are the default values that will be overridden by any
 #	explicit values found in /usr/lib/rpm/rpmrc or /etc/rpmrc.
 #
-%_builddir		%{_topdir}/BUILD
-%_bzip2bin		@BZIP2BIN@
-%_dbpath		%{_var}/lib/rpm
-%_defaultdocdir		/usr/doc
-%_fixperms		@FIXPERMS@
-%_gzipbin		@GZIPBIN@
-%_rpmdir		%{_topdir}/RPMS
-%_rpmfilename		%{ARCH}/%{NAME}-%{VERSION}-%{RELEASE}.%{ARCH}.rpm
-%_sourcedir		%{_topdir}/SOURCES
-%_specdir		%{_topdir}/SPECS
-%_srcrpmdir		%{_topdir}/SRPMS
-%_tmppath		%{_var}/tmp
-%_topdir		%{_usrsrc}/redhat
+#%_builddir		%{_topdir}/BUILD
+#%_bzip2bin		@BZIP2BIN@
+#%_dbpath		%{_var}/lib/rpm
+#%_defaultdocdir		/usr/doc
+#%_fixperms		@FIXPERMS@
+#%_gzipbin		@GZIPBIN@
+#%_rpmdir		%{_topdir}/RPMS
+#%_rpmfilename		%{ARCH}/%{NAME}-%{VERSION}-%{RELEASE}.%{ARCH}.rpm
+#%_sourcedir		%{_topdir}/SOURCES
+#%_specdir		%{_topdir}/SPECS
+#%_srcrpmdir		%{_topdir}/SRPMS
+#%_tmppath		%{_var}/tmp
+#%_topdir		%{_usrsrc}/redhat
 
 #==============================================================================
 # ---- per-platform macros.
