diff -Nur --exclude *~ rpm-4.0.2/lib/header.h rpm-4.0.2.new/lib/header.h
--- rpm-4.0.2/lib/header.h	Wed Feb 14 16:19:41 2001
+++ rpm-4.0.2.new/lib/header.h	Fri May 18 00:08:13 2001
@@ -66,7 +66,7 @@
 /* WARNING: 1 means success, 0 means failure (yes, this is backwards) */
 
 #include <stdio.h>
-#include <rpmio.h>
+#include "rpmio.h"
 
 #ifdef __cplusplus
 extern "C" {
@@ -702,7 +702,7 @@
 {
     if (data) {
 	/*@-branchstate@*/
-	if (type == -1 ||
+	if ((int)type == -1 ||
 	    type == RPM_STRING_ARRAY_TYPE ||
 	    type == RPM_I18NSTRING_TYPE ||
 	    type == RPM_BIN_TYPE)
@@ -713,7 +713,7 @@
 }
 
 #if !defined(__HEADER_PROTOTYPES__)
-#include <hdrinline.h>
+#include "hdrinline.h"
 #endif
 
 #ifdef __cplusplus
