--- rpm-5.2.0/macros.in.wiget	2010-04-24 18:05:03.413900239 +0200
+++ rpm-5.2.0/macros.in	2010-04-24 18:06:23.963891557 +0200
@@ -220,7 +220,7 @@ pakietu oraz przy odpluskwianiu samego p
 %endif\
 %{nil}
 
-%_defaultdocdir		%{_usr}/doc
+%_defaultdocdir		%{_usr}/share/doc
 
 #	The path to the pgp executable (legacy, use %{__pgp} instead).
 %_pgpbin		%{__pgp}
@@ -253,14 +253,14 @@ pakietu oraz przy odpluskwianiu samego p
 %_buildrootdir		@BUILDROOTDIR_MACRO@
 
 #       Build root path, where %install installs the package during build.
-%buildroot              %{_buildrootdir}/%{name}-root
+%buildroot		%{_tmppath}/%{name}-%{version}-root-%(id -u -n)
 
-#	Directory where temporaray files can be created.
-%_tmppath		%{_var}/tmp
+#	Directory where temporary files can be created.
+%_tmppath		%(echo "${TMPDIR:-/tmp}")
 %tmpdir			%{_tmppath}
 
 #	Path to top of build area.
-%_topdir		@PKGSRCDIR_MACRO@
+%_topdir		%(echo $HOME)/rpm
 
 #==============================================================================
 # ---- Optional macros.
@@ -367,8 +367,8 @@ pakietu oraz przy odpluskwianiu samego p
 #		"w6.lzdio"	lzma level 6 (legacy, stable).
 #		"w6.xzdio"	xz level 6 (obsoletes lzma, unstable).
 #
-#%_source_payload	w9.gzdio
-#%_binary_payload	w9.gzdio
+%_source_payload	w9.gzdio
+%_binary_payload	w9.lzdio
 
 #	Archive formats to use for source/binary package payloads.
 #		"cpio"		cpio archive (default)
@@ -1411,17 +1411,17 @@ print (t)\
 %_exec_prefix		%{_prefix}
 %_bindir		%{_exec_prefix}/bin
 %_sbindir		%{_exec_prefix}/sbin
-%_libexecdir		%{_exec_prefix}/libexec
+%_libexecdir		%{_exec_prefix}/lib
 %_datadir		%{_prefix}/share
-%_sysconfdir		%{_prefix}/etc
-%_sharedstatedir	%{_prefix}/com
-%_localstatedir		%{_prefix}/var
+%_sysconfdir		/etc
+%_sharedstatedir	/var/lib
+%_localstatedir		/var
 %_lib			lib
 %_libdir		%{_exec_prefix}/%{_lib}
 %_includedir		%{_prefix}/include
 %_oldincludedir		/usr/include
-%_infodir		%{_prefix}/info
-%_mandir		%{_prefix}/man
+%_infodir		%{_prefix}/share/info
+%_mandir		%{_prefix}/share/man
 %_localedir		%{_datadir}/locale
 
 %_initrddir		%{_sysconfdir}/rc.d/init.d
@@ -1755,7 +1755,33 @@ PreReq: python >= %{python_version}, pyt
 # helpers are also used by %{_rpmhome}/rpmdeps {--provides|--requires}.
 #%__executable_provides	%{_rpmhome}/executabledeps.sh --provides
 #%__executable_requires	%{_rpmhome}/executabledeps.sh --requires
-%__scriptlet_requires	%{__bash} --rpm-requires
+#%__scriptlet_requires	%{__bash} --rpm-requires
 
+# PLD rpm macros
+%_enable_debug_packages 1
+
+# provides don't obsolete
+%_upgrade_tag name
+
+#-----------------------------------------------------------------
+# CFLAGS and LDFLAGS used to build
+
+%debuginfocflags	%{expand:%%define __dic_%{?_enable_debug_packages} 1}%{?__dic_1: -gdwarf-2 -g2}%{expand:%%undefine __dic_%{?_enable_debug_packages}}
+# -feliminate-dwarf2-dups disabled until PR ld/3290 is fixed.
+
+%debugcflags	-O0 -g -Wall
+%optldflags	-Wl,--as-needed
+
+# Warning: those macros are overwritten by macros.build,
+#          left here for compatibility
+%rpmcflags	%{?debug:%debugcflags}%{!?debug:%optflags}%{?debuginfocflags}
+%rpmcxxflags	%{rpmcflags}
+%rpmldflags	%{!?no_build_with_as_needed:-Wl,--as-needed}
+
+# arch macros
+%ix86	i386 i486 i586 i686 pentium3 pentium4 athlon
+%x8664	x86_64 amd64 ia32e
+%ppc	ppc ppc7400 ppc7450
+  
 # \endverbatim
 #*/
