diff -Nur rpm-4.0.4.orig/build/files.c rpm-4.0.4/build/files.c
--- rpm-4.0.4.orig/build/files.c	Sun Jun 30 00:07:54 2002
+++ rpm-4.0.4/build/files.c	Sun Jun 30 00:13:08 2002
@@ -2412,7 +2412,8 @@
 
 /**
  */
-static int generateDepends(Spec spec, Package pkg, TFI_t cpioList, int multiLib)
+static int generateDepends(Spec spec, Package pkg, TFI_t cpioList, int multiLib,
+		StringBuf fileList, int *fileListLen)
 	/*@globals rpmGlobalMacroContext,
 		fileSystem, internalState @*/
 	/*@modifies cpioList, rpmGlobalMacroContext,
@@ -2450,6 +2451,10 @@
 	writeBytes += strlen(fi->dnl[fi->dil[i]]);
 	appendLineStringBuf(writeBuf, fi->bnl[i]);
 	writeBytes += strlen(fi->bnl[i]) + 1;
+	appendStringBuf(fileList, fi->dnl[fi->dil[i]]);
+	*fileListLen += strlen(fi->dnl[fi->dil[i]]);
+	appendLineStringBuf(fileList, fi->bnl[i]);
+	*fileListLen += strlen(fi->bnl[i]) + 1;
     }
 
     for (dm = depMsgs; dm->msg != NULL; dm++) {
@@ -2655,10 +2660,54 @@
     versions = hfd(versions, dvt);
 }
 
+int checkFiles(StringBuf fileList, int fileListLen)
+{
+    StringBuf readBuf = NULL;
+    char* myargv[3] = {NULL, NULL, NULL};
+    int rc = 0;
+    char *buf;
+    
+    myargv[0] = rpmExpand("%{__check_files}", NULL);
+    if (!(myargv[0] && *myargv[0] != '%')) {
+        rc = -1;
+        goto exit;
+    }    
+    myargv[1] = rpmExpand("%{buildroot}", NULL);
+    if (!(myargv[1] && *myargv[1] != '%')) {
+        rc = -1;
+        goto exit;
+    }
+    
+    rpmMessage(RPMMESS_NORMAL, _("Searching for omited files: (using %s)...\n"),
+               myargv[0]);
+		    
+    readBuf = getOutputFrom(NULL, myargv, getStringBuf(fileList), fileListLen, 0);
+    
+    if (readBuf) {
+        buf = getStringBuf(readBuf);
+        if ((*buf != '\0') & (*buf != '\n'))
+            rpmMessage(RPMMESS_WARNING, \
+	                _("found installed files not placed in any package:\n---\n%s---\n"),
+                       buf);
+    }
+    
+exit:
+    
+    freeStringBuf(readBuf);
+    _free(myargv[0]);
+    _free(myargv[1]);
+
+    return rc;
+}
+
 int processBinaryFiles(Spec spec, int installSpecialDoc, int test)
 {
     Package pkg;
     int res = 0;
+    StringBuf fileList;
+    int fileListLen = 0;
+    
+    fileList = newStringBuf();
     
     for (pkg = spec->packages; pkg != NULL; pkg = pkg->next) {
 	const char *n, *v, *r;
@@ -2678,14 +2727,24 @@
      * XXX existence (rather than value) that will need to change as well.
      */
 	if (headerIsEntry(pkg->header, RPMTAG_MULTILIBS)) {
-	    (void) generateDepends(spec, pkg, pkg->cpioList, 1);
-	    (void) generateDepends(spec, pkg, pkg->cpioList, 2);
+	    (void) generateDepends(spec, pkg, pkg->cpioList, 1, fileList, &fileListLen);
+	    (void) generateDepends(spec, pkg, pkg->cpioList, 2, fileList, &fileListLen);
 	} else
-	    (void) generateDepends(spec, pkg, pkg->cpioList, 0);
+	    (void) generateDepends(spec, pkg, pkg->cpioList, 0, fileList, &fileListLen);
 	/*@-noeffect@*/
 	printDeps(pkg->header);
 	/*@=noeffect@*/
     }
 
+    /* Now we have in fileList list of files from all packages.
+     * We pass it to a script which do the work of finding missing
+     * and duplicated files.
+     */
+    
+    if (res == 0) 
+        checkFiles(fileList, fileListLen);
+    
+    freeStringBuf(fileList);
+    
     return res;
 }
diff -Nur rpm-4.0.4.orig/platform.in rpm-4.0.4/platform.in
--- rpm-4.0.4.orig/platform.in	Sun Jun 30 00:07:54 2002
+++ rpm-4.0.4/platform.in	Sun Jun 30 00:13:45 2002
@@ -89,6 +89,10 @@
 #	Expanded at end of %install scriptlet.
 #
 
+# Gets file list on input and buildroot as first parameter
+# and returns list of files not found in any package
+%__check_files		@prefix@/lib/rpm/check-files
+
 %__arch_install_post   @ARCH_INSTALL_POST@
 
 %__os_install_post    \
