--- rpm-4.3/rpmio/rpmsw.c.orig	2003-04-10 20:09:13.000000000 +0200
+++ rpm-4.3/rpmio/rpmsw.c	2003-11-17 21:55:44.649426712 +0100
@@ -27,7 +27,7 @@
 /*@unchecked@*/
 static int rpmsw_initialized = 0;
 
-#if defined(__linux__) && defined(__i386__) && !defined(RPM_VENDOR_PLD)
+#if defined(__linux__) && defined(__i386__)
 /* Swiped from glibc-2.3.2 sysdeps/i386/i686/hp-timing.h */
 
 #define	HP_TIMING_ZERO(Var)	(Var) = (0)
