--- rpm-4.5/rpmio/rpmrpc.c~	2010-12-19 14:19:57.678043380 +0100
+++ rpm-4.5/rpmio/rpmrpc.c	2010-12-19 14:20:39.353812967 +0100
@@ -1711,6 +1711,17 @@
 if (_rpmio_debug)
 fprintf(stderr, "*** Glob(%s,0x%x,%p,%p)\n", pattern, (unsigned)flags, (void *)errfunc, pglob);
 /*@=castfcnptr@*/
+
+    /* same as upstream glob with difference that gl_stat is Lstat now */
+    pglob->gl_closedir = closedir;
+    pglob->gl_readdir = readdir;
+    pglob->gl_opendir = opendir;
+    pglob->gl_lstat = Lstat;
+    pglob->gl_stat = Lstat;
+
+/*@=type@*/
+	flags |= GLOB_ALTDIRFUNC;
+
     switch (ut) {
     case URL_IS_HTTPS:
     case URL_IS_HTTP:
--- rpm-4.5/configure.ac~	2010-12-19 13:46:37.917863585 +0100
+++ rpm-4.5/configure.ac	2010-12-19 13:50:49.826071048 +0100
@@ -1004,8 +1004,8 @@
 #fi
 #
 #if test "$rpm_cv_glob" = yes; then
-    AC_DEFINE(USE_GNU_GLOB, 1, [Use the included glob.c?])
-    AC_LIBOBJ(glob)
+#    AC_DEFINE(USE_GNU_GLOB, 1, [Use the included glob.c?])
+#    AC_LIBOBJ(glob)
     AC_LIBOBJ(fnmatch)
 #fi
 

