--- rpm-4.4.2/Doxyfile.in.orig	2005-07-04 21:00:46.000000000 +0200
+++ rpm-4.4.2/Doxyfile.in	2005-07-23 14:02:43.454384568 +0200
@@ -383,29 +383,7 @@
                          @top_srcdir@/rpmrc \
                          @top_srcdir@/rpm2cpio.c \
                          @top_srcdir@/system.h \
-                         @top_srcdir@/build/build.c \
-                         @top_srcdir@/build/buildio.h \
-                         @top_srcdir@/build/expression.c \
-                         @top_srcdir@/build/files.c \
-                         @top_srcdir@/build/misc.c \
-                         @top_srcdir@/build/names.c \
-                         @top_srcdir@/build/pack.c \
-                         @top_srcdir@/build/parseBuildInstallClean.c \
-                         @top_srcdir@/build/parseChangelog.c \
-                         @top_srcdir@/build/parseDescription.c \
-                         @top_srcdir@/build/parseFiles.c \
-                         @top_srcdir@/build/parsePreamble.c \
-                         @top_srcdir@/build/parsePrep.c \
-                         @top_srcdir@/build/parseReqs.c \
-                         @top_srcdir@/build/parseScript.c \
-                         @top_srcdir@/build/parseSpec.c \
-                         @top_srcdir@/build/poptBT.c \
-                         @top_srcdir@/build/reqprov.c \
-                         @top_srcdir@/build/rpmbuild.h \
-                         @top_srcdir@/build/rpmfc.c \
-                         @top_srcdir@/build/rpmfc.h \
-                         @top_srcdir@/build/rpmspec.h \
-                         @top_srcdir@/build/spec.c \
+                         @top_srcdir@/build/ \
                          @top_srcdir@/doc/manual/builddependencies \
                          @top_srcdir@/doc/manual/buildroot \
                          @top_srcdir@/doc/manual/conditionalbuilds \
@@ -420,195 +398,13 @@
                          @top_srcdir@/doc/manual/spec \
                          @top_srcdir@/doc/manual/triggers \
                          @top_srcdir@/doc/manual/tsort \
-                         @top_srcdir@/file/src/apprentice.c \
-                         @top_srcdir@/file/src/ascmagic.c \
-                         @top_srcdir@/file/src/compress.c \
-                         @top_srcdir@/file/src/file.c \
-                         @top_srcdir@/file/src/file.h \
-                         @top_srcdir@/file/src/fsmagic.c \
-                         @top_srcdir@/file/src/names.h \
-                         @top_srcdir@/file/src/print.c \
-                         @top_srcdir@/file/src/readelf.c \
-                         @top_srcdir@/file/src/readelf.h \
-                         @top_srcdir@/file/src/softmagic.c \
-                         @top_srcdir@/file/src/tar.h \
-                         @top_srcdir@/lib/cpio.c \
-                         @top_srcdir@/lib/cpio.h \
-                         @top_srcdir@/lib/depends.c \
-                         @top_srcdir@/lib/formats.c \
-                         @top_srcdir@/lib/fs.c \
-                         @top_srcdir@/lib/fsm.c \
-                         @top_srcdir@/lib/fsm.h \
-                         @top_srcdir@/lib/manifest.c \
-                         @top_srcdir@/lib/manifest.h \
-                         @top_srcdir@/lib/misc.c \
-                         @top_srcdir@/lib/misc.h \
-                         @top_srcdir@/lib/package.c \
-                         @top_srcdir@/lib/poptALL.c \
-                         @top_srcdir@/lib/poptI.c \
-                         @top_srcdir@/lib/poptQV.c \
-                         @top_srcdir@/lib/psm.c \
-                         @top_srcdir@/lib/psm.h \
-                         @top_srcdir@/lib/query.c \
-                         @top_srcdir@/lib/rpmal.c \
-                         @top_srcdir@/lib/rpmal.h \
-                         @top_srcdir@/lib/rpmchecksig.c \
-                         @top_srcdir@/lib/rpmcli.h \
-                         @top_srcdir@/lib/rpmds.c \
-                         @top_srcdir@/lib/rpmds.h \
-                         @top_srcdir@/lib/rpmfi.c \
-                         @top_srcdir@/lib/rpmfi.h \
-                         @top_srcdir@/lib/rpmgi.c \
-                         @top_srcdir@/lib/rpmgi.h \
-                         @top_srcdir@/lib/rpminstall.c \
-                         @top_srcdir@/lib/rpmlead.c \
-                         @top_srcdir@/lib/rpmlead.h \
-                         @top_srcdir@/lib/rpmlib.h \
-                         @top_srcdir@/lib/rpmlibprov.c \
-                         @top_srcdir@/lib/rpmps.c \
-                         @top_srcdir@/lib/rpmps.h \
-                         @top_srcdir@/lib/rpmrc.c \
-                         @top_srcdir@/lib/rpmsx.c \
-                         @top_srcdir@/lib/rpmsx.h \
-                         @top_srcdir@/lib/rpmte.c \
-                         @top_srcdir@/lib/rpmte.h \
-                         @top_srcdir@/lib/rpmts.c \
-                         @top_srcdir@/lib/rpmts.h \
-                         @top_srcdir@/lib/rpmvercmp.c \
-                         @top_srcdir@/lib/signature.c \
-                         @top_srcdir@/lib/signature.h \
-                         @top_srcdir@/lib/stringbuf.c \
-                         @top_srcdir@/lib/stringbuf.h \
-                         @top_srcdir@/lib/transaction.c \
-                         @top_srcdir@/lib/verify.c \
-			 @top_srcdir@/lua/lapi.c \
-			 @top_srcdir@/lua/lapi.h \
-			 @top_srcdir@/lua/lcode.c \
-			 @top_srcdir@/lua/lcode.h \
-			 @top_srcdir@/lua/ldebug.c \
-			 @top_srcdir@/lua/ldebug.h \
-			 @top_srcdir@/lua/ldo.c \
-			 @top_srcdir@/lua/ldo.h \
-			 @top_srcdir@/lua/ldump.c \
-			 @top_srcdir@/lua/lfunc.c \
-			 @top_srcdir@/lua/lfunc.h \
-			 @top_srcdir@/lua/lgc.c \
-			 @top_srcdir@/lua/lgc.h \
-			 @top_srcdir@/lua/llex.c \
-			 @top_srcdir@/lua/llex.h \
-			 @top_srcdir@/lua/llimits.h \
-			 @top_srcdir@/lua/lmem.c \
-			 @top_srcdir@/lua/lmem.h \
-			 @top_srcdir@/lua/lobject.c \
-			 @top_srcdir@/lua/lobject.h \
-			 @top_srcdir@/lua/lopcodes.c \
-			 @top_srcdir@/lua/lopcodes.h \
-			 @top_srcdir@/lua/lparser.c \
-			 @top_srcdir@/lua/lparser.h \
-			 @top_srcdir@/lua/lstate.c \
-			 @top_srcdir@/lua/lstate.h \
-			 @top_srcdir@/lua/lstring.c \
-			 @top_srcdir@/lua/lstring.h \
-			 @top_srcdir@/lua/ltable.c \
-			 @top_srcdir@/lua/ltable.h \
-			 @top_srcdir@/lua/ltests.c \
-			 @top_srcdir@/lua/ltm.c \
-			 @top_srcdir@/lua/ltm.h \
-			 @top_srcdir@/lua/lundump.c \
-			 @top_srcdir@/lua/lundump.h \
-			 @top_srcdir@/lua/lvm.c \
-			 @top_srcdir@/lua/lvm.h \
-			 @top_srcdir@/lua/lzio.c \
-			 @top_srcdir@/lua/lzio.h \
-                         @top_srcdir@/rpmdb/db3.c \
-                         @top_srcdir@/rpmdb/dbconfig.c \
-                         @top_srcdir@/rpmdb/fprint.c \
-                         @top_srcdir@/rpmdb/fprint.h \
-                         @top_srcdir@/rpmdb/hdrNVR.c \
-                         @top_srcdir@/rpmdb/hdrinline.h \
-                         @top_srcdir@/rpmdb/header.c \
-                         @top_srcdir@/rpmdb/header.h \
-                         @top_srcdir@/rpmdb/header_internal.c \
-                         @top_srcdir@/rpmdb/header_internal.h \
-                         @top_srcdir@/rpmdb/legacy.c \
-                         @top_srcdir@/rpmdb/legacy.h \
-                         @top_srcdir@/rpmdb/merge.c \
-                         @top_srcdir@/rpmdb/poptDB.c \
-                         @top_srcdir@/rpmdb/rpmhash.c \
-                         @top_srcdir@/rpmdb/rpmhash.h \
-                         @top_srcdir@/rpmdb/rpmdb.c \
-                         @top_srcdir@/rpmdb/rpmdb.h \
-                         @top_srcdir@/rpmdb/tagname.c \
-                         @top_srcdir@/rpmdb/tagtbl.c \
-                         @top_srcdir@/rpmio/argv.c \
-                         @top_srcdir@/rpmio/argv.h \
-                         @top_srcdir@/rpmio/digest.c \
-                         @top_srcdir@/rpmio/fts.c \
-                         @top_srcdir@/rpmio/fts.h \
-                         @top_srcdir@/rpmio/macro.c \
-                         @top_srcdir@/rpmio/rpmlog.c \
-                         @top_srcdir@/rpmio/rpmlog.h \
-                         @top_srcdir@/rpmio/rpmerr.h \
-                         @top_srcdir@/rpmio/rpmio.c \
-                         @top_srcdir@/rpmio/rpmio.h \
-                         @top_srcdir@/rpmio/rpmio_internal.h \
-                         @top_srcdir@/rpmio/rpmmacro.h \
-                         @top_srcdir@/rpmio/rpmmalloc.c \
-                         @top_srcdir@/rpmio/rpmmessages.h \
-                         @top_srcdir@/rpmio/rpmpgp.c \
-                         @top_srcdir@/rpmio/rpmpgp.h \
-                         @top_srcdir@/rpmio/rpmrpc.c \
-                         @top_srcdir@/rpmio/rpmsq.c \
-                         @top_srcdir@/rpmio/rpmsq.h \
-                         @top_srcdir@/rpmio/rpmsw.c \
-                         @top_srcdir@/rpmio/rpmsw.h \
-                         @top_srcdir@/rpmio/rpmurl.h \
-                         @top_srcdir@/rpmio/strcasecmp.c \
-                         @top_srcdir@/rpmio/stubs.c \
-                         @top_srcdir@/rpmio/ugid.c \
-                         @top_srcdir@/rpmio/ugid.h \
-                         @top_srcdir@/rpmio/url.c \
-                         @top_srcdir@/popt/findme.c \
-                         @top_srcdir@/popt/findme.h \
-                         @top_srcdir@/popt/popt.c \
-                         @top_srcdir@/popt/popt.h \
-                         @top_srcdir@/popt/poptconfig.c \
-                         @top_srcdir@/popt/popthelp.c \
-                         @top_srcdir@/popt/poptint.h \
-                         @top_srcdir@/popt/poptparse.c \
-                         @top_srcdir@/python/header-py.c \
-                         @top_srcdir@/python/header-py.h \
-                         @top_srcdir@/python/rpmal-py.c \
-                         @top_srcdir@/python/rpmal-py.h \
-                         @top_srcdir@/python/rpmdb-py.c \
-                         @top_srcdir@/python/rpmdb-py.h \
-                         @top_srcdir@/python/rpmds-py.c \
-                         @top_srcdir@/python/rpmds-py.h \
-                         @top_srcdir@/python/rpmfd-py.c \
-                         @top_srcdir@/python/rpmfd-py.h \
-                         @top_srcdir@/python/rpmfi-py.c \
-                         @top_srcdir@/python/rpmfi-py.h \
-                         @top_srcdir@/python/rpmfts-py.c \
-                         @top_srcdir@/python/rpmfts-py.h \
-                         @top_srcdir@/python/rpmmi-py.c \
-                         @top_srcdir@/python/rpmmi-py.h \
-                         @top_srcdir@/python/rpmmodule.c \
-                         @top_srcdir@/python/rpmrc-py.c \
-                         @top_srcdir@/python/rpmrc-py.h \
-                         @top_srcdir@/python/rpmte-py.c \
-                         @top_srcdir@/python/rpmte-py.h \
-                         @top_srcdir@/python/rpmts-py.c \
-                         @top_srcdir@/python/rpmts-py.h \
-                         @top_srcdir@/tools/dump.c \
-                         @top_srcdir@/tools/dumpdb.c \
-                         @top_srcdir@/tools/javadeps.c \
-                         @top_srcdir@/tools/rpmcache.c \
-                         @top_srcdir@/tools/rpmdeps.c \
-                         @top_srcdir@/tools/rpmgraph.c \
-                         @top_srcdir@/tools/rpminject.c \
-                         @top_srcdir@/tools/rpmtool.c \
-                         @top_srcdir@/tools/rpmxml.c \
-                         @top_srcdir@/tools/rpmxp.c
+                         @top_srcdir@/file/src/ \
+                         @top_srcdir@/lib/ \
+			 @top_srcdir@/lua/ \
+                         @top_srcdir@/rpmdb/ \
+                         @top_srcdir@/rpmio/ \
+                         @top_srcdir@/python/ \
+                         @top_srcdir@/tools/
 
 # If the value of the INPUT tag contains directories, you can use the 
 # FILE_PATTERNS tag to specify one or more wildcard pattern (like *.cpp 
@@ -617,8 +413,8 @@
 # *.c *.cc *.cxx *.cpp *.c++ *.java *.ii *.ixx *.ipp *.i++ *.inl *.h *.hh *.hxx *.hpp 
 # *.h++ *.idl *.odl *.cs *.php *.php3 *.inc
 
-FILE_PATTERNS          = *.c \
-                         *.h
+#FILE_PATTERNS          = *.c \
+#                         *.h
 
 # The RECURSIVE tag can be used to turn specify whether or not subdirectories 
 # should be searched for input files as well. Possible values are YES and NO. 
