--- rpm-4.4.7/lua/Makefile.am.orig	2007-03-06 08:22:58.000000000 +0100
+++ rpm-4.4.7/lua/Makefile.am	2007-03-09 10:46:58.000000000 +0100
@@ -17,7 +17,7 @@
 
 lua_lua_SOURCES = lua/lua.c
 lua_lua_CFLAGS = -DLUA_USERCONFIG='"$(srcdir)/local/userconfig.c"'
-lua_lua_LDADD = $(LDADD) -L/usr/lib -lncurses
+lua_lua_LDADD = $(LDADD) -lncurses
 luac_luac_SOURCES = luac/luac.c luac/print.c lopcodes.c
 luac_luac_CFLAGS = -DLUA_OPNAMES
 
