diff -urN rpm-4.0.2.org/acconfig.h rpm-4.0.2/acconfig.h
--- rpm-4.0.2.org/acconfig.h	Fri Oct 12 18:23:02 2001
+++ rpm-4.0.2/acconfig.h	Fri Oct 12 18:50:55 2001
@@ -29,6 +29,9 @@
 /* Define as 1 if you have catgets and don't want to use GNU gettext.  */
 #undef HAVE_CATGETS
 
+/* */
+#undef HAVE_GETMNTENT
+
 /* Define as 1 if you have gettext and don't want to use GNU gettext.  */
 #undef HAVE_GETTEXT
 
diff -urN rpm-4.0.2.org/configure.in rpm-4.0.2/configure.in
--- rpm-4.0.2.org/configure.in	Fri Oct 12 18:23:08 2001
+++ rpm-4.0.2/configure.in	Fri Oct 12 18:25:18 2001
@@ -20,7 +20,6 @@
 LIBOBJS=
 
 dnl Checks for programs.
-AC_ARG_PROGRAM
 AC_PROG_CC
 if test "$ac_cv_prog_gcc" = yes; then
     CFLAGS="$CFLAGS -D_GNU_SOURCE -Wall -Wpointer-arith -Wstrict-prototypes -Wmissing-prototypes -Wno-char-subscripts"
