--- rpm-4.3/python/Makefile.am.wiget	2003-05-06 17:44:12.000000000 +0200
+++ rpm-4.3/python/Makefile.am	2003-06-10 13:22:13.859888700 +0200
@@ -29,45 +29,35 @@
 	rpmfi-py.h rpmmi-py.h rpmrc-py.h rpmte-py.h rpmts-py.h
 
 mylibs= \
-	$(top_builddir)/lib/.libs/librpm.so \
-	$(top_builddir)/rpmdb/.libs/librpmdb.so \
-	$(top_builddir)/rpmio/.libs/librpmio.so \
-	$(top_builddir)/popt/.libs/libpopt.so \
+	$(top_builddir)/lib/librpm.la \
+ 	$(top_builddir)/rpmdb/librpmdb.la \
+ 	$(top_builddir)/rpmio/librpmio.la \
+	-lbeecrypt \
+ 	-lpopt \
 	@WITH_LIBELF_LIB@
 
 LDADD =
 
 pythondir = $(pylibdir)/site-packages
-python_PROGRAMS = rpmmodule.so
+python_LTLIBRARIES = rpmmodule.la poptmodule.la
 
 rpmdbdir = $(pylibdir)/site-packages/rpmdb
-rpmdb_PROGRAMS = _rpmdb.so
+rpmdb_LTLIBRARIES = _rpmdb.la
 
-noinst_PROGRAMS = poptmodule.so
+_rpmdb_la_SOURCES = _rpmdb.c
+_rpmdb_la_LIBADD = $(mylibs) $(LIBS)
+_rpmdb_la_LDFLAGS = -module -avoid-version
 
-rpmmodule_so_SOURCES =
-rpmmodule_so_LDFLAGS = $(mylibs) $(LIBS) -shared -Wl,-soname,rpmmodule.so
+poptmodule_la_SOURCES = poptmodule.c
+poptmodule_la_LIBADD = $(mylibs) $(LIBS)
+poptmodule_la_LDFLAGS = -module -avoid-version
 
-_rpmdb_so_SOURCES = _rpmdb.c
-_rpmdb_so_LDFLAGS = $(mylibs) $(LIBS) -shared -Wl,-soname,_rpmdb.so
-
-poptmodule_so_SOURCES = poptmodule.c
-poptmodule_so_LDFLAGS = $(mylibs) $(LIBS) -shared -Wl,-soname,poptmodule.so
-
-noinst_LTLIBRARIES = librpmmodule.la
-librpmmodule_la_SOURCES = rpmmodule.c header-py.c \
+rpmmodule_la_SOURCES = rpmmodule.c header-py.c \
 	rpmbc-py.c rpmmpw-py.c rpmrng-py.c \
 	rpmal-py.c rpmds-py.c rpmdb-py.c rpmfd-py.c rpmfts-py.c \
 	rpmfi-py.c rpmmi-py.c rpmrc-py.c rpmte-py.c rpmts-py.c
-
-rpmmodule.so$(EXEEXT): $(librpmmodule_la_OBJECTS)
-	$(CC) -o $@ $(librpmmodule_la_OBJECTS) $(rpmmodule_so_LDFLAGS)
-
-_rpmdb.so$(EXEEXT): _rpmdb.lo
-	$(CC) -o $@ _rpmdb.lo $(_rpmdb_so_LDFLAGS)
-
-poptmodule.so$(EXEEXT): poptmodule.lo
-	$(CC) -o $@ poptmodule.lo $(poptmodule_so_LDFLAGS)
+rpmmodule_la_LIBADD = $(mylibs) $(LIBS)
+rpmmodule_la_LDFLAGS = -module -avoid-version 
 
 # rpmmodule.c header-py.c \
 # 	rpmal-py.c rpmds-py.c rpmdb-py.c rpmfd-py.c rpmfi-py.c rpmmi-py.c \
