--- rpm-4.4.8/lib/rpmfc.c.orig	2007-02-14 07:31:50.000000000 +0100
+++ rpm-4.4.8/lib/rpmfc.c	2007-04-08 16:48:00.273560592 +0200
@@ -773,7 +773,7 @@
     } else
     if (fc->fcolor->vals[fc->ix] & RPMFC_PHP) {
 	xx = rpmfcHelper(fc, 'P', "php");
-	if (is_executable)
+	/* not only executable, files run by httpd usually are not */
 	    xx = rpmfcHelper(fc, 'R', "php");
     }
 
@@ -1056,7 +1056,7 @@
 	    else if (_suffix(s, ".pc"))
 		ftype = "pkgconfig file";
 
-	    /* XXX all files with extension ".php" are pkgconfig for now. */
+	    /* XXX all files with extension ".php" are php scripts now. */
 	    else if (_suffix(s, ".php"))
 		ftype = "PHP script text";
 
