--- rpm-4.3/python/rpmdebug-py.c.orig	2005-01-04 23:12:21.975043528 +0100
+++ rpm-4.3/python/rpmdebug-py.c	2005-01-04 23:13:42.899741104 +0100
@@ -2,7 +2,9 @@
 /*@unchecked@*/
 extern PyTypeObject PyCode_Type;
 /*@unchecked@*/
-extern PyTypeObject PyDictIter_Type;
+extern PyTypeObject PyDictIterKey_Type;
+extern PyTypeObject PyDictIterValue_Type;
+extern PyTypeObject PyDictIterItem_Type;
 /*@unchecked@*/
 extern PyTypeObject PyFrame_Type;
 
@@ -45,7 +47,9 @@
     if (o->ob_type == &PyCode_Type)	return "Code";
     if (o->ob_type == &PyComplex_Type)	return "Complex";
     if (o->ob_type == &PyDict_Type)	return "Dict";
-    if (o->ob_type == &PyDictIter_Type)	return "DictIter";
+    if (o->ob_type == &PyDictIterKey_Type)	return "DictIterKey";
+    if (o->ob_type == &PyDictIterValue_Type)	return "DictIterValue";
+    if (o->ob_type == &PyDictIterItem_Type)	return "DictIterItem";
     if (o->ob_type == &PyFile_Type)	return "File";
     if (o->ob_type == &PyFloat_Type)	return "Float";
     if (o->ob_type == &PyFrame_Type)	return "Frame";
--- rpm-4.3/python/rpmrc-py.c.orig	2005-01-04 23:12:26.489357248 +0100
+++ rpm-4.3/python/rpmrc-py.c	2005-01-04 23:15:03.677461024 +0100
@@ -214,8 +214,12 @@
 {
 if (_rc_debug)
 fprintf(stderr, "*** rpmrc_iter(%p[%s])\n", s, lbl(s));
-    if (s->ob_type == &PyDictIter_Type)
-	return PyDictIter_Type.tp_iter(s);
+    if (s->ob_type == &PyDictIterKey_Type)
+	return PyDictIterKey_Type.tp_iter(s);
+    if (s->ob_type == &PyDictIterValue_Type)
+	return PyDictIterValue_Type.tp_iter(s);
+    if (s->ob_type == &PyDictIterItem_Type)
+	return PyDictIterItem_Type.tp_iter(s);
     return PyDict_Type.tp_iter(s);
 }
 
@@ -227,8 +231,12 @@
 {
 if (_rc_debug)
 fprintf(stderr, "*** rpmrc_iternext(%p[%s])\n", s, lbl(s));
-    if (s->ob_type == &PyDictIter_Type)
-	return PyDictIter_Type.tp_iternext(s);
+    if (s->ob_type == &PyDictIterKey_Type)
+	return PyDictIterKey_Type.tp_iternext(s);
+    if (s->ob_type == &PyDictIterValue_Type)
+	return PyDictIterValue_Type.tp_iternext(s);
+    if (s->ob_type == &PyDictIterItem_Type)
+	return PyDictIterItem_Type.tp_iternext(s);
     return NULL;
 }
 
@@ -240,8 +248,12 @@
 {
 if (_rc_debug)
 fprintf(stderr, "*** rpmrc_next(%p[%s],%p)\n", s, lbl(s), args);
-    if (s->ob_type == &PyDictIter_Type)
-	return PyDictIter_Type.tp_methods[0].ml_meth(s, args);
+    if (s->ob_type == &PyDictIterKey_Type)
+	return PyDictIterKey_Type.tp_methods[0].ml_meth(s, args);
+    if (s->ob_type == &PyDictIterValue_Type)
+	return PyDictIterValue_Type.tp_methods[0].ml_meth(s, args);
+    if (s->ob_type == &PyDictIterItem_Type)
+	return PyDictIterItem_Type.tp_methods[0].ml_meth(s, args);
     return NULL;
 }
 
