--- ./macros.in~	2005-10-30 16:24:30.000000000 +0200
+++ ./macros.in	2005-10-30 16:24:32.000000000 +0200
@@ -223,7 +223,7 @@
 %_tmppath		%{_var}/tmp
 
 #	Path to top of build area.
-%_topdir		%{_usrsrc}/redhat
+%_topdir         %(echo $HOME)/rpm
 
 #	The path to the unzip executable (legacy, use %{__unzip} instead).
 %_unzipbin		%{__unzip}
@@ -1276,5 +1276,18 @@
 #
 %find_lang	@RPMCONFIGDIR@/find-lang.sh %{buildroot}
 
+# PLD rpm macros
+%_enable_debug_packages 0
+
+#-----------------------------------------------------------------
+# CFLAGS and LDFLAGS used to build
+
+%debuginfocflags	%{expand:%%define __dic_%{?_enable_debug_packages} 1}%{?__dic_1: -ggdb}%{expand:%%undefine __dic_%{?_enable_debug_packages}}
+
+%debugcflags	-O0 -g -Wall
+%rpmcflags	%{?debug:%debugcflags}%{!?debug:%optflags}%{?debuginfocflags}
+%rpmcxxflags	%{?debug:%debugcflags}%{!?debug:%optflags}%{?debuginfocflags}
+%rpmldflags	%{nil}
+
 # \endverbatim
 #*/
