--- rpm-4.3/macros.in.alreadey_patched	2004-05-31 16:47:32.496632272 +0200
+++ rpm-4.3/macros.in	2004-05-31 16:46:14.976417136 +0200
@@ -107,12 +107,16 @@
 #	The location of the rpm database file(s) after "rpm --rebuilddb".
 %_dbpath_rebuild	%{_dbpath}
 
-#
-#	Path to script that creates debug symbols in a /usr/lib/debug
-#	shadow tree.
-%__debug_install_post   \
-   @RPMCONFIGDIR@/find-debuginfo.sh %{_builddir}/%{?buildsubdir}\
-%{nil}
+#------------------------------------------------------------------
+# Path to script that creates debug symbols in a /usr/lib/debug
+# shadow tree.
+#%no_debug_install_post 1
+%__debug_install_post { \
+%{!?no_debug_install_post: \
+	/usr/lib/rpm/find-debuginfo.sh %{_builddir}/%{?buildsubdir}\
+	%{nil}\
+} }
+#------------------------------------------------------------------
 
 #	Template for debug information sub-package.
 %debug_package \
