--- rpm/lib/depends.c	2007-02-13 20:39:00.386623000 +0100
+++ rpm/lib/depends.c	2007-02-19 19:48:40.490623000 +0100
@@ -507,6 +507,10 @@
 	goto unsatisfied;
     }
 
+    /* Ignore cpuinfo() and uname() deps */
+    if (!strncmp(Name, "cpuinfo(", sizeof("cpuinfo(")-1) || !strncmp(Name, "uname(", sizeof("uname(")-1))
+	    goto exit;
+
     /* Search added packages for the dependency. */
     if (rpmalSatisfiesDepend(ts->addedPackages, dep, NULL) != NULL) {
 	/*
