--- rpm-4.4.3/build/parseChangelog.c.orig	2005-11-12 01:20:12.000000000 +0100
+++ rpm-4.4.3/build/parseChangelog.c	2005-11-18 19:46:50.357322048 +0100
@@ -255,7 +255,7 @@
 	line = xstrtolocale(line);
 	appendStringBuf(sb, spec->line);
 	line = _free(line);
-	if ((rc = readLine(spec, STRIP_COMMENTS)) > 0) {
+	if ((rc = readLine(spec, STRIP_COMMENTS | STRIP_NOEXPAND)) > 0) {
 	    nextPart = PART_NONE;
 	    break;
 	}
--- rpm-4.3/build/parseSpec.c	Fri May 16 00:08:57 2003
+++ rpm-5.0.1/build/parseSpec.c	2008-01-31 00:43:18.388524335 +0200
@@ -168,7 +168,7 @@
 
 /**
  */
-static int copyNextLineFromOFI(Spec spec, OFI_t * ofi)
+static int copyNextLineFromOFI(Spec spec, OFI_t * ofi, int strip)
 	/*@globals rpmGlobalMacroContext, h_errno,
 		fileSystem @*/
 	/*@modifies spec->nextline, spec->nextpeekc, spec->lbuf, spec->line,
@@ -230,12 +230,16 @@
 /*@=mods@*/
 
 	/* Don't expand macros (eg. %define) in false branch of %if clause */
+	/* Also don't expand macros in %changelog where we set STRIP_NOEXPAND flag */
+	/* (first line is ommited, so if there is e.g. %date macro, it will be expanded */
+	if (!(strip & STRIP_NOEXPAND)) {
 	if (spec->readStack->reading &&
 	    expandMacros(spec, spec->macros, spec->lbuf, spec->lbuf_len)) {
 		rpmlog(RPMLOG_ERR, _("line %d: %s\n"),
 			spec->lineNum, spec->lbuf);
 		return RPMRC_FAIL;
 	}
+	}
 	spec->nextline = spec->lbuf;
     }
     return 0;
@@ -273,6 +277,7 @@
     SKIPSPACE(s);
 
     match = -1;
+    if (! (strip & STRIP_NOEXPAND)) {
     if (!spec->readStack->reading && !strncmp("%if", s, sizeof("%if")-1)) {
 	match = 0;
     } else if (! strncmp("%ifarch", s, sizeof("%ifarch")-1)) {
@@ -346,7 +351,7 @@
 
       /* Copy next file line into the spec line buffer */
 
-      if ((rc = copyNextLineFromOFI(spec, ofi)) != 0) {
+      if ((rc = copyNextLineFromOFI(spec, ofi, strip)) != 0) {
 	if (rc == RPMRC_FAIL)
 	    goto retry;
 	return rc;
@@ -354,6 +359,7 @@
 	ofi = spec->fileStack;
 	goto retry;
     }
+    }
 
     if (match != -1) {
 	rl = xmalloc(sizeof(*rl));
--- rpm-4.3/build/rpmbuild.h.wiget	Sat May 10 17:19:33 2003
+++ rpm-4.3/build/rpmbuild.h	Fri May 16 00:06:47 2003
@@ -75,6 +75,7 @@
 #define STRIP_NOTHING             0
 #define STRIP_TRAILINGSPACE (1 << 0)
 #define STRIP_COMMENTS      (1 << 1)
+#define STRIP_NOEXPAND      (1 << 2)
 
 #ifdef __cplusplus
 extern "C" {
