--- rpm-4.3.orig/build/parsePreamble.c	2003-11-16 13:47:23.000000000 +0100
+++ rpm-4.3/build/parsePreamble.c	2004-02-29 15:40:58.466804704 +0100
@@ -942,6 +943,12 @@
 	headerCopyTags(spec->packages->header, pkg->header,
 			(uint32_t *)copyTagsDuringParse);
 
+    if (headerGetEntry(pkg->header, RPMTAG_EPOCH, NULL, NULL, NULL) == 0) {
+    	int num = 0;
+	headerAddEntry(pkg->header, RPMTAG_EPOCH, RPM_UINT32_TYPE, &num, 1);
+	addMacro(spec->macros, "epoch", NULL, "0", RMIL_SPEC);
+    }
+    
     if (checkForRequired(pkg->header, NVR) != RPMRC_OK)
 	return RPMRC_FAIL;
 
