diff -durN rpm-4.3.orig/python/Makefile.am rpm-4.3/python/Makefile.am
--- rpm-4.3.orig/python/Makefile.am	2004-12-21 11:12:21.000000000 +0100
+++ rpm-4.3/python/Makefile.am	2004-12-21 11:19:51.246590360 +0100
@@ -6,7 +6,7 @@
 
 PYVER= @WITH_PYTHON_VERSION@
 
-pylibdir = $(prefix)/lib/python${PYVER}
+pylibdir = $(libdir)/python${PYVER}
 pyincdir = $(prefix)/include/python${PYVER}
 
 SUBDIRS = rpmdb mpw
diff -durN rpm-4.3.orig/python/rpmdb/Makefile.am rpm-4.3/python/rpmdb/Makefile.am
--- rpm-4.3.orig/python/rpmdb/Makefile.am	2004-12-21 11:12:21.000000000 +0100
+++ rpm-4.3/python/rpmdb/Makefile.am	2004-12-21 11:20:09.519812408 +0100
@@ -9,7 +9,7 @@
 EXTRA_DIST = \
 	__init__.py dbobj.py db.py dbrecio.py dbshelve.py dbtables.py dbutils.py
 
-pylibdir = $(prefix)/lib/python${PYVER}
+pylibdir = $(libdir)/python${PYVER}
 
 rpmdbdir = $(pylibdir)/site-packages/rpmdb
 rpmdb_SCRIPTS = \
