diff -Nru rpm-4.0/rpmrc.in rpm-4.0.new/rpmrc.in
--- rpm-4.0/rpmrc.in	Wed Sep 13 20:03:46 2000
+++ rpm-4.0.new/rpmrc.in	Wed Nov 15 19:11:00 2000
@@ -9,7 +9,7 @@
 # Values for RPM_OPT_FLAGS for various platforms
 
 # XXX this is the i386 optflags that rpm has traditionally used in the past.
-#optflags: i386 -O2 -m486 -fno-strength-reduce
+#optflags: i386 -O2 -march=i386
 optflags: i386 -O2 -march=i386 -mcpu=i686
 optflags: i486 -O2 -march=i486
 optflags: i586 -O2 -march=i586
@@ -291,5 +291,5 @@
 buildarch_compat: ia64: noarch
 buildarch_compat: s390: noarch
 
-macrofiles:	@RPMCONFIGDIR@/macros:@RPMCONFIGDIR@/%{_target}/macros:@SYSCONFIGDIR@/macros.specspo:@SYSCONFIGDIR@/macros:@SYSCONFIGDIR@/%{_target}/macros:~/.rpmmacros
+macrofiles:	@RPMCONFIGDIR@/macros:@RPMCONFIGDIR@/macros.pld:@RPMCONFIGDIR@/%{_target}/macros:@SYSCONFIGDIR@/macros.specspo:@SYSCONFIGDIR@/macros:@SYSCONFIGDIR@/%{_target}/macros:~/.rpmmacros
 
