--- rpm-4.0.4/rpmdb/dbconfig.c.wiget	Sun Jan 20 17:48:23 2002
+++ rpm-4.0.4/rpmdb/dbconfig.c	Wed Feb 13 22:45:40 2002
@@ -15,7 +15,7 @@
 /*@=redef@*/
 #endif
 
-#include <db3/db.h>
+#include DB_H
 
 #include <rpmlib.h>
 #include <rpmmacro.h>
--- rpm-4.0.4/rpmdb/db3.c.wiget	Sun Jan 20 18:09:56 2002
+++ rpm-4.0.4/rpmdb/db3.c	Wed Feb 13 22:45:57 2002
@@ -23,7 +23,7 @@
 /*@=redef@*/
 #endif
 
-#include <db3/db.h>
+#include DB_H
 
 #include <rpmlib.h>
 #include <rpmmacro.h>
--- rpm-4.0.4/system.h.wiget	Wed Feb 13 22:46:12 2002
+++ rpm-4.0.4/system.h	Wed Feb 13 22:48:21 2002
@@ -581,4 +581,13 @@
 #ifndef MOUNTED
 #define MOUNTED "/etc/mnttab"
 #endif
+
+#ifdef HAVE_DB4_DB_H
+#define DB_H <db4/db.h>
+#elif defined(HAVE_DB3_DB_H)
+#define DB_H <db3/db.h>
+#else
+#define DB_H <db.h>
+#endif
+
 #endif	/* H_SYSTEM */
--- rpm-4.0.4/configure.in.wiget	Wed Feb 13 22:48:32 2002
+++ rpm-4.0.4/configure.in	Wed Feb 13 22:53:31 2002
@@ -393,14 +393,16 @@
 if test $withval = no ; then
 dnl ------------------ without internal db
 
-AC_CHECK_HEADERS(db3/db.h)
+AC_CHECK_HEADERS(db4/db.h db3/db.h)
 
 dnl Check for Berkeley db3 API.
 AC_CHECK_FUNC(db_create, [DBLIBSRCS="$DBLIBSRCS db3.c"],
-  AC_CHECK_LIB(db-3.2, db_create, [DBLIBSRCS="$DBLIBSRCS db3.c"; libdb3="-ldb-3.2"],
-    AC_CHECK_LIB(db-3.1, db_create, [DBLIBSRCS="$DBLIBSRCS db3.c"; libdb3="-ldb-3.1"],
-      AC_CHECK_LIB(db-3.0, db_create, [DBLIBSRCS="$DBLIBSRCS db3.c"; libdb3="-ldb-3.0"],
-        AC_CHECK_LIB(db, db_create, [DBLIBSRCS="$DBLIBSRCS db3.c"; libdb3="-ldb"],
+  AC_CHECK_LIB(db-4.0, db_create, [DBLIBSRCS="$DBLIBSRCS db3.c"; libdb3="-ldb-4.0"],
+    AC_CHECK_LIB(db-3.2, db_create, [DBLIBSRCS="$DBLIBSRCS db3.c"; libdb3="-ldb-3.2"],
+      AC_CHECK_LIB(db-3.1, db_create, [DBLIBSRCS="$DBLIBSRCS db3.c"; libdb3="-ldb-3.1"],
+        AC_CHECK_LIB(db-3.0, db_create, [DBLIBSRCS="$DBLIBSRCS db3.c"; libdb3="-ldb-3.0"],
+          AC_CHECK_LIB(db, db_create, [DBLIBSRCS="$DBLIBSRCS db3.c"; libdb3="-ldb"],
+          ,$libthread)
         ,$libthread)
       ,$libthread)
     ,$libthread)
