diff -ru rpm-3.0.2.old/rpmrc.in rpm-3.0.2/rpmrc.in
--- rpm-3.0.2.old/rpmrc.in	Thu Sep  9 22:08:24 1999
+++ rpm-3.0.2/rpmrc.in	Thu Sep  9 22:11:31 1999
@@ -118,12 +118,12 @@
 buildarchtranslate: osfmach3_i486: i386
 buildarchtranslate: osfmach3_i386: i386
 
-buildarchtranslate: i986: i386
-buildarchtranslate: i886: i386
-buildarchtranslate: i786: i386
-buildarchtranslate: i686: i386
-buildarchtranslate: i586: i386
-buildarchtranslate: i486: i386
+buildarchtranslate: i986: i686
+buildarchtranslate: i886: i686
+buildarchtranslate: i786: i686
+buildarchtranslate: i686: i686
+buildarchtranslate: i586: i586
+buildarchtranslate: i486: i486
 buildarchtranslate: i386: i386
 
 buildarchtranslate: osfmach3_ppc: ppc
@@ -133,7 +133,8 @@
 buildarchtranslate: sun4c: sparc
 buildarchtranslate: sun4d: sparc
 buildarchtranslate: sun4m: sparc
-buildarchtranslate: sun4u: sparc64
+buildarchtranslate: sun4u: sparc
+buildarchtranslate: sparc64: sparc
 
 buildarchtranslate: atarist: m68kmint
 buildarchtranslate: atariste: m68kmint
