--- rpm-4.3/build/files.c.orig	2003-11-24 19:10:54.000000000 +0100
+++ rpm-4.3/build/files.c	2003-11-24 19:20:05.827568008 +0100
@@ -2119,7 +2119,7 @@
 	goto exit;
 
     /* Verify that file attributes scope over hardlinks correctly. */
-    if (checkHardLinks(&fl))
+    if (checkHardLinks(&fl) && !rpmExpandNumeric("%{_hack_dontneed_PartialHardlinkSets}"))
 	(void) rpmlibNeedsFeature(pkg->header,
 			"PartialHardlinkSets", "4.0.4-1");
 
