--- rpm-4.4.9/macros.in	2008-03-27 19:45:51.090959034 +0200
+++ rpm-4.4.9/macros.in	2008-04-04 23:09:48.869762337 +0300
@@ -271,6 +271,9 @@
 #
 #%distribution
 
+# PLD Linux Release
+%pld_release	ti
+
 #	Configurable distribution URL, same as DistURL: tag in a specfile.
 #	The URL will be used to supply reliable information to tools like
 #	rpmfind.
@@ -1464,6 +1467,8 @@
 # -feliminate-dwarf2-dups disabled until PR ld/3290 is fixed.
 
 %debugcflags	-O0 -g -Wall
+%optldflags	-Wl,--as-needed
+%optcppflags	%{nil}
 
 # Warning: those macros are overwritten by macros.build,
 #          left here for compatibility
