--- rpm-4.4.2/rpmio/Makefile.am.orig	2005-10-07 13:52:53.000000000 +0200
+++ rpm-4.4.2/rpmio/Makefile.am	2005-11-08 15:56:58.000000000 +0100
@@ -26,8 +26,6 @@
 
 BEECRYPTLOBJS = $(shell test X"@WITH_BEECRYPT_SUBDIR@" != X && cat $(top_builddir)/@WITH_BEECTYPT_SUBDIR@/listobjs)
 
-LDFLAGS = -L$(RPM_BUILD_ROOT)$(usrlibdir) -L$(DESTDIR)$(usrlibdir)
-
 usrlibdir = $(libdir)@MARK64@
 usrlib_LTLIBRARIES = librpmio.la
 librpmio_la_SOURCES = \
--- rpm-4.4.2/rpmdb/Makefile.am.orig	2005-10-07 13:52:53.000000000 +0200
+++ rpm-4.4.2/rpmdb/Makefile.am	2005-11-08 15:56:49.000000000 +0100
@@ -37,9 +37,6 @@
 # XXX watchout, ../db3/libdb.la created by this Makefile may surprise
 libdb_la = 
 
-# XXX grrr, RPM_BUILD_ROOT prevents build pollution if/when -lrpm different
-LDFLAGS = -L$(RPM_BUILD_ROOT)$(usrlibdir) -L$(DESTDIR)$(usrlibdir)
-
 usrlibdir = $(libdir)@MARK64@
 usrlib_LTLIBRARIES = librpmdb.la
 librpmdb_la_SOURCES = \
--- rpm-4.4.2/lib/Makefile.am.orig	2005-10-07 13:52:53.000000000 +0200
+++ rpm-4.4.2/lib/Makefile.am	2005-11-08 15:56:22.000000000 +0100
@@ -29,8 +29,6 @@
 mylibs = librpm.la
 LIBS =
 
-LDFLAGS = -L$(RPM_BUILD_ROOT)$(usrlibdir) -L$(DESTDIR)$(usrlibdir)
-
 usrlibdir = $(libdir)@MARK64@
 usrlib_LTLIBRARIES = librpm.la
 librpm_la_SOURCES = \
--- rpm-4.4.2/build/Makefile.am.orig	2005-03-14 11:03:48.000000000 +0100
+++ rpm-4.4.2/build/Makefile.am	2005-11-08 15:56:12.000000000 +0100
@@ -22,8 +22,6 @@
 pkginc_HEADERS = rpmbuild.h rpmfc.h rpmfile.h rpmspec.h
 noinst_HEADERS = buildio.h
 
-LDFLAGS = -L$(RPM_BUILD_ROOT)$(usrlibdir) -L$(DESTDIR)$(usrlibdir)
-
 usrlibdir = $(libdir)@MARK64@
 usrlib_LTLIBRARIES = librpmbuild.la
 librpmbuild_la_SOURCES = \
