--- rpm-4.5/lib/psm.c.org	2008-11-21 17:20:34.293584455 +0100
+++ rpm-4.5/lib/psm.c	2008-11-21 17:21:41.482728047 +0100
@@ -2114,8 +2114,8 @@
 	    psm->countCorrection = -1;
 
 	    if (!(rpmtsFlags(ts) & RPMTRANS_FLAG_NOPOSTUN)) {
-		rc = rpmpsmNext(psm, PSM_SCRIPT);
-		if (rc) break;
+		if (rpmpsmNext(psm, PSM_SCRIPT))
+		    rpmMessage(RPMMESS_VERBOSE, _("Ignoring failed %%postun scriptlet\n"));
 	    }
 
 	    if (!(rpmtsFlags(ts) & RPMTRANS_FLAG_NOTRIGGERPOSTUN)) {
