diff -urbB rpm-4.4.1.org/build/parseSpec.c rpm-4.4.1/build/parseSpec.c
--- rpm-4.4.1.org/build/parseSpec.c	2005-06-18 13:56:00.311780000 +0200
+++ rpm-4.4.1/build/parseSpec.c	2005-06-18 13:56:29.453350760 +0200
@@ -233,6 +233,15 @@
     return 0;
 }
 /*@=boundswrite@*/
+static const char *getAlternateArch(const char *arch)
+{
+    const char *alternate_arch = NULL;
+    if (! strncmp("x86_64", arch, sizeof("x86_64")-1))
+      alternate_arch = "amd64";
+    else if (! strncmp("amd64", arch, sizeof("amd64")-1))
+      alternate_arch = "x86_64";
+    return alternate_arch;
+}
 
 /*@-boundswrite@*/
 int readLine(Spec spec, int strip)
@@ -325,13 +334,15 @@
 	match = 0;
     } else if (! strncmp("%ifarch", s, sizeof("%ifarch")-1)) {
 	const char *arch = rpmExpand("%{_target_cpu}", NULL);
+	const char *alternate_arch = getAlternateArch(arch);
 	s += 7;
-	match = matchTok(arch, s);
+	match = matchTok(arch, s) || (alternate_arch && matchTok(alternate_arch, s));
 	arch = _free(arch);
     } else if (! strncmp("%ifnarch", s, sizeof("%ifnarch")-1)) {
 	const char *arch = rpmExpand("%{_target_cpu}", NULL);
+	const char *alternate_arch = getAlternateArch(arch);
 	s += 8;
-	match = !matchTok(arch, s);
+	match = !matchTok(arch, s) && (!alternate_arch || !matchTok(alternate_arch, s));
 	arch = _free(arch);
     } else if (! strncmp("%ifos", s, sizeof("%ifos")-1)) {
 	const char *os = rpmExpand("%{_target_os}", NULL);
diff -urbB rpm-4.4.1.org/installplatform rpm-4.4.1/installplatform
--- rpm-4.4.1.org/installplatform	2005-01-26 04:39:54.000000000 +0100
+++ rpm-4.4.1/installplatform	2005-06-18 13:57:16.874141704 +0200
@@ -31,6 +31,7 @@
 
 case "$arch" in
   i[3456]86|pentium[34]|athlon) SUBSTS='s_i386_i386_ s_i386_i486_ s_i386_i586_ s_i386_i686_ s_i386_pentium3_ s_i386_pentium4_ s_i386_athlon_' ;;
+  x86_64*|amd64*) SUBSTS="s/${arch}/x86_64/ s/${arch}/amd64/" ;;
   alpha*) SUBSTS='s_alpha_alpha_ s_alpha_alphaev5_ s_alpha_alphaev56_ s_alpha_alphapca56_ s_alpha_alphaev6_ s_alpha_alphaev67_' ;;
   sparc*) SUBSTS='s_sparc\(64\|v9\)_sparc_ s_sparc64_sparcv9_;s_sparc\([^v]\|$\)_sparcv9\1_ s_sparcv9_sparc64_;s_sparc\([^6]\|$\)_sparc64\1_' ;;
   powerpc*|ppc*) SUBSTS='s_ppc64_ppc_ s_ppc\([^6ip]\|$\)_ppc64\1_ s_ppc\([^6ip]\|$\)_ppciseries_ s_ppc\([^6ip]\|$\)_ppcpseries_ s_ppc\([^6ip]\|$\)_ppc64iseries_ s_ppc\([^6ip]\|$\)_ppc64pseries_' ;;
--- rpm-4.4.2.2/lib/rpmrc.c~	2007-12-18 01:50:21.574997679 +0200
+++ rpm-4.4.2.2/lib/rpmrc.c	2007-12-18 01:58:40.593251514 +0200
@@ -1334,6 +1334,12 @@
 		un.machine[1] = class;
 	}
 #	endif
+#   if defined(__linux__) && defined(__amd64__)
+    {
+	    /* Defaults to amd64 if compiler supports __amd64__ */
+        strcpy(un.machine, "amd64");
+    }
+#   endif
 
 	/* the uname() result goes through the arch_canon table */
 	canon = lookupInCanonTable(un.machine,
diff -urbB rpm-4.4.1.org/rpmrc.in rpm-4.4.1/rpmrc.in
--- rpm-4.4.1.org/rpmrc.in	2005-06-18 13:56:00.305781000 +0200
+++ rpm-4.4.1/rpmrc.in	2005-06-18 13:56:29.455350456 +0200
@@ -230,9 +230,9 @@
 
 buildarchtranslate: ia64: ia64
 
-buildarchtranslate: x86_64: x86_64
-buildarchtranslate: amd64: x86_64
-buildarchtranslate: ia32e: x86_64
+buildarchtranslate: x86_64: amd64
+buildarchtranslate: amd64: amd64
+buildarchtranslate: ia32e: amd64
 
 #############################################################
 # Architecture compatibility
Tylko w rpm-4.4.1: rpmrc.in.orig
