diff -Nru rpm-4.0.1/rpmrc.in rpm-4.0.1.new/rpmrc.in
--- rpm-4.0.1/rpmrc.in	Wed Jan 31 13:19:02 2001
+++ rpm-4.0.1.new/rpmrc.in	Wed Jan 31 13:20:51 2001
@@ -9,7 +9,7 @@
 # Values for RPM_OPT_FLAGS for various platforms
 
 # XXX this is the i386 optflags that rpm has traditionally used in the past.
-#optflags: i386 -O2 -m486 -fno-strength-reduce
+#optflags: i386 -O2 -march=i386
 optflags: i386 -O2 -march=i386 -mcpu=i686
 optflags: i486 -O2 -march=i486
 optflags: i586 -O2 -march=i586
@@ -289,5 +289,5 @@
 buildarch_compat: ia64: noarch
 buildarch_compat: s390: noarch
 
-macrofiles:	@RPMCONFIGDIR@/macros:@RPMCONFIGDIR@/%{_target}/macros:@SYSCONFIGDIR@/macros.specspo:@SYSCONFIGDIR@/macros.db1:@SYSCONFIGDIR@/macros:@SYSCONFIGDIR@/%{_target}/macros:~/.rpmmacros
+macrofiles:	@RPMCONFIGDIR@/macros:@RPMCONFIGDIR@/macros.pld:@RPMCONFIGDIR@/%{_target}/macros:@SYSCONFIGDIR@/macros.specspo:@SYSCONFIGDIR@/macros.db1:@SYSCONFIGDIR@/macros:@SYSCONFIGDIR@/%{_target}/macros:~/.rpmmacros
 
