--- rpm-4.0/lib/db3.c.wiget	Fri Aug  4 21:48:35 2000
+++ rpm-4.0/lib/db3.c	Wed Nov 15 22:28:52 2000
@@ -9,7 +9,7 @@
 typedef	int int32_t;
 #endif
 
-#include <db3/db.h>
+#include <db.h>
 
 #include <rpmlib.h>
 #include <rpmmacro.h>
--- rpm-4.0/configure.in.wiget	Wed Nov 15 22:32:50 2000
+++ rpm-4.0/configure.in	Wed Nov 15 22:31:57 2000
@@ -396,7 +396,7 @@
 AC_HEADER_DIRENT
 AC_HEADER_TIME
 
-AC_CHECK_HEADERS(db_185.h db1/db.h db3/db.h)
+AC_CHECK_HEADERS(db_185.h db1/db.h db3/db.h db.h)
 AC_CHECK_HEADERS(fcntl.h getopt.h grp.h memory.h netdb.h pwd.h utime.h)
 
 AC_CHECK_HEADERS(sys/socket.h sys/select.h)
--- rpm-4.0.2/lib/dbconfig.c.wiget	Wed Feb 21 12:05:37 2001
+++ rpm-4.0.2/lib/dbconfig.c	Wed Feb 21 12:06:42 2001
@@ -4,7 +4,7 @@
 
 #include "system.h"
 
-#include <db3/db.h>
+#include <db.h>
 
 #include <rpmlib.h>
 #include <rpmmacro.h>
