diff -ur rpm-4.4.4/Makefile.am rpm-4.4.4.morearchs/Makefile.am
--- rpm-4.4.4/Makefile.am	2006-02-11 15:01:38.000000000 +0000
+++ rpm-4.4.4.morearchs/Makefile.am	2006-02-11 18:06:51.000000000 +0000
@@ -188,6 +188,8 @@
 	powerpc*) $(mkinstalldirs) $(DESTDIR)$(pkgsrcdir)/RPMS/ppc ;\
 		$(mkinstalldirs) $(DESTDIR)$(pkgsrcdir)/RPMS/ppciseries ;\
 		$(mkinstalldirs) $(DESTDIR)$(pkgsrcdir)/RPMS/ppcpseries ;\
+		$(mkinstalldirs) $(DESTDIR)$(pkgsrcdir)/RPMS/ppc7400 ;\
+		$(mkinstalldirs) $(DESTDIR)$(pkgsrcdir)/RPMS/ppc7450 ;\
 		$(mkinstalldirs) $(DESTDIR)$(pkgsrcdir)/RPMS/ppc64 ;\
 		$(mkinstalldirs) $(DESTDIR)$(pkgsrcdir)/RPMS/ppc64iseries ;\
 		$(mkinstalldirs) $(DESTDIR)$(pkgsrcdir)/RPMS/ppc64pseries ;;\
diff -ur rpm-4.4.4/installplatform rpm-4.4.4.morearchs/installplatform
--- rpm-4.4.4/installplatform	2005-10-25 21:19:24.000000000 +0000
+++ rpm-4.4.4.morearchs/installplatform	2006-02-11 18:02:55.000000000 +0000
@@ -33,7 +33,7 @@
   i[3456]86|pentium[34]|athlon) SUBSTS='s_i386_i386_ s_i386_i486_ s_i386_i586_ s_i386_i686_ s_i386_pentium3_ s_i386_pentium4_ s_i386_athlon_' ;;
   alpha*) SUBSTS='s_alpha_alpha_ s_alpha_alphaev5_ s_alpha_alphaev56_ s_alpha_alphapca56_ s_alpha_alphaev6_ s_alpha_alphaev67_' ;;
   sparc*) SUBSTS='s_sparc\(64\|v9\)_sparc_ s_sparc64_sparcv9_;s_sparc\([^v]\|$\)_sparcv9\1_ s_sparcv9_sparc64_;s_sparc\([^6]\|$\)_sparc64\1_' ;;
-  powerpc*|ppc*) SUBSTS='s_ppc64_ppc_ s_ppc\([^6ip]\|$\)_ppc64\1_ s_ppc\([^6ip]\|$\)_ppciseries_ s_ppc\([^6ip]\|$\)_ppcpseries_ s_ppc\([^6ip]\|$\)_ppc64iseries_ s_ppc\([^6ip]\|$\)_ppc64pseries_' ;;
+  powerpc*|ppc*) SUBSTS='s_ppc64_ppc_ s_ppc\([^6ip]\|$\)_ppc64\1_ s_ppc\([^6ip]\|$\)_ppciseries_ s_ppc\([^6ip]\|$\)_ppcpseries_ s_ppc\([^6ip]\|$\)_ppc7400_ s_ppc\([^6ip]\|$\)_ppc7450_ s_ppc\([^6ip]\|$\)_ppc64iseries_ s_ppc\([^6ip]\|$\)_ppc64pseries_' ;;
   s390*) SUBSTS='s_s390x_s390_ s_s390\([^x]\|$\)_s390x\1_' ;;
   *) SUBSTS=y___ ;;
 esac
diff -ur rpm-4.4.4/lib/rpmrc.c rpm-4.4.4.morearchs/lib/rpmrc.c
--- rpm-4.4.4/lib/rpmrc.c	2006-02-11 15:01:38.000000000 +0000
+++ rpm-4.4.4.morearchs/lib/rpmrc.c	2006-02-11 18:00:45.000000000 +0000
@@ -1328,7 +1328,12 @@
 
 	    if ( pvr ) {
 		pvr >>= 16;
-		if ( pvr >= 0x40)
+		if ( (pvr & ~0x3) == 0x8000)
+		    // 0x8000: 7450, 0x8001: 7455, 0x8002: 7457, 0x8003: 7447A
+		    strcpy(un.machine, "ppc7450");
+		else if ( pvr == 0x800C || pvr == 12 )
+		    strcpy(un.machine, "ppc7400\n");
+		else if ( pvr >= 0x40)
 		    strcpy(un.machine, "ppcpseries");
 		else if ( (pvr == 0x36) || (pvr == 0x37) )
 		    strcpy(un.machine, "ppciseries");
diff -ur rpm-4.4.4/macros.in rpm-4.4.4.morearchs/macros.in
--- rpm-4.4.4/macros.in	2006-02-11 15:01:38.000000000 +0000
+++ rpm-4.4.4.morearchs/macros.in	2006-02-11 18:17:36.000000000 +0000
@@ -1244,6 +1244,7 @@
 #   conditionals.
 #
 %ix86   i386 i486 i586 i686 pentium3 pentium4 athlon
+%ppc    ppc ppc7400 ppc7450
 
 #------------------------------------------------------------------------
 # Use in %install to generate locale specific file lists. For example,
diff -ur rpm-4.4.4/rpmrc.in rpm-4.4.4.morearchs/rpmrc.in
--- rpm-4.4.4/rpmrc.in	2006-02-11 15:01:38.000000000 +0000
+++ rpm-4.4.4.morearchs/rpmrc.in	2006-02-11 18:11:55.000000000 +0000
@@ -42,6 +42,8 @@
 optflags: m68k -O2 -fomit-frame-pointer%{!?nospecflags:%{?specflags: %{specflags}}%{?specflags_m68k: %{specflags_m68k}}}
 
 optflags: ppc -O2 -fsigned-char%{!?nospecflags:%{?specflags: %{specflags}}%{?specflags_ppc: %{specflags_ppc}}}
+optflags: ppc7400 -O2 -mcpu=7400 -fsigned-char%{!?nospecflags:%{?specflags: %{specflags}}%{?specflags_ppc: %{specflags_ppc}}}
+optflags: ppc7450 -O2 -mcpu=7450 -fsigned-char%{!?nospecflags:%{?specflags: %{specflags}}%{?specflags_ppc: %{specflags_ppc}}}
 optflags: ppc8260 -O2 -fsigned-char%{!?nospecflags:%{?specflags: %{specflags}}%{?specflags_ppc8260: %{specflags_ppc8260}}}
 optflags: ppc8560 -O2 -fsigned-char%{!?nospecflags:%{?specflags: %{specflags}}%{?specflags_ppc8560: %{specflags_ppc8560}}}
 optflags: ppc32dy4 -O2 -fsigned-char%{!?nospecflags:%{?specflags: %{specflags}}%{?specflags_ppc32dy4: %{specflags_ppc32dy4}}}
@@ -110,6 +112,8 @@
 arch_canon:	mips:	mips	4
 
 arch_canon:	ppc:	ppc	5
+arch_canon:	ppc7400:	ppc7400	5
+arch_canon:	ppc7450:	ppc7450	5
 arch_canon:	ppc8260:	ppc8260	5
 arch_canon:	ppc8560:	ppc8560	5
 arch_canon:	ppc32dy4:	ppc32dy4	5
@@ -212,7 +216,8 @@
 buildarchtranslate: sun4u: sparc64
 buildarchtranslate: sparc64: sparc64
 
-buildarchtranslate: osfmach3_ppc: ppc
+buildarchtranslate: ppc7400: ppc7400
+buildarchtranslate: ppc7450: ppc7450
 buildarchtranslate: powerpc: ppc
 buildarchtranslate: powerppc: ppc
 buildarchtranslate: ppc8260: ppc
@@ -263,6 +268,8 @@
 arch_compat: osfmach3_i486: i486 osfmach3_i386
 arch_compat: osfmach3_i386: i486
 
+arch_compat: ppc7450: ppc7400
+arch_compat: ppc7400: ppc
 arch_compat: osfmach3_ppc: ppc
 arch_compat: powerpc: ppc
 arch_compat: powerppc: ppc
@@ -375,6 +382,8 @@
 
 buildarch_compat: m68k: noarch
 
+buildarch_compat: ppc7450: ppc7400
+buildarch_compat: ppc7400: ppc
 buildarch_compat: ppc8260: noarch
 buildarch_compat: ppc8560: noarch
 buildarch_compat: ppc32dy4: noarch
