--- rpm-4.0.4/configure.in.orig	Sat Mar 16 20:30:55 2002
+++ rpm-4.0.4/configure.in	Sat Mar 16 20:32:13 2002
@@ -17,8 +17,6 @@
 dnl Set of available languages.
 ALL_LINGUAS="cs da de en_RN es eu_ES fi fr gl hu id is it ja ko no pl pt pt_BR ro ru sk sl sr sv tr uk wa zh zh_CN.GB2312"
 
-LIBOBJS=
-
 dnl Checks for programs.
 AC_PROG_CC
 AC_AIX
@@ -452,7 +450,7 @@
 done
 
 AC_CHECK_FUNC(fork, [], [echo "using vfork() instead of fork()";
-	LIBOBJS=fakefork.o])
+	AC_LIBOBJ([fakefork])])
 
 dnl AmigaOS and IXEmul have a fork() dummy
     case "$target" in
@@ -489,7 +487,7 @@
 AC_CHECK_FUNCS(personality)
 
 dnl XXX these are needed only by rpmgettext right now
-AM_FUNC_ERROR_AT_LINE
+AC_FUNC_ERROR_AT_LINE
 
 dnl ==> AM_GNU_GETTEXT does this (and more)
 dnl AC_CHECK_HEADERS(unistd.h limits.h locale.h malloc.h string.h sys/param.h)
@@ -726,7 +724,7 @@
 
 AC_FUNC_FNMATCH
 if test $ac_cv_func_fnmatch_works = no; then
-   LIBOBJS="$LIBOBJS fnmatch.o"
+   AC_LIBOBJ([fnmatch])
 dnl XXX no expansion needed
 dnl AC_DEFINE_UNQUOTED(fnmatch, rpl_fnmatch)
 fi
@@ -771,7 +769,7 @@
 dnl 
 dnl if test $rpm_cv_regex = yes; then
 dnl     AC_DEFINE(USE_GNU_REGEX)
-dnl     LIBOBJS="$LIBOBJS regex.o"
+dnl     AC_LIBOBJ([regex.o])
 dnl fi
 
 dnl
@@ -810,7 +808,8 @@
 
 if test "$rpm_cv_glob" = yes; then
     AC_DEFINE(USE_GNU_GLOB)
-    LIBOBJS="$LIBOBJS glob.o fnmatch.o"
+    AC_LIBOBJ([glob.o])
+    AC_LIBOBJ([fnmatch.o])
 fi
 dnl
 dnl Auto-detect which python bindings should be built.
@@ -899,7 +897,7 @@
       AC_CHECK_LIB(c_r, getmntinfo_r, [LIBS="$LIBS -lc_r"; 
 					AC_DEFINE(HAVE_GETMNTINFO_R)], [
 		 AC_DEFINE([USE_GETMNTENT], 1, [Defined if getmntent replacement is used])
-                 LIBOBJS="$LIBOBJS getmntent.o"])])])])
+                 AC_LIBOBJ([getmntent])])])])])
 
 AC_CHECK_FUNC(lchown,
    [__CHOWN_RHF="%{__chown} -Rhf"
@@ -1014,7 +1012,7 @@
 	CFLAGS="$CFLAGS -D__USE_FIXED_PROTOTYPES__"
 	AC_DEFINE(NEED_STRINGS_H)
 	AC_DEFINE(NEED_MYREALLOC)
-	LIBOBJS="$LIBOBJS myrealloc.o"
+	AC_LIBOBJ([myrealloc])
 fi
 
 #
@@ -1103,10 +1101,9 @@
 AC_SUBST(RPMCANONVENDOR)
 AC_SUBST(RPMCANONOS)
 
-if test -n "$LIBOBJS" ; then
+if test -n "$LIB@&t@OBJS" ; then
     LIBMISC='$(top_builddir)/misc/libmisc.a'
 fi
-AC_SUBST(LIBOBJS)
 AC_SUBST(LIBMISC)
 
 if test X"$prefix" = XNONE ; then
@@ -1149,7 +1146,7 @@
 dnl AC_DEFINE_UNQUOTED(testdir, "$testdir")
 AC_SUBST(testdir)
 
-if test -n "$LIBOBJS"; then
+if test -n "$LIB@&t@OBJS"; then
     INCPATH="-I\$(top_builddir)/misc $INCPATH"
 fi
 AC_SUBST(INCPATH)
