diff -ur rpm-4.4.6.org/macros.in rpm-4.4.6/macros.in
--- rpm-4.4.6.org/macros.in	2006-04-08 21:04:54.000000000 +0200
+++ rpm-4.4.6/macros.in	2006-06-30 22:03:15.076727750 +0200
@@ -168,15 +168,20 @@
 %debug_package \
 %ifnarch noarch\
 %global __debug_package 1\
-%package debug\
+%package debuginfo\
 Summary: Debug information for package %{name}\
+Summary(pl): Informacje dla debuggera dla pakietu %{name}\
 Group: Development/Debug\
 AutoReqProv: 0\
-%description debug\
+%description debuginfo\
 This package provides debug information for package %{name}.\
 Debug information is useful when developing applications that use this\
 package or when debugging this package.\
-%files debug -f debugfiles.list\
+%description debuginfo -l pl\
+Ten pakiet dostarcza informacje dla debuggera dla pakietu %{name}.\
+Informacje te s± przydatne przy rozwijaniu aplikacji u¿ywaj±cych tego\
+pakietu oraz przy odpluskwianiu samego pakietu.\
+%files debuginfo -f debugfiles.list\
 %defattr(-,root,root)\
 %endif\
 %{nil}
@@ -961,7 +966,7 @@
 %__spec_install_pre	%{___build_pre}
 %__spec_install_body	%{___build_body}
 %__spec_install_post\
-%{?__debug_package:%{__debug_install_post}}\
+%{expand:%%define __sip_%{?_enable_debug_packages} 1}%{?__sip_1:%{?__debug_package:%{__debug_install_post}}}%{expand:%%undefine __sip_%{?_enable_debug_packages}}\
 %{__arch_install_post}\
 %{__os_install_post}\
 %{nil}
diff -ur rpm-4.4.6.org/platform.in rpm-4.4.6/platform.in
--- rpm-4.4.6.org/platform.in	2006-06-30 22:01:46.743207000 +0200
+++ rpm-4.4.6/platform.in	2006-06-30 22:03:15.080728000 +0200
@@ -61,7 +61,7 @@
 %{nil}
 
 %__spec_install_post\
-    %{?__debug_package:%{__debug_install_post}}\
+    %{expand:%%define __sip_%{?_enable_debug_packages} 1}%{?__sip_1:%{?__debug_package:%{__debug_install_post}}}%{expand:%%undefine __sip_%{?_enable_debug_packages}}\
     %{__arch_install_post}\
     %{__os_install_post}\
 %{nil}
diff -ur rpm-4.4.6.org/scripts/find-debuginfo.sh rpm-4.4.6/scripts/find-debuginfo.sh
--- rpm-4.4.6.org/scripts/find-debuginfo.sh	2006-02-12 15:59:51.000000000 +0100
+++ rpm-4.4.6/scripts/find-debuginfo.sh	2006-06-30 22:03:53.451126000 +0200
@@ -15,7 +15,8 @@
 
 strip_to_debug()
 {
-  eu-strip --remove-comment -f "$1" "$2" || :
+  objcopy --only-keep-debug --remove-section .comment "$2" "$1"
+  objcopy --add-gnu-debuglink="$1" "$2"
 }
 
 # Strip ELF binaries
@@ -30,7 +31,7 @@
 	[ -f "${debugfn}" ] && continue
 
 	echo extracting debug info from $f
-	/usr/lib/rpm/debugedit -b "$RPM_BUILD_DIR" -d /usr/src/debug -l "$SOURCEFILE" "$f"
+	/usr/bin/debugedit -b "$RPM_BUILD_DIR" -d /usr/src/debug -l "$SOURCEFILE" "$f"
 
 	# A binary already copied into /usr/lib/debug doesn't get stripped,
 	# just has its file names collected and adjusted.
@@ -54,4 +55,5 @@
 find ${RPM_BUILD_ROOT}/usr/src/debug -type d -print0 | xargs -0 chmod a+rx
 
 find ${RPM_BUILD_ROOT}/usr/lib/debug -type f | sed -n -e "s#^$RPM_BUILD_ROOT##p" > $LISTFILE
+find ${RPM_BUILD_ROOT}/usr/lib/debug -mindepth 1 -type d | sed -n -e "s#^$RPM_BUILD_ROOT#%dir #p" >> $LISTFILE
 find ${RPM_BUILD_ROOT}/usr/src/debug -mindepth 1 -maxdepth 1 | sed -n -e "s#^$RPM_BUILD_ROOT##p" >> $LISTFILE
