--- rpm-4.0.4/rpmio/Makefile.am.orig	Tue May 21 19:07:54 2002
+++ rpm-4.0.4/rpmio/Makefile.am	Tue May 21 19:26:14 2002
@@ -24,6 +24,7 @@
 LIBS =	@LIBS@ @WITH_ZLIB_LIB@ -lrt -lpthread
 
 BEECRYPTLOBJS = $(shell cat $(top_builddir)/beecrypt/listobjs)
+BEECRYPTOBJS = $(BEECRYPTLOBJS:%.lo=%.o)
 
 lib_LTLIBRARIES = librpmio.la
 librpmio_la_SOURCES = digest.c macro.c rpmio.c rpmlog.c rpmmalloc.c \
@@ -45,6 +46,12 @@
 	for lo in $(BEECRYPTLOBJS); do \
 	  [ -f $$lo ] || $(LN_S) $(top_builddir)/beecrypt/$$lo $$lo ; \
 	done
+	for lo in $(BEECRYPTOBJS) ; do \
+	  [ ! -f $$lo -a -f $(top_builddir)/beecrypt/$$lo ] && $(LN_S) \
+		$(top_builddir)/beecrypt/$$lo $$lo ; \
+	  [ ! -f .libs/$$lo -a -f $(top_builddir)/beecrypt/.libs/$$lo ] && $(LN_S) \
+		../$(top_builddir)/beecrypt/.libs/$$lo .libs/$$lo ; \
+	done
 	touch $@
 
 clean-local:
