diff -urN rpm-4.0.2/scripts/perl.prov rpm-4.0.2-grabba/scripts/perl.prov
--- rpm-4.0.2/scripts/perl.prov	Wed Feb 27 13:04:35 2002
+++ rpm-4.0.2-grabba/scripts/perl.prov	Wed Feb 27 13:47:06 2002
@@ -116,7 +116,7 @@
     # package name so we report all namespaces as if they were
     # provided packages (really ugly).
 
-    if (m/^\s*package\s+([_:a-zA-Z0-9]+)\s*;/) {
+    if (m/^\s*package\s+([_:a-zA-Z0-9]+)\s*;/ && !$require{$1}) {
       $package=$1;
       undef $version;
       $require{$package}=undef;
@@ -132,10 +132,15 @@
     #ExtUtils/Install.pm:$VERSION = substr q$Revision$, 10;
     #CGI/Apache.pm:$VERSION = (qw$Revision$)[1];
     #DynaLoader.pm:$VERSION = $VERSION = "1.03";     # avoid typo warning
+    #Cwd.pm:our $VERSION = '2.04';
+
+    # here is example of VERSION line from RPC::PlClient module
+
+    # RPC/PlClient.pm:$RPC::PlClient::VERSION = '0.2016';
 
     if ( 
 	($package) && 
-	(m/^\s*\$VERSION\s*=\s+/)
+	(m/^\s*(our)?\s*\$(${package}::)?VERSION\s*=/)
        ) {
 
       # first see if the version string contains the string
