--- rpm-4.2/build/rpmfc.c.wiget	Mon Jan 20 23:16:09 2003
+++ rpm-4.2/build/rpmfc.c	Tue Apr  1 09:56:51 2003
@@ -450,6 +450,8 @@
 
   { "perl script text",		RPMFC_PERL|RPMFC_INCLUDE },
   { "Perl5 module source text", RPMFC_PERL|RPMFC_MODULE|RPMFC_INCLUDE },
+  
+  { "PHP script text",		RPMFC_PHP|RPMFC_INCLUDE },
 
   { "current ar archive",	RPMFC_STATIC|RPMFC_LIBRARY|RPMFC_ARCHIVE|RPMFC_INCLUDE },
 
@@ -618,6 +621,7 @@
 
 	fc->sb_java = freeStringBuf(fc->sb_java);
 	fc->sb_perl = freeStringBuf(fc->sb_perl);
+	fc->sb_php = freeStringBuf(fc->sb_php);
 	fc->sb_python = freeStringBuf(fc->sb_python);
 
     }
@@ -705,6 +709,8 @@
 	    fc->fcolor->vals[fc->ix] |= RPMFC_PERL;
 	else if (!strcmp(bn, "python"))
 	    fc->fcolor->vals[fc->ix] |= RPMFC_PYTHON;
+	else if (!strcmp(bn, "php"))
+	    fc->fcolor->vals[fc->ix] |= RPMFC_PHP;
 
 	break;
     }
@@ -723,6 +729,10 @@
 	if (is_executable)
 	    xx = rpmfcHelper(fc, 'R', "python");
     }
+    if (fc->fcolor->vals[fc->ix] & RPMFC_PHP) {
+		xx = rpmfcHelper(fc, 'P', "php");
+	    xx = rpmfcHelper(fc, 'R', "php");
+	}
 
     return 0;
 }
@@ -1068,6 +1068,7 @@
     { rpmfcELF,		RPMFC_ELF },
     { rpmfcSCRIPT,	(RPMFC_SCRIPT|RPMFC_PERL) },
     { rpmfcSCRIPT,	(RPMFC_SCRIPT|RPMFC_PYTHON) },
+    { rpmfcSCRIPT,	(RPMFC_SCRIPT|RPMFC_PHP) },
     { NULL, 0 }
 };
 
@@ -1227,6 +1227,9 @@
 /*@-branchstate@*/
 	if (slen >= sizeof(".pm") && !strcmp(s+slen-(sizeof(".pm")-1), ".pm"))
 	    ftype = "Perl5 module source text";
+	/* XXX all files with extension ".php" are PHP modules for now. */
+	else if (slen >= sizeof(".php") && !strcmp(s+slen-(sizeof(".php")-1), ".php"))
+	    ftype = "PHP script text";
 	/* XXX skip all files in /dev/ which are (or should be) %dev dummies. */
 	else if (slen >= fc->brlen+sizeof("/dev/") && !strncmp(s+fc->brlen, "/dev/", sizeof("/dev/")-1))
 	    ftype = "";
--- rpm-4.2/build/rpmfc.h.wiget	Tue Apr  1 10:00:23 2003
+++ rpm-4.2/build/rpmfc.h	Tue Apr  1 10:01:22 2003
@@ -40,6 +40,7 @@
     StringBuf sb_java;	/*!< concatenated list of java colored files. */
     StringBuf sb_perl;	/*!< concatenated list of perl colored files. */
     StringBuf sb_python;/*!< concatenated list of python colored files. */
+    StringBuf sb_php;   /*!< concatenated list of php colored files. */
 
 };
 
