--- rpm-4.5/scripts/rpm2cpio~	2007-06-11 08:56:55.000000000 +0300
+++ rpm-4.5/scripts/rpm2cpio	2008-08-19 10:31:53.658158936 +0300
@@ -31,6 +31,6 @@
     BZh)      dd if="$pkg" ibs=$o skip=1 2>/dev/null | bunzip2 ;;
     "$gz"*)   dd if="$pkg" ibs=$o skip=1 2>/dev/null | gunzip ;;
     # no magic in old lzma format, if unknown we assume that's lzma for now
-    *)        dd if="$pkg" ibs=$o skip=1 2>/dev/null | lzma d -si -so ;;
+    *)        dd if="$pkg" ibs=$o skip=1 2>/dev/null | lzma --format=alone -d ;;
     #*)        echo "Unrecognized rpm file: $pkg"; return 1 ;;
 esac
