--- rpm-5.3.1/configure.ac.wiget	2010-05-30 16:43:06.000000000 +0200
+++ rpm-5.3.1/configure.ac	2010-05-30 16:43:38.482725491 +0200
@@ -2199,6 +2199,7 @@ AC_CONFIG_FILES([ po/Makefile.in
     tools/Makefile
     scripts/Makefile
     scripts/perl.req scripts/perl.prov
+    scripts/php.req scripts/php.prov
     rpmdb/DB_CONFIG
     macros/macros macros/macros.rpmbuild
     macros/java macros/libtool macros/mono macros/perl 
