diff -Nur rpm-4.0.2.orig/scripts/perl.req rpm-4.0.2/scripts/perl.req
--- rpm-4.0.2.orig/scripts/perl.req	Thu Jan 24 15:17:27 2002
+++ rpm-4.0.2/scripts/perl.req	Thu Jan 24 16:04:53 2002
@@ -43,7 +43,7 @@
   foreach (@ARGV) {
     if (m=/usr/(sbin|bin|lib|share|X11R6/(lib|bin))/=) {
       if (! m=(/(doc|man|info|usr/src)/|\.(so|ph|h|html|pod)$)=) {
-        process_file($_);
+        process_file($_) if -f "$_";
       }
     }
   }
@@ -55,7 +55,7 @@
   foreach (<>) {
      if (m=/usr/(sbin|bin|lib|share|X11R6/(lib|bin))/=) {
        if (! m=(/(doc|man|info|usr/src)/|\.(so|ph|h|html|pod)$)=) {
-         process_file($_);
+         process_file($_) if -f "$_";
        }
      }
   }
