--- rpm-4.4.8/lib/Makefile.am.orig	2007-01-13 15:09:49.000000000 +0100
+++ rpm-4.4.8/lib/Makefile.am	2007-04-08 17:28:55.353467622 +0200
@@ -47,7 +47,6 @@
 	$(top_builddir)/rpmio/librpmio.la \
 	@WITH_POPT_LIB@ \
 	@WITH_SELINUX_LIB@ \
-	-lresolv \
 	@LTLIBINTL@
 
 if HAVE_LD_VERSION_SCRIPT
--- rpm-4.4.8/rpmio/Makefile.am.orig	2007-04-08 16:55:58.468811386 +0200
+++ rpm-4.4.8/rpmio/Makefile.am	2007-04-08 17:29:16.078648683 +0200
@@ -49,7 +49,7 @@
 	@WITH_SYCK_LIB@ \
 	@WITH_MAGIC_LIB@ \
 	@WITH_ZLIB_LIB@ \
-	-lpthread -lresolv \
+	-lpthread \
 	@LTLIBINTL@
 librpmio_la_LIBADD = # $(BEECRYPTLOBJS)
 librpmio_la_DEPENDENCIES = # .created
--- rpm-4.4.3/configure.ac.orig	2005-11-19 11:30:59.197389000 +0100
+++ rpm-4.4.3/configure.ac	2005-11-19 16:17:10.397970072 +0100
@@ -439,7 +439,7 @@
 ])
 
 AC_CHECK_HEADERS(aio.h)
-AC_SEARCH_LIBS(aio_read, [c rt aio posix4])
+dnl not used? AC_SEARCH_LIBS(aio_read, [c rt aio posix4])
 
 dnl Better not use fchmod at all.
 AC_CHECK_FUNC(fchmod)
