diff -urN rpm-4.4.8.org/configure.ac rpm-4.4.8/configure.ac
--- rpm-4.4.8.org/configure.ac	2007-02-11 03:21:02.000000000 +0100
+++ rpm-4.4.8/configure.ac	2007-04-10 18:00:13.715685518 +0200
@@ -81,6 +81,8 @@
 AC_PATH_PROG(CTAGS, ctags, /bin/true)
 AC_PATH_PROG(CSCOPE, cscope, /bin/true)
 
+CPPFLAGS="$CPPFLAGS -Dglob=rpm_glob -Dglobfree=rpm_globfree -Dglob64=rpm_glob64 -Dglobfree64=rpm_globfree64"
+
 dnl
 dnl This now uses libtool. Put
 dnl	LDFLAGS_STATIC="-all"
diff -urN rpm-4.4.8.org/rpmio/librpmio.vers rpm-4.4.8/rpmio/librpmio.vers
--- rpm-4.4.8.org/rpmio/librpmio.vers	2007-02-02 02:36:47.000000000 +0100
+++ rpm-4.4.8/rpmio/librpmio.vers	2007-04-10 17:59:13.905039270 +0200
@@ -79,10 +79,8 @@
     Fwrite;
     gidToGname;
     Glob;
-    glob64;
     Glob_error;
     Globfree;
-    globfree64;
     Glob_pattern_p;
     gnameToGid;
     gzdio;
@@ -164,6 +162,10 @@
     rpmGetPath;
     rpmGlob;
     rpmGlobalMacroContext;
+    rpm_glob;
+    rpm_glob64;
+    rpm_globfree;
+    rpm_globfree64;
     rpmhookArgsFree;
     rpmhookArgsNew;
     rpmhookCall;
