--- rpm-4.4.9/lib/misc.c~	2007-01-21 19:33:05.000000000 +0200
+++ rpm-4.4.9/lib/misc.c	2008-04-29 23:27:33.808321000 +0300
@@ -176,8 +176,10 @@
 	/* XXX FIXME: errno may not be correct for ufdio */
     } while ((fd == NULL || Ferror(fd)) && errno == EEXIST);
 
-    if (fd == NULL || Ferror(fd))
+    if (fd == NULL || Ferror(fd)) {
+	rpmError(RPMERR_SCRIPT, _("error creating temporary file %s\n"), tempfn);
 	goto errxit;
+    }
 
     switch(temput) {
     case URL_IS_PATH:
