--- rpm-4.0.2/lib/Makefile.am.wiget	Wed Feb 14 20:44:20 2001
+++ rpm-4.0.2/lib/Makefile.am	Wed Feb 21 11:57:02 2001
@@ -22,7 +22,7 @@
 
 lib_LTLIBRARIES = librpm.la
 librpm_la_SOURCES = \
-	cpio.c $(DBLIBSRCS) dbconfig.c depends.c \
+	cpio.c dbconfig.c depends.c \
 	formats.c fprint.c fs.c hash.c header.c install.c \
 	md5.c md5sum.c misc.c package.c problems.c \
 	poptBT.c poptQV.c query.c rpmchecksig.c rpmdb.c rpminstall.c \
