--- rpm-5.1b2/Doxyfile.in~	2008-04-13 19:14:33.000000000 +0300
+++ rpm-5.1b2/Doxyfile.in	2008-04-13 19:15:06.904571320 +0300
@@ -503,7 +503,6 @@
                          @top_srcdir@/doc/manual/spec \
                          @top_srcdir@/doc/manual/triggers \
                          @top_srcdir@/doc/manual/tsort \
-                         @top_srcdir@/file/src/tar.h \
                          @top_srcdir@/lib/depends.c \
                          @top_srcdir@/lib/formats.c \
                          @top_srcdir@/lib/fs.c \
