--- rpm-4.3/rpmio/Makefile.am.wiget	Fri May 16 00:16:34 2003
+++ rpm-4.3/rpmio/Makefile.am	Fri May 16 00:18:17 2003
@@ -70,34 +70,34 @@
 	$(LINT) $(DEFS) $(INCLUDES) $(librpmio_la_SOURCES)
 
 tdigest_SOURCES = tdigest.c
-tdigest_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la -lgcrypt
+tdigest_LDADD = librpmio.la -lpopt -lgcrypt
 
 tdir_SOURCES = tdir.c
 tdir_LDFLAGS = -all-static
-tdir_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la
+tdir_LDADD = librpmio.la -lpopt
 
 tfts_SOURCES = tfts.c
 tfts_LDFLAGS = -all-static
-tfts_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la
+tfts_LDADD = librpmio.la -lpopt
 
 tglob_SOURCES = tglob.c
 tglob_LDFLAGS = -all-static
-tglob_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la
+tglob_LDADD = librpmio.la -lpopt
 
 tinv_SOURCES = tinv.c
 tinv_LDFLAGS = -all-static
-tinv_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la
+tinv_LDADD = librpmio.la -lpopt
 
 tkey_SOURCES = tkey.c
 tkey_LDFLAGS = -all-static
-tkey_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la
+tkey_LDADD = librpmio.la -lpopt
 
 tring_SOURCES = tring.c
 tring_LDFLAGS = -all-static
-tring_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la
+tring_LDADD = librpmio.la -lpopt
 
 trpmio_SOURCES = trpmio.c
-trpmio_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la
+trpmio_LDADD = librpmio.la -lpopt
 
 tsw_SOURCES = tsw.c
 tsw_LDFLAGS = librpmio.la
--- rpm-4.3/rpmdb/Makefile.am.wiget	Tue Mar 25 13:49:44 2003
+++ rpm-4.3/rpmdb/Makefile.am	Fri May 16 00:16:34 2003
@@ -49,7 +49,7 @@
 	tagname.c tagtbl.c
 librpmdb_la_LDFLAGS = -release @VERSION@ $(LDFLAGS) \
 	$(top_builddir)/rpmio/librpmio.la \
-	$(top_builddir)/popt/libpopt.la \
+	@WITH_POPT_LIB@ \
 	@WITH_LIBELF_LIB@ \
 	@libdb3@
 librpmdb_la_LIBADD = $(DBLIBOBJS) $(DB3LOBJS)
--- rpm-4.3/lib/Makefile.am.wiget	Tue Apr  8 18:38:08 2003
+++ rpm-4.3/lib/Makefile.am	Fri May 16 00:16:34 2003
@@ -39,7 +39,7 @@
 librpm_la_LDFLAGS = -release @VERSION@ $(LDFLAGS) \
 	$(top_builddir)/rpmdb/librpmdb.la \
 	$(top_builddir)/rpmio/librpmio.la \
-	$(top_builddir)/popt/libpopt.la
+	@WITH_POPT_LIB@
 
 getdate.c: getdate.y
 	@echo expect 10 shift/reduce conflicts
--- rpm-4.3/tools/Makefile.am.wiget	Thu Apr 17 18:45:19 2003
+++ rpm-4.3/tools/Makefile.am	Fri May 16 00:16:34 2003
@@ -39,7 +39,7 @@
 
 debugedit_SOURCES =	debugedit.c hashtab.c
 debugedit_LDADD =	@LDFLAGS_STATIC@ @WITH_LIBELF_LIB@ \
-	$(top_builddir)/popt/libpopt.la
+	-lpopt
 
 javadeps_SOURCES =	javadeps.c
 
--- rpm-4.3/configure.ac.wiget	Fri May 16 00:16:34 2003
+++ rpm-4.3/configure.ac	Fri May 16 00:16:44 2003
@@ -410,6 +410,17 @@
 ])
 AC_SUBST(WITH_LIBDWARF_INCLUDE)
 
+WITH_POPT_SUBDIR=
+WITH_POPT_LIB=
+AC_CHECK_HEADERS([popt.h])
+AC_CHECK_LIB(popt, poptGetContext, [
+	AC_DEFINE(HAVE_LIBPOPT, 1, [Define to 1 if you have libpopt.])
+	WITH_POPT_LIB="-lpopt"
+	],[WITH_POPT_SUBDIR="popt"
+	WITH_POPT_LIB="\$(top_builddir)/popt/libpopt.la"])
+	
+AC_SUBST(WITH_POPT_SUBDIR)
+AC_SUBST(WITH_POPT_LIB)
 AC_CHECK_FUNC(setreuid, [], [
     AC_CHECK_LIB(ucb, setreuid, [if echo $LIBS | grep -- -lucb >/dev/null ;then :; else LIBS="$LIBS -lc -lucb" USEUCB=y;fi])
 ])
@@ -470,6 +481,7 @@
 
 dnl Check for Berkeley db3 API.
 AC_CHECK_FUNC(db_create, [DBLIBSRCS="$DBLIBSRCS db3.c"],
+  AC_CHECK_LIB(db-4.1, db_create, [DBLIBSRCS="$DBLIBSRCS db3.c"; libdb3="-ldb-4.1"],
   AC_CHECK_LIB(db-3.2, db_create, [DBLIBSRCS="$DBLIBSRCS db3.c"; libdb3="-ldb-3.2"],
     AC_CHECK_LIB(db-3.1, db_create, [DBLIBSRCS="$DBLIBSRCS db3.c"; libdb3="-ldb-3.1"],
       AC_CHECK_LIB(db-3.0, db_create, [DBLIBSRCS="$DBLIBSRCS db3.c"; libdb3="-ldb-3.0"],
@@ -478,6 +490,7 @@
       )
     )
   )
+  )
 )
 
 if test X"$DBLIBSRCS" = X; then
@@ -1236,7 +1249,7 @@
 dnl # XXX Propagate -lucb to popt ...
 dnl export LIBS INCPATH CONFIG_SITE
 
-AC_CONFIG_SUBDIRS(popt beecrypt zlib elfutils file db3)
+AC_CONFIG_SUBDIRS(beecrypt file)
 
 AC_OUTPUT([ Doxyfile Makefile rpmrc macros platform rpmpopt rpm.spec
 	scripts/perl.req scripts/perl.prov
@@ -1260,7 +1273,7 @@
 	python/rpmdb/test/Makefile
 	python/mpw/Makefile
 	python/mpw/test/Makefile
-  ], [	echo timestamp > popt/stamp-h.in
+  ], [	
 	echo timestamp > beecrypt/stamp-h.in
 	echo timestamp > stamp-h.in
   ]
--- rpm-4.3/Makefile.am.wiget	Mon May  5 10:44:39 2003
+++ rpm-4.3/Makefile.am	Fri May 16 00:16:34 2003
@@ -14,7 +14,7 @@
 	po/*.in po/*.po po/rpm.pot \
 	rpm.magic rpmpopt-$(VERSION) rpmqv.c rpm.c
 
-SUBDIRS = intl po @WITH_ZLIB_SUBDIR@ @WITH_ELFUTILS_SUBDIR@ file @WITH_DB_SUBDIR@ popt beecrypt rpmio rpmdb lib build misc @WITH_PYTHON_SUBDIR@ tools scripts tests doc .
+SUBDIRS = intl po @WITH_ZLIB_SUBDIR@ @WITH_ELFUTILS_SUBDIR@ file @WITH_DB_SUBDIR@ @WITH_POPT_SUBDIR@ beecrypt rpmio rpmdb lib build misc @WITH_PYTHON_SUBDIR@ tools scripts tests doc .
 
 INCLUDES = \
 	-I$(top_srcdir)/build \
@@ -34,7 +34,7 @@
 	$(top_builddir)/lib/librpm.la \
 	$(top_builddir)/rpmdb/librpmdb.la \
 	$(top_builddir)/rpmio/librpmio.la \
-	$(top_builddir)/popt/libpopt.la \
+	-lpopt \
 	@WITH_ZLIB_LIB@ \
 	@INTLLIBS@ \
 	@LIBMISC@
@@ -105,7 +105,6 @@
 	    -load lib/rpmlib.lcd \
 	    -load rpmdb/rpmdb.lcd \
 	    -load rpmio/rpmio.lcd \
-	    -load popt/popt.lcd \
 		$(DEFS) $(INCLUDES) rpmqv.c $(rpmb_SOURCES)
 
 .PHONY:	lint
