--- rpm/lib/signature.c.orig	2003-12-23 06:25:05.000000000 +0000
+++ rpm/lib/signature.c	2004-08-03 21:26:09.340102703 +0000
@@ -1226,7 +1226,7 @@
 	goto exit;
 
     (void) rpmswEnter(rpmtsOp(ts, RPMTS_OP_SIGNATURE), 0);
-    if (rsavrfy(&dig->rsa_pk, &dig->rsahm, &dig->c))
+    if (rsavrfy(&dig->rsa_pk.n, &dig->rsa_pk.e, &dig->rsahm, &dig->c))
 	res = RPMRC_OK;
     else
 	res = RPMRC_FAIL;
--- rpm/rpmio/rpmio_internal.h.orig	2003-11-23 19:52:39.000000000 +0000
+++ rpm/rpmio/rpmio_internal.h	2004-08-03 21:15:42.497336945 +0000
@@ -9,7 +9,7 @@
 #include <rpmio.h>
 #include <rpmurl.h>
 
-#include <beecrypt.api.h>
+#include <beecrypt/api.h>
 #include <rpmpgp.h>
 #include <rpmsw.h>
 
