diff -urN rpm-4.3.org/macros.in rpm-4.3/macros.in
--- rpm-4.3.org/macros.in	2004-05-20 22:09:48.582353920 +0200
+++ rpm-4.3/macros.in	2004-05-20 22:12:42.086977208 +0200
@@ -118,15 +118,20 @@
 %debug_package \
 %ifnarch noarch\
 %global __debug_package 1\
-%package debug\
+%package debuginfo\
 Summary: Debug information for package %{name}\
+Summary(pl): Informacje dla debuggera dla pakietu %{name}\
 Group: Development/Debug\
 AutoReqProv: 0\
-%description debug\
+%description debuginfo\
 This package provides debug information for package %{name}.\
 Debug information is useful when developing applications that use this\
 package or when debugging this package.\
-%files debug -f debugfiles.list\
+%description debuginfo -l pl\
+Ten pakiet dostarcza informacje dla debuggera dla pakietu %{name}.\
+Informacje te s± przydatne przy rozwijaniu aplikacji u¿ywaj±cych tego\
+pakietu oraz przy odpluskwianiu samego pakietu.\
+%files debuginfo -f debugfiles.list\
 %defattr(-,root,root)\
 %endif\
 %{nil}
@@ -1247,7 +1252,12 @@
 unset DISPLAY ||:\
 %{nil}
 
-%install %{?_enable_debug_packages:%{?buildsubdir:%{debug_package}}}\
+%install \
+%if %{?_enable_debug_packages:1}%{!?_enable_debug_packages:0}\
+%if %{_enable_debug_packages}\
+%{?buildsubdir:%{debug_package}}\
+%endif\
+%endif\
 %%install\
 LANG=C\
 export LANG\
