--- rpm-4.0.4/python/Makefile.am.wiget2	Tue Feb 12 23:20:17 2002
+++ rpm-4.0.4/python/Makefile.am	Tue Feb 12 23:21:09 2002
@@ -34,10 +34,10 @@
 noinst_LTLIBRARIES = librpmmodule.la
 librpmmodule_la_SOURCES = rpmmodule.c hash.c upgrade.c
 
-rpmmodule.so: $(librpmmodule_la_OBJECTS)
+rpmmodule.so$(EXEEXT): $(librpmmodule_la_OBJECTS)
 	$(LINK) -o $@ $(librpmmodule_la_OBJECTS) $(rpmmodule_so_LDFLAGS)
 
-poptmodule.so: $(poptmodule_so_OBJECTS)
+poptmodule.so$(EXEEXT): $(poptmodule_so_OBJECTS)
 	$(LINK) -o $@ $(poptmodule_so_OBJECTS) $(poptmodule_so_LDFLAGS)
 
 .PHONY:	lclint
