--- rpm-4.3/configure.ac.orig	2004-12-23 12:17:23.000217152 +0000
+++ rpm-4.3/configure.ac	2004-12-23 12:19:30.094895824 +0000
@@ -817,6 +817,16 @@
 WITH_PYTHON_VERSION=$withval
 if test $withval = auto ; then
 
+AC_MSG_CHECKING(for python 2.4)
+AC_RUN_IFELSE([AC_LANG_SOURCE([[
+#include <python2.4/Python.h>
+main() {
+  exit(strncmp("2.3", PY_VERSION, 3));
+} ]])],[withval=yes],[withval=no],[withval=yes])
+  AC_MSG_RESULT($withval)
+  if test $withval = yes ; then
+    WITH_PYTHON_VERSION="2.4"
+  else
 AC_MSG_CHECKING(for python 2.3)
 AC_RUN_IFELSE([AC_LANG_SOURCE([[
 #include <python2.3/Python.h>
@@ -851,7 +861,7 @@
       fi
     fi
   fi
-
+fi
 fi
 
 if test "$WITH_PYTHON_VERSION" != no ; then
