Index: macros.in
===================================================================
RCS file: /cvs/devel/rpm/macros.in,v
retrieving revision 1.154.2.61
diff -u -b -B -w -p -r1.154.2.61 macros.in
--- macros.in	3 Feb 2007 20:02:07 -0000	1.154.2.61
+++ macros.in	4 Feb 2007 06:37:17 -0000
@@ -928,7 +928,10 @@ print (t)\
 %_vendor		@RPMCANONVENDOR@
 %_os			@RPMCANONOS@
 %_gnu			-gnu
-%_target_platform	%{_target_cpu}-%{_vendor}-%{_target_os}%{?_gnu}
+
+
+
+%_target_platform	%{_target_cpu}-%{_target_vendor}-%{_target_os}%{?_gnu}
 
 #
 # Define a generic value for optflags. Normally overridden by per-target macros.
Index: lib/rpmrc.c
===================================================================
RCS file: /cvs/devel/rpm/lib/rpmrc.c,v
retrieving revision 2.193.2.27
diff -u -b -B -w -p -r2.193.2.27 rpmrc.c
--- lib/rpmrc.c	21 Jan 2007 21:11:46 -0000	2.193.2.27
+++ lib/rpmrc.c	4 Feb 2007 06:37:30 -0000
@@ -1774,8 +1774,6 @@ static void rpmRebuildTargetVars(const c
     addMacro(NULL, "_target", NULL, ct, RMIL_RPMRC);
     delMacro(NULL, "_target_cpu");
     addMacro(NULL, "_target_cpu", NULL, ca, RMIL_RPMRC);
-    delMacro(NULL, "_target_vendor");
-    addMacro(NULL, "_target_vendor", NULL, "unknown", RMIL_RPMRC);
     delMacro(NULL, "_target_os");
     addMacro(NULL, "_target_os", NULL, co, RMIL_RPMRC);
 /*
