--- rpm-4.3/rpmrc.in.wiget	2004-03-29 16:08:10.816846952 +0200
+++ rpm-4.3/rpmrc.in	2004-03-29 16:08:10.825845538 +0200
@@ -189,7 +189,8 @@
 buildarchtranslate: sun4m: sparc
 buildarchtranslate: sparcv8: sparc
 buildarchtranslate: sparcv9: sparc
-buildarchtranslate: sun4u: sparc64
+buildarchtranslate: sun4u: sparc
+buildarchtranslate: sparc64: sparc
 
 buildarchtranslate: osfmach3_ppc: ppc
 buildarchtranslate: powerpc: ppc
