diff -uNr rpm-4.3/rpmio/rpmsq.c rpm-4.3.fixed/rpmio/rpmsq.c
--- rpm-4.3/rpmio/rpmsq.c	2004-01-04 03:13:09.000000000 +0100
+++ rpm-4.3.fixed/rpmio/rpmsq.c	2004-02-04 20:29:39.000000000 +0100
@@ -135,20 +135,20 @@
 #define	DO_UNLOCK()	pthread_mutex_unlock(&rpmsigTbl_lock);
 #define	INIT_LOCK()	\
     {	pthread_mutexattr_t attr; \
-	(void) pthread_mutexattr_init(&attr); \
-	(void) pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_RECURSIVE); \
-	(void) pthread_mutex_init (&rpmsigTbl_lock, &attr); \
-	(void) pthread_mutexattr_destroy(&attr); \
+	pthread_mutexattr_init(&attr); \
+	pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_RECURSIVE); \
+	pthread_mutex_init (&rpmsigTbl_lock, &attr); \
+	pthread_mutexattr_destroy(&attr); \
 	rpmsigTbl_sigchld->active = 0; \
     }
 #define	ADD_REF(__tbl)	(__tbl)->active++
 #define	SUB_REF(__tbl)	--(__tbl)->active
 #define	CLEANUP_HANDLER(__handler, __arg, __oldtypeptr) \
-    (void) pthread_setcanceltype (PTHREAD_CANCEL_ASYNCHRONOUS, (__oldtypeptr));\
+	pthread_setcanceltype (PTHREAD_CANCEL_ASYNCHRONOUS, (__oldtypeptr));\
 	pthread_cleanup_push((__handler), (__arg));
 #define	CLEANUP_RESET(__execute, __oldtype) \
-    (void) pthread_cleanup_pop(__execute); \
-    (void) pthread_setcanceltype ((__oldtype), &(__oldtype));
+	pthread_cleanup_pop(__execute); \
+	pthread_setcanceltype ((__oldtype), &(__oldtype));
 
 #define	SAME_THREAD(_a, _b)	pthread_equal(((pthread_t)_a), ((pthread_t)_b))
 
