--- rpm-4.4.9/build/parseBuildInstallClean.c	2008-01-30 23:16:55.347346942 +0200
+++ rpm-4.4.9/build/parseBuildInstallClean.c	2008-01-30 23:22:24.241459876 +0200
@@ -51,7 +51,7 @@
 	if (s && *s)
 	    *iobp = rpmiobAppend(*iobp, s, 0);
 	s = _free(s);
-#if !defined(RPM_VENDOR_OPENPKG) /* still-support-section-clean */
+#if !defined(RPM_VENDOR_OPENPKG) && !defined(RPM_VENDOR_PLD) /* still-support-section-clean */
 	/* OpenPKG still wishes to use "%clean" script/section */
 	iobp = NULL;	/* XXX skip %clean from spec file. */
 #endif
