diff -ur rpm-4.0.4/build/files.c rpm-4.0.4-/build/files.c
--- rpm-4.0.4/build/files.c	Thu Feb 14 18:28:48 2002
+++ rpm-4.0.4-/build/files.c	Fri May 17 11:16:53 2002
@@ -1002,6 +1002,16 @@
 	    appendStringBuf(pkg->specialDoc, "cp -pr ");
 	    appendStringBuf(pkg->specialDoc, specialDocBuf);
 	    appendLineStringBuf(pkg->specialDoc, " $DOCDIR");
+
+	    {
+	        char *compress_doc;
+
+		compress_doc = rpmExpand("%{__compress_doc}", NULL);
+		if (compress_doc && *compress_doc != '%')
+	    	    appendLineStringBuf(pkg->specialDoc, compress_doc);
+		if (compress_doc)
+		  free(compress_doc);
+	    }
 	}
     }
 
diff -ur rpm-4.0.4/platform.in rpm-4.0.4-/platform.in
--- rpm-4.0.4/platform.in	Fri May 17 11:18:12 2002
+++ rpm-4.0.4-/platform.in	Fri May 17 11:17:44 2002
@@ -102,6 +102,9 @@
     %{__os_install_post}\
 %{nil}
 
+# Executed at after copying %doc to DOCDIR, with $DOCDIR set
+%__compress_doc		@prefix@/lib/rpm/compress-doc%{?_noautocompressdoc: --noautocompressdoc='%{_noautocompressdoc}'}
+
 @redhat@#---------------------------------------------------------------------
 @redhat@#	Expanded at end of %prep
 @redhat@#
