--- rpm-4.4.2/macros.in~	2006-05-24 11:55:23.556060140 +0300
+++ rpm-4.4.2/macros.in	2006-05-24 11:58:19.559999273 +0300
@@ -327,8 +327,8 @@
 #		"w9.gzdio"	gzip level 9 (default).
 #		"w9.bzdio"	bzip2 level 9.
 #
-#%_source_payload	w9.gzdio
-#%_binary_payload	w9.gzdio
+%_source_payload	w9.gzdio
+%_binary_payload	w9.bzdio
 
 #	The signature to use and the location of configuration files for
 #	signing packages with PGP.
