--- rpm-4.4.2.2/Doxyfile.in~	2007-12-20 00:39:37.204213809 +0200
+++ rpm-4.4.2.2/Doxyfile.in	2007-12-20 00:55:37.952346428 +0200
@@ -398,7 +398,6 @@
                          @top_srcdir@/doc/manual/spec \
                          @top_srcdir@/doc/manual/triggers \
                          @top_srcdir@/doc/manual/tsort \
-                         @top_srcdir@/file/src/
                          @top_srcdir@/lib/
 						 @top_srcdir@/lua/
                          @top_srcdir@/rpmdb/
