--- rpm-5.4.10/build/files.c~	2013-01-24 14:24:39.161982722 +0100
+++ rpm-5.4.10/build/files.c	2013-01-24 14:27:50.620325354 +0100
@@ -1045,6 +1045,7 @@
 		pkg->specialDoc = rpmiobAppend(pkg->specialDoc, "DOCDIR=\"$RPM_BUILD_ROOT\"", 0);
 		pkg->specialDoc = rpmiobAppend(pkg->specialDoc, buf, 1);
 		pkg->specialDoc = rpmiobAppend(pkg->specialDoc, "export DOCDIR", 1);
+		pkg->specialDoc = rpmiobAppend(pkg->specialDoc, "if test -d \"$DOCDIR\" ; then rmdir \"$DOCDIR\" ; fi", 1);
 		mkdir_p = rpmExpand("%{?__mkdir_p}%{!?__mkdir_p:mkdir -p}", NULL);
 		if (!mkdir_p)
 		    mkdir_p = xstrdup("mkdir -p");
