diff -urN rpm.org/configure.ac rpm/configure.ac
--- rpm.org/configure.ac	2004-06-14 11:46:10.220092656 +0200
+++ rpm/configure.ac	2004-06-14 11:47:13.499472728 +0200
@@ -428,6 +428,18 @@
 AC_SUBST(WITH_BEECRYPT_INCLUDE)
 AC_SUBST(WITH_BEECRYPT_LIB)
 
+WITH_POPT_SUBDIR=
+WITH_POPT_LIB=
+AC_CHECK_HEADERS([popt.h])
+AC_CHECK_LIB(popt, poptGetContext, [
+       AC_DEFINE(HAVE_LIBPOPT, 1, [Define to 1 if you have libpopt.])
+       WITH_POPT_LIB="-lpopt"
+       ],[WITH_POPT_SUBDIR="popt"
+       WITH_POPT_LIB="\$(top_builddir)/popt/libpopt.la"])
+
+AC_SUBST(WITH_POPT_SUBDIR)
+AC_SUBST(WITH_POPT_LIB)
+
 AC_CHECK_FUNC(setreuid, [], [
     AC_CHECK_LIB(ucb, setreuid, [if echo $LIBS | grep -- -lucb >/dev/null ;then :; else LIBS="$LIBS -lc -lucb" USEUCB=y;fi])
 ])
@@ -1203,7 +1215,7 @@
 dnl # XXX Propagate -lucb to popt ...
 dnl export LIBS INCPATH CONFIG_SITE
 
-AC_CONFIG_SUBDIRS(popt file db3)
+AC_CONFIG_SUBDIRS(file)
 
 AC_CONFIG_FILES([ Doxyfile Makefile rpmrc macros platform rpmpopt rpm.spec
 	scripts/perl.req scripts/perl.prov
@@ -1227,7 +1239,7 @@
 	python/rpmdb/test/Makefile
  	lua/Makefile
   ])
-AC_CONFIG_COMMANDS([default],[[	echo timestamp > popt/stamp-h.in
+AC_CONFIG_COMMANDS([default],[[	
 	echo timestamp > stamp-h.in
   
 ]],[[]])
diff -urN rpm.org/lib/Makefile.am rpm/lib/Makefile.am
--- rpm.org/lib/Makefile.am	2004-06-14 11:46:08.925289496 +0200
+++ rpm/lib/Makefile.am	2004-06-14 11:47:13.502472272 +0200
@@ -40,7 +40,7 @@
 librpm_la_LDFLAGS = -release 4.3 $(LDFLAGS) \
 	$(top_builddir)/rpmdb/librpmdb.la \
 	$(top_builddir)/rpmio/librpmio.la \
-	$(top_builddir)/popt/libpopt.la \
+	-lpopt \
 	@WITH_SELINUX_LIB@
 
 getdate.c: getdate.y
diff -urN rpm.org/Makefile.am rpm/Makefile.am
--- rpm.org/Makefile.am	2004-06-14 11:46:04.409975928 +0200
+++ rpm/Makefile.am	2004-06-14 11:48:04.236759488 +0200
@@ -14,7 +14,7 @@
 	po/*.in po/*.po po/rpm.pot \
 	rpm.magic rpmpopt-$(VERSION) rpmqv.c rpm.c
 
-SUBDIRS = intl po @WITH_ZLIB_SUBDIR@ @WITH_ELFUTILS_SUBDIR@ file @WITH_DB_SUBDIR@ popt @WITH_BEECRYPT_SUBDIR@ lua rpmio rpmdb lib build misc @WITH_PYTHON_SUBDIR@ tools scripts tests doc .
+SUBDIRS = intl po @WITH_ZLIB_SUBDIR@ @WITH_ELFUTILS_SUBDIR@ file @WITH_DB_SUBDIR@ @WITH_BEECRYPT_SUBDIR@ lua rpmio rpmdb lib build misc @WITH_PYTHON_SUBDIR@ tools scripts tests doc .
 
 INCLUDES = \
 	-I$(top_srcdir)/build \
@@ -22,7 +22,6 @@
 	-I$(top_srcdir)/rpmdb \
 	-I$(top_srcdir)/rpmio \
 	@WITH_BEECRYPT_INCLUDE@ \
-	-I$(top_srcdir)/popt \
 	@WITH_LIBELF_INCLUDE@ \
 	@INCPATH@
 
@@ -34,7 +33,7 @@
 	$(top_builddir)/lib/librpm.la \
 	$(top_builddir)/rpmdb/librpmdb.la \
 	$(top_builddir)/rpmio/librpmio.la \
-	$(top_builddir)/popt/libpopt.la \
+	-lpopt \
 	@WITH_ZLIB_LIB@ \
 	@INTLLIBS@ \
 	@LIBMISC@
@@ -107,7 +106,6 @@
 	    -load lib/rpmlib.lcd \
 	    -load rpmdb/rpmdb.lcd \
 	    -load rpmio/rpmio.lcd \
-	    -load popt/popt.lcd \
 		$(DEFS) $(INCLUDES) rpmqv.c $(rpmb_SOURCES)
 
 .PHONY:	lint
@@ -117,9 +115,7 @@
 		`make -s sources -C lib` \
 		`make -s sources -C rpmdb` \
 		`make -s sources -C rpmio` \
-		`make -s sources -C beecrypt` \
-		`make -s sources -C file/src` \
-		`make -s sources -C popt`
+		`make -s sources -C file/src`
 
 CVSTAG = r$(subst .,-,$(VERSION))
 
diff -urN rpm.org/rpmdb/Makefile.am rpm/rpmdb/Makefile.am
--- rpm.org/rpmdb/Makefile.am	2004-06-14 11:46:09.726167744 +0200
+++ rpm/rpmdb/Makefile.am	2004-06-14 11:47:13.524468928 +0200
@@ -9,7 +9,6 @@
 	-I$(top_srcdir)/lib \
 	-I$(top_srcdir)/rpmio \
 	@WITH_BEECRYPT_INCLUDE@ \
-	-I$(top_srcdir)/popt \
 	@WITH_LIBELF_INCLUDE@ \
 	@INCPATH@
 
@@ -51,7 +50,7 @@
 	tagname.c tagtbl.c
 librpmdb_la_LDFLAGS = -release 4.3 \
 	$(top_builddir)/rpmio/librpmio.la \
-	$(top_builddir)/popt/libpopt.la \
+	@WITH_POPT_LIB@ \
 	@WITH_LIBELF_LIB@
 librpmdb_la_LIBADD = $(DBLIBOBJS) $(libdb_la)
 librpmdb_la_DEPENDENCIES = $(DBLIBOBJS) $(libdb_la)
--- rpm.org/rpmio/Makefile.am.org	2004-11-18 17:42:53.756263795 +0100
+++ rpm.org/rpmio/Makefile.am	2004-11-18 17:43:09.526885309 +0100
@@ -12,7 +12,6 @@
 	-I$(top_srcdir) \
 	@WITH_BEECRYPT_INCLUDE@ \
 	@WITH_LUA_INCLUDE@ \
-	-I$(top_srcdir)/popt \
 	@INCPATH@
 
 pkgincdir = $(pkgincludedir)
@@ -70,43 +69,43 @@
 tax_LDADD = librpmio.la
 
 tdigest_SOURCES = tdigest.c
-tdigest_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la -lgcrypt
+tdigest_LDADD = librpmio.la -lpopt -lgcrypt
 
 tdir_SOURCES = tdir.c
 tdir_LDFLAGS = -all-static
-tdir_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la
+tdir_LDADD = librpmio.la -lpopt
 
 tfts_SOURCES = tfts.c
 tfts_LDFLAGS = -all-static
-tfts_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la
+tfts_LDADD = librpmio.la -lpopt
 
 tget_SOURCES = tget.c
 tget_LDFLAGS = -all-static
-tget_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la
+tget_LDADD = librpmio.la -lpopt
 
 tput_SOURCES = tput.c
 tput_LDFLAGS = -all-static
-tput_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la
+tput_LDADD = librpmio.la -lpopt
 
 
 tglob_SOURCES = tglob.c
 tglob_LDFLAGS = -all-static
-tglob_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la
+tglob_LDADD = librpmio.la -lpopt
 
 tinv_SOURCES = tinv.c
 tinv_LDFLAGS = -all-static
-tinv_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la
+tinv_LDADD = librpmio.la -lpopt
 
 tkey_SOURCES = tkey.c
 tkey_LDFLAGS = -all-static
-tkey_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la
+tkey_LDADD = librpmio.la -lpopt
 
 tring_SOURCES = tring.c
 tring_LDFLAGS = -all-static
-tring_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la
+tring_LDADD = librpmio.la -lpopt
 
 trpmio_SOURCES = trpmio.c
-trpmio_LDADD = librpmio.la $(top_builddir)/popt/libpopt.la
+trpmio_LDADD = librpmio.la -lpopt
 
 tsw_SOURCES = tsw.c
 tsw_LDFLAGS = librpmio.la
diff -urN rpm.org/tools/Makefile.am rpm/tools/Makefile.am
--- rpm.org/tools/Makefile.am	2004-06-14 11:46:09.906140384 +0200
+++ rpm/tools/Makefile.am	2004-06-14 11:47:13.532467712 +0200
@@ -9,7 +9,6 @@
 	-I$(top_srcdir)/rpmdb \
 	-I$(top_srcdir)/rpmio \
 	@WITH_BEECRYPT_INCLUDE@ \
-	-I$(top_srcdir)/popt \
 	@WITH_LIBELF_INCLUDE@ \
 	@WITH_LIBDWARF_INCLUDE@ \
 	-I$(top_srcdir)/file/src \
@@ -37,7 +36,7 @@
 
 debugedit_SOURCES =	debugedit.c hashtab.c
 debugedit_LDADD =	@LDFLAGS_STATIC@ @WITH_LIBELF_LIB@ \
-	$(top_builddir)/popt/libpopt.la
+	-lpopt
 
 javadeps_SOURCES =	javadeps.c
 
--- rpm/python/Makefile.am.org	2004-11-18 20:01:55.658219108 +0100
+++ rpm/python/Makefile.am	2004-11-18 20:02:10.571929388 +0100
@@ -16,7 +16,6 @@
 	-I$(top_srcdir)/rpmdb \
 	-I$(top_srcdir)/rpmio \
 	@WITH_BEECRYPT_INCLUDE@ \
-	-I$(top_srcdir)/popt \
 	-I$(top_srcdir)/build \
 	-I$(pyincdir) \
 	@WITH_LIBELF_INCLUDE@ \
@@ -31,7 +30,7 @@
 	$(top_builddir)/lib/librpm.la \
 	$(top_builddir)/rpmdb/librpmdb.la \
 	$(top_builddir)/rpmio/librpmio.la \
-	$(top_builddir)/popt/libpopt.la \
+	-lpopt \
 	$(top_builddir)/build/librpmbuild.la \
 	@WITH_LIBELF_LIB@
 
