--- rpm-4.3/configure.ac.wiget	2003-06-11 02:15:29.000000000 +0200
+++ rpm-4.3/configure.ac	2003-06-11 02:15:29.000000000 +0200
@@ -1273,7 +1273,7 @@
 dnl # XXX Propagate -lucb to popt ...
 dnl export LIBS INCPATH CONFIG_SITE
 
-AC_CONFIG_SUBDIRS(beecrypt file)
+AC_CONFIG_SUBDIRS(file)
 
 AC_OUTPUT([ Doxyfile Makefile rpmrc macros platform rpmpopt rpm.spec
 	scripts/perl.req scripts/perl.prov
@@ -1298,7 +1298,6 @@
 	python/mpw/Makefile
 	python/mpw/test/Makefile
   ], [	
-	echo timestamp > beecrypt/stamp-h.in
 	echo timestamp > stamp-h.in
   ]
 )
