--- rpm-4.3/file/Makefile.am.orig	2003-04-16 18:37:23.000000000 +0200
+++ rpm-4.3/file/Makefile.am	2003-06-08 21:37:17.000000000 +0200
@@ -2,4 +2,4 @@
 
 EXTRA_DIST = LEGAL.NOTICE MAINT Makefile.std
 
-SUBDIRS = src magic doc
+SUBDIRS = src doc
--- rpm-4.3/file/src/Makefile.am.orig	2003-04-17 18:44:59.000000000 +0200
+++ rpm-4.3/file/src/Makefile.am	2003-06-08 21:36:44.000000000 +0200
@@ -26,7 +26,7 @@
 file_LDADD =	libfmagic.la
 
 pkglibdir =  @prefix@/lib/rpm
-MAGIC	 = $(pkglibdir)/magic
+MAGIC	 = /usr/share/file/magic
 CPPFLAGS = -DMAGIC='"$(MAGIC)"'
 
 listobjs:
