--- rpm-4.4.2.2/rpmrc.in~	2007-12-18 00:45:31.504518760 +0200
+++ rpm-4.4.2.2/rpmrc.in	2007-12-18 00:47:02.070572082 +0200
@@ -186,12 +186,12 @@
 buildarchtranslate: osfmach3_i486: i386
 buildarchtranslate: osfmach3_i386: i386
 
-buildarchtranslate: athlon: i386
-buildarchtranslate: pentium4: i386
-buildarchtranslate: pentium3: i386
-buildarchtranslate: i686: i386
-buildarchtranslate: i586: i386
-buildarchtranslate: i486: i386
+buildarchtranslate: athlon: athlon
+buildarchtranslate: pentium4: pentium4
+buildarchtranslate: pentium3: pentium3
+buildarchtranslate: i686: i686
+buildarchtranslate: i586: i586
+buildarchtranslate: i486: i486
 buildarchtranslate: i386: i386
 
 buildarchtranslate: alphaev5: alpha
@@ -217,7 +217,8 @@
 buildarchtranslate: sparcv8: sparc
 buildarchtranslate: sparcv9: sparc
 buildarchtranslate: sparcv9v: sparc
-buildarchtranslate: sun4u: sparc64
+buildarchtranslate: sun4u: sparc
+buildarchtranslate: sparc64: sparc
 buildarchtranslate: sparc64v: sparc64
 
 buildarchtranslate: osfmach3_ppc: ppc
