--- rpm-4.9.0/scripts/find-debuginfo.sh.wiget	2011-03-25 18:57:10.367956883 +0100
+++ rpm-4.9.0/scripts/find-debuginfo.sh	2011-03-25 19:05:58.372477326 +0100
@@ -309,6 +322,9 @@ pattern_list()
   sed -n -r "\#^$ptn #s/ .*\$//p" "$LINKSFILE" | append_uniq "$out"
 }
 
+# get rid of top level dir, we want it be only in filesystem-debuginfo package
+sed -i -e '/^%dir \/usr\/lib\/debug$/d' "$LISTFILE"
+
 #
 # When given multiple -o switches, split up the output as directed.
 #
