diff -Nur rpm-4.4.6.orig/macros.in rpm-4.4.6.macros/macros.in
--- rpm-4.4.6.orig/macros.in	2006-08-05 20:52:48.000000000 +0200
+++ rpm-4.4.6.macros/macros.in	2006-08-05 20:53:57.000000000 +0200
@@ -219,7 +219,7 @@
 %_tmppath		%{_var}/tmp
 
 #	Path to top of build area.
-%_topdir		%{_usrsrc}/rpm
+%_topdir		%(echo $HOME)/rpm
 
 #	The path to the unzip executable (legacy, use %{__unzip} instead).
 %_unzipbin		%{__unzip}
@@ -330,8 +330,8 @@
 #		"w9.gzdio"	gzip level 9 (default).
 #		"w9.bzdio"	bzip2 level 9.
 #
-#%_source_payload	w9.gzdio
-#%_binary_payload	w9.gzdio
+%_source_payload	w9.gzdio
+%_binary_payload	w9.lzdio
 
 #	Archive formats to use for source/binary package payloads.
 #		"cpio"		cpio archive (default)
@@ -909,10 +909,10 @@
   export RPM_BUILD_ROOT}\
   %{?_javaclasspath:CLASSPATH=\"%{_javaclasspath}\"\
   export CLASSPATH}\
-  unset PERL_MM_OPT\
+  unset PERL_MM_OPT || :\
   LANG=C\
   export LANG\
-  unset DISPLAY\
+  unset DISPLAY || :\
   \
   %{verbose:set -x}%{!verbose:exec > /dev/null}\
   umask 022\
@@ -1350,5 +1350,22 @@
 #%__executable_provides	@RPMCONFIGDIR@/executabledeps.sh --provides
 #%__executable_requires	@RPMCONFIGDIR@/executabledeps.sh --requires
 
+# PLD rpm macros
+%_enable_debug_packages 1
+
+#-----------------------------------------------------------------
+# CFLAGS and LDFLAGS used to build
+
+%debuginfocflags	%{expand:%%define __dic_%{?_enable_debug_packages} 1}%{?__dic_1: -ggdb}%{expand:%%undefine __dic_%{?_enable_debug_packages}}
+
+%debugcflags	-O0 -g -Wall
+%optldflags	-Wl,--as-needed
+
+# Warning: those macros are overwritten by macros.build,
+#          left here for compatibility
+%rpmcflags	%{?debug:%debugcflags}%{!?debug:%optflags}%{?debuginfocflags}
+%rpmcxxflags	%{rpmcflags}
+%rpmldflags	%{!?no_build_with_as_needed:-Wl,--as-needed}
+
 # \endverbatim
 #*/
