diff -Nru rpm-4.0/rpmpopt.in rpm-4.0.new/rpmpopt.in
--- rpm-4.0/rpmpopt.in	Wed Nov 15 19:18:35 2000
+++ rpm-4.0.new/rpmpopt.in	Wed Nov 15 19:18:15 2000
@@ -80,6 +80,14 @@
 # Build policies enabled from command line. Last policy applies.
 rpm alias --buildpolicy --define '__os_install_post @RPMCONFIGDIR@/brp-!#:+'
 
+# Make RPM build tree
+rpm exec --install-build-tree install-build-tree
+
+# Compiling with debuginfo may be enabled by --debug
+rpm alias --debug --define 'debug 1'
+
+# Don't remove $BUILDROOT if --noclean
+
 # Choose db interface:
 #	0	same as 1
 #	1	native db1 interface (e.g. linux glibc libdb1 routines).
