--- rpm-4.2/configure.ac.orig	2003-05-15 10:53:44.000000000 +0200
+++ rpm-4.2/configure.ac	2003-05-17 13:00:00.000000000 +0200
@@ -897,6 +897,18 @@
     AC_MSG_RESULT($withval)
     if test $withval = yes ; then
       WITH_PYTHON_VERSION="1.5"
+    else
+	  AC_MSG_CHECKING(for python 2.3)
+    AC_TRY_RUN([
+	#include <python2.3/Python.h>
+	main() {
+	 exit(strncmp("2.3", PY_VERSION, 3));
+	 } ],
+	withval=yes, withval=no, withval=yes)
+	AC_MSG_RESULT($withval)
+	if test $withval = yes ; then
+	WITH_PYTHON_VERSION="2.3"
+	fi
     fi
   fi
 
--- rpm-4.2/python/Makefile.am.orig	2003-05-15 10:49:57.000000000 +0200
+++ rpm-4.2/python/Makefile.am	2003-05-17 13:04:14.000000000 +0200
@@ -6,7 +6,7 @@
 
 PYVER= @WITH_PYTHON_VERSION@
 
-pylibdir = $(shell python -c 'import sys; print sys.path[1]')
+pylibdir = $(prefix)/lib/python${PYVER}
 pyincdir = $(prefix)/include/python${PYVER}
 
 SUBDIRS = rpmdb test
--- rpm-4.2/python/rpmdb/Makefile.am~	2003-01-09 00:21:38.000000000 +0100
+++ rpm-4.2/python/rpmdb/Makefile.am	2003-05-17 13:03:53.000000000 +0200
@@ -4,7 +4,7 @@
 
 PYVER= @WITH_PYTHON_VERSION@
 
-pylibdir = $(shell python -c 'import sys; print sys.path[1]')
+pylibdir = $(prefix)/lib/python${PYVER}
 
 EXTRA_DIST = \
 	__init__.py dbobj.py db.py dbrecio.py dbshelve.py dbtables.py dbutils.py
