--- rpm-5.4.10/lib/query.c~	2012-05-08 02:06:44.000000000 +0200
+++ rpm-5.4.10/lib/query.c	2012-09-21 13:42:12.013607913 +0200
@@ -732,7 +732,7 @@
 	qva->qva_mi = rpmtsInitIterator(ts, RPMTAG_FILEPATHS, fn, 0);
 	if (qva->qva_mi == NULL && !provides_checked && !gotpattern) {
 	    qva->qva_mi = rpmtsInitIterator(ts, RPMTAG_PROVIDENAME, fn, 0);
-#if defined(RPM_VENDOR_MANDRIVA)
+#if defined(RPM_VENDOR_MANDRIVA) || defined(RPM_VENDOR_PLD)
 	if(rpmmiCount(qva->qva_mi) == 0)
 	    qva->qva_mi = rpmmiFree(qva->qva_mi);
 #endif
