--- rpm-5.4.16/macros/macros.in~	2016-03-16 00:27:05.000000000 +0200
+++ rpm-5.4.16/macros/macros.in	2016-03-16 00:27:42.513047506 +0200
@@ -561,9 +561,10 @@
 
 # Horowitz Key Protocol server configuration
 #
-%_hkp_keyserver         hkp://keys.rpm5.org
+#%_hkp_keyserver         hkp://keys.rpm5.org
 #%_hkp_keyserver         hkp://keys.n3npq.net
 #%_hkp_keyserver         hkp://pool.sks-keyservers.net
+%_hkp_keyserver         hkp://ha.pool.sks-keyservers.net
 %_hkp_keyserver_query   %{_hkp_keyserver}/pks/lookup?op=get&search=
 

@@ -698,23 +698,7 @@
 # XXX	Note: that there cannot be any whitespace within the string "p>q",
 #	and that both p and q are package names (i.e. no version/release).
 #
-%_dependency_whiteout_caos_core		\
-	perl>perl-Filter		\
-	pam>coreutils			\
-	pam>initscripts			\
-	glibc-common>glibc		\
-	glibc>nscd			\
-	filesystem>setup
-
-%_dependency_whiteout_OWL31		\
-	dhcp-server>dhcp		\
-	which>bash			\
-
-
-%_dependency_whiteout			\
-	%{?_dependency_whiteout_OWL31} \
-	%{?_dependency_whiteout_system} \
-	%{nil}
+%_dependency_whiteout	%{nil}
 
 #
 # Default path used for serializing transactions with a  fcntl lock.
