--- rpm-4.5/Makefile.am~	2008-06-10 01:48:59.000000000 +0300
+++ rpm-4.5/Makefile.am	2009-10-15 00:00:46.263391619 +0300
@@ -51,6 +51,15 @@
 
 noinst_HEADERS = build.h debug.h system.h
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = rpm.pc
+rpm.pc: $(top_srcdir)/rpm.pc.in
+	@$(SED) \
+	  -e "s,[@]RPMCONFIGDIR[@],$(rpmconfigdir),g" \
+	  < $(top_srcdir)/rpm.pc.in > rpm.pc.tmp \
+	  && ( cd $(top_builddir) && ./config.status --file=${subdir}/rpm.pc:${subdir}/rpm.pc.tmp )
+EXTRA_DIST += rpm.pc.in
+
 rpm_SOURCES =		$(top_srcdir)/build.c
 rpm_LDFLAGS =		$(myLDFLAGS) $(staticLDFLAGS)
 rpm_LDADD =		rpm.o $(top_builddir)/build/.libs/librpmbuild.a $(myLDADD)
--- /dev/null	2007-02-13 18:29:53.000000000 +0200
+++ rpm/rpm.pc.in	2009-10-14 23:59:01.260069165 +0300
@@ -0,0 +1,15 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+rpmhome=@RPMCONFIGDIR@
+
+Name: RPM
+Description: RPM Package Manager
+Version: @VERSION@
+URL: http://rpm5.org
+# Requires: 
+# Conflicts:
+Cflags: -I${includedir}
+Libs: -L${libdir} -lrpm -lrpmio
+Libs.private: -lpopt -lrt -lpthread @WITH_SELINUX_LIB@ @WITH_SQLITE3_LIB@ @WITH_DB_LIB@ @WITH_BZ2_LIB@ @WITH_ZLIB_LIB@ @WITH_NSS_LIB@ @WITH_LUA_LIB@
