--- rpm-5.4.10/lib/verify.c.orig	2012-09-22 18:33:59.791810053 +0200
+++ rpm-5.4.10/lib/verify.c	2012-09-22 18:33:34.921810920 +0200
@@ -594,7 +594,7 @@
 	    continue;
 
 	/* If not verifying %ghost, skip ghost files. */
-	if (!FF_ISSET(qva->qva_fflags, GHOST) && FF_ISSET(fflags, GHOST))
+	if (FF_ISSET(qva->qva_fflags, GHOST) && FF_ISSET(fflags, GHOST))
 	    continue;
 
 	/* Gather per-file data into a carrier. */
