--- rpm-5.4.10/lib/rpmfc.c.old	2012-09-25 22:34:46.184039778 +0200
+++ rpm-5.4.10/lib/rpmfc.c	2012-09-25 22:35:21.947179368 +0200
@@ -921,7 +921,7 @@ static int rpmfcSCRIPT(rpmfc fc)
 #endif
 	    xx = rpmfcHelper(fc, 'R', "ruby");
     } else
-    if (fc->fcolor->vals[fc->ix] & RPMFC_FONT) {
+    if ((fc->fcolor->vals[fc->ix] & RPMFC_FONT) == RPMFC_FONT) {
        xx = rpmfcHelper(fc, 'P', "font");
        /* XXX: currently of no use, but for the sake of consistency... */
        xx = rpmfcHelper(fc, 'R', "font");
