--- rpm-5.4.10/scripts/rubygems.rb~	2013-04-26 00:58:58.622064212 +0300
+++ rpm-5.4.10/scripts/rubygems.rb	2013-04-26 11:42:07.842839682 +0300
@@ -39,8 +39,9 @@
   # way fugly, but we make the assumption that if the package has
   # this file, the package is the current ruby version, and should
   # therefore provide ruby(abi) = version
   if provides and path.match(RbConfig::CONFIG["archdir"] + "/rbconfig.rb")
      abi_provide = true
+     ruby_versioned = true
   elsif path.match(specpatt)
     ruby_versioned = true
     gems.push(path.chomp)
