--- rpm-4.16.0/build/build.c~	2020-05-28 12:04:24.000000000 +0200
+++ rpm-4.16.0/build/build.c	2020-10-31 18:57:03.678941782 +0100
@@ -401,7 +401,7 @@
 		goto exit;
 
 	if (((what & RPMBUILD_PACKAGEBINARY) && !test) &&
-	    (rc = packageBinaries(spec, cookie, (didBuild == 0))))
+	    (rc = packageBinaries(spec, cookie, ((didBuild == 0) && !rpmExpandNumeric("%{?disable_short_circuited_deps}")))))
 		goto exit;
 	
 	if ((what & RPMBUILD_CLEAN) &&
