--- rpm-4.16.1.2/build/rpmfc.cc~	2021-01-24 09:43:15.000000000 +0100
+++ rpm-4.16.1.2/build/rpmfc.cc	2021-02-19 09:14:46.031544586 +0100
@@ -1231,7 +1231,7 @@
 	int extension_index = 0;
 	int fcolor = RPMFC_BLACK;
 	rpm_mode_t mode = (fmode ? fmode[ix] : 0);
-	int is_executable = (mode & (S_IXUSR|S_IXGRP|S_IXOTH));
+	int is_executable = ((mode & S_IFREG) && (mode & (S_IXUSR|S_IXGRP|S_IXOTH)));
 
 	switch (mode & S_IFMT) {
 	case S_IFCHR:	ftype = "character special";	break;
