--- rpm-5.4.18/configure.ac~	2017-07-18 16:11:41.000000000 +0300
+++ rpm-5.4.18/configure.ac	2017-07-18 16:14:03.712760045 +0300
@@ -1872,6 +1872,9 @@ AC_ARG_WITH(
 	    "fedora")
 		    AC_DEFINE([RPM_VENDOR_FEDORA], 1, Vendor is Fedora)
 		    ;;
+	    "pld")
+		    AC_DEFINE([RPM_VENDOR_PLD], 1, Vendor is PLD)
+		    ;;
 	    esac
 	    AC_MSG_RESULT([$withval])
     else
@@ -6264,6 +6264,7 @@
     rpmdb/DB_CONFIG
     macros/macros macros/macros.rpmbuild
     macros/cmake macros/gstreamer macros/java macros/kernel macros/libtool
+    macros/pld
     macros/mandriva macros/suse macros/fedora macros/mono macros/mqtt
     macros/perl macros/pkgconfig macros/php
     macros/python macros/ruby macros/selinux macros/tcl
--- rpm-5.4.10/Makefile.am~	2012-03-17 01:24:19.000000000 +0100
+++ rpm-5.4.10/Makefile.am	2012-09-03 00:08:35.599623394 +0200
@@ -117,7 +117,7 @@
 pkgcfgdir =	$(pkgdatadir)/macros.d
 pkgcfg_DATA = \
 	macros/cmake macros/gstreamer macros/java macros/kernel macros/libtool \
-	macros/mandriva macros/mono macros/perl macros/pkgconfig macros/php \
+	macros/pld macros/mono macros/perl macros/pkgconfig macros/php \
 	macros/python macros/ruby macros/selinux macros/tcl
 
 noinst_HEADERS = build.h debug.h system.h
